{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd\n",
    "import os \n",
    "import networkx as nx\n",
    "import obonet\n",
    "from Bio.SeqRecord import SeqRecord\n",
    "from Bio.Seq import Seq\n",
    "from Bio import SeqIO\n",
    "from collections import defaultdict\n",
    "from sklearn.model_selection import ShuffleSplit\n",
    "from sklearn.preprocessing import MultiLabelBinarizer\n",
    "from tqdm import tqdm\n",
    "from collections import Counter\n",
    "import subprocess\n",
    "import requests, sys\n",
    "import json"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "np.random.seed(42)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Read in the raw datasets"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "\n",
    "# without this mmseqs isn't recognized as a module\n",
    "\n",
    "os.environ[\"PATH\"] = (\n",
    "    \"/shared/ifbstor1/projects/deepmar/conda_environments/conda_envs/PlasmoFP/bin:\"\n",
    "    + os.environ[\"PATH\"]\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Uniprot Data: (329022, 9)\n"
     ]
    }
   ],
   "source": [
    "TSV_path = \"../uniprot_taxonomy_bacteria_20_11_2025.tsv\"\n",
    "\n",
    "#read in the datasets\n",
    "uniprot = pd.read_csv(TSV_path, sep=\"\\t\")\n",
    "print(\"Uniprot Data:\", uniprot.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "def create_seq_records(data):\n",
    "    seq_records = []\n",
    "    for index, row in data.iterrows():\n",
    "        seq_record = SeqRecord(Seq(row['Sequence']), id=row['Entry'], description=row['Organism'])\n",
    "        seq_records.append(seq_record)\n",
    "    assert len(seq_records) == data.shape[0]\n",
    "    return seq_records"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "uniprot[\"GOAssertion\"] = \"AA\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(301300, 10)\n"
     ]
    }
   ],
   "source": [
    "total_bacteria_function = uniprot[uniprot[\"Gene Ontology (molecular function)\"].notnull()]\n",
    "\n",
    "print(total_bacteria_function.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "329022\n"
     ]
    }
   ],
   "source": [
    "total_bacteria_records = create_seq_records(uniprot)\n",
    "print(len(total_bacteria_records))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "329022"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "SeqIO.write(total_bacteria_records, \"total_bacteria.fasta\", \"fasta\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "seq_records_bacteria_function = create_seq_records(total_bacteria_function)\n",
    "\n",
    "assert len(seq_records_bacteria_function) == total_bacteria_function.shape[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "301300"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "SeqIO.write(seq_records_bacteria_function, \"total_bacteria_function_for_reduce.fasta\", \"fasta\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/shared/ifbstor1/projects/deepmar/conda_environments/conda_envs/PlasmoFP/bin/mmseqs'"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import shutil\n",
    "shutil.which(\"mmseqs\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "! mkdir reduced_90"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true,
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Creating databases...\n",
      "function/function_redundant_db exists and will be overwritten\n",
      "createdb /shared/projects/deepmar/data/bacteria_sequences/function_only/total_bacteria_function_for_reduce.fasta function/function_redundant_db \n",
      "\n",
      "MMseqs Version:                    \t18.8cc5c\n",
      "Database type                      \t0\n",
      "Shuffle input database             \ttrue\n",
      "Createdb mode                      \t0\n",
      "Write lookup file                  \t1\n",
      "Offset of numeric ids              \t0\n",
      "Threads                            \t256\n",
      "Compressed                         \t0\n",
      "Mask residues                      \t0\n",
      "Mask residues probability          \t0.9\n",
      "Mask lower case residues           \t0\n",
      "Mask lower letter repeating N times\t0\n",
      "Use GPU                            \t0\n",
      "Verbosity                          \t3\n",
      "\n",
      "Converting sequences\n",
      "[==============================\n",
      "Time for merging to function_redundant_db_h: 0h 0m 0s 113ms\n",
      "Time for merging to function_redundant_db: 0h 0m 0s 221ms\n",
      "Database type: Aminoacid\n",
      "Time for processing: 0h 0m 1s 135ms\n",
      "component/component_redundant_db exists and will be overwritten\n",
      "createdb /shared/projects/deepmar/data/bacteria_sequences/function_only/empty.fasta component/component_redundant_db \n",
      "\n",
      "MMseqs Version:                    \t18.8cc5c\n",
      "Database type                      \t0\n",
      "Shuffle input database             \ttrue\n",
      "Createdb mode                      \t0\n",
      "Write lookup file                  \t1\n",
      "Offset of numeric ids              \t0\n",
      "Threads                            \t256\n",
      "Compressed                         \t0\n",
      "Mask residues                      \t0\n",
      "Mask residues probability          \t0.9\n",
      "Mask lower case residues           \t0\n",
      "Mask lower letter repeating N times\t0\n",
      "Use GPU                            \t0\n",
      "Verbosity                          \t3\n",
      "\n",
      "Converting sequences\n",
      "[\n",
      "Time for merging to component_redundant_db_h: 0h 0m 0s 70ms\n",
      "Time for merging to component_redundant_db: 0h 0m 0s 65ms\n",
      "Database type: Aminoacid\n",
      "Time for processing: 0h 0m 0s 394ms\n",
      "createdb /shared/projects/deepmar/data/bacteria_sequences/function_only/empty.fasta process/process_redundant_db \n",
      "\n",
      "MMseqs Version:                    \t18.8cc5c\n",
      "Database type                      \t0\n",
      "Shuffle input database             \ttrue\n",
      "Createdb mode                      \t0\n",
      "Write lookup file                  \t1\n",
      "Offset of numeric ids              \t0\n",
      "Threads                            \t256\n",
      "Compressed                         \t0\n",
      "Mask residues                      \t0\n",
      "Mask residues probability          \t0.9\n",
      "Mask lower case residues           \t0\n",
      "Mask lower letter repeating N times\t0\n",
      "Use GPU                            \t0\n",
      "Verbosity                          \t3\n",
      "\n",
      "Converting sequences\n",
      "[\n",
      "Time for merging to process_redundant_db_h: 0h 0m 0s 60ms\n",
      "Time for merging to process_redundant_db: 0h 0m 0s 59ms\n",
      "Database type: Aminoacid\n",
      "Time for processing: 0h 0m 0s 294ms\n",
      "Clustering sequences...\n",
      "Create directory tmp\n",
      "cluster function/function_redundant_db function/function_clusters_redundant tmp --min-seq-id 0.9 -c 0.5 --cov-mode 1 \n",
      "\n",
      "MMseqs Version:                     \t18.8cc5c\n",
      "Substitution matrix                 \taa:blosum62.out,nucl:nucleotide.out\n",
      "Seed substitution matrix            \taa:VTML80.out,nucl:nucleotide.out\n",
      "Sensitivity                         \t4\n",
      "k-mer length                        \t0\n",
      "Target search mode                  \t0\n",
      "k-score                             \tseq:2147483647,prof:2147483647\n",
      "Alphabet size                       \taa:21,nucl:5\n",
      "Max sequence length                 \t65535\n",
      "Max results per query               \t20\n",
      "Split database                      \t0\n",
      "Split mode                          \t2\n",
      "Split memory limit                  \t0\n",
      "Coverage threshold                  \t0.5\n",
      "Coverage mode                       \t1\n",
      "Compositional bias                  \t1\n",
      "Compositional bias scale            \t1\n",
      "Diagonal scoring                    \ttrue\n",
      "Exact k-mer matching                \t0\n",
      "Mask residues                       \t1\n",
      "Mask residues probability           \t0.9\n",
      "Mask lower case residues            \t0\n",
      "Mask lower letter repeating N times \t0\n",
      "Minimum diagonal score              \t15\n",
      "Selected taxa                       \t\n",
      "Include identical seq. id.          \tfalse\n",
      "Spaced k-mers                       \t1\n",
      "Preload mode                        \t0\n",
      "Pseudo count a                      \tsubstitution:1.100,context:1.400\n",
      "Pseudo count b                      \tsubstitution:4.100,context:5.800\n",
      "Spaced k-mer pattern                \t\n",
      "Local temporary path                \t\n",
      "Threads                             \t256\n",
      "Compressed                          \t0\n",
      "Verbosity                           \t3\n",
      "Add backtrace                       \tfalse\n",
      "Alignment mode                      \t3\n",
      "Alignment mode                      \t0\n",
      "Allow wrapped scoring               \tfalse\n",
      "E-value threshold                   \t0.001\n",
      "Seq. id. threshold                  \t0.9\n",
      "Min alignment length                \t0\n",
      "Seq. id. mode                       \t0\n",
      "Alternative alignments              \t0\n",
      "Max reject                          \t2147483647\n",
      "Max accept                          \t2147483647\n",
      "Score bias                          \t0\n",
      "Realign hits                        \tfalse\n",
      "Realign score bias                  \t-0.2\n",
      "Realign max seqs                    \t2147483647\n",
      "Correlation score weight            \t0\n",
      "Gap open cost                       \taa:11,nucl:5\n",
      "Gap extension cost                  \taa:1,nucl:2\n",
      "Zdrop                               \t40\n",
      "Rescore mode                        \t0\n",
      "Remove hits by seq. id. and coverage\tfalse\n",
      "Sort results                        \t0\n",
      "Cluster mode                        \t0\n",
      "Max connected component depth       \t1000\n",
      "Similarity type                     \t2\n",
      "Weight file name                    \t\n",
      "Cluster Weight threshold            \t0.9\n",
      "Single step clustering              \tfalse\n",
      "Cascaded clustering steps           \t3\n",
      "Cluster reassign                    \tfalse\n",
      "Remove temporary files              \tfalse\n",
      "Force restart with latest tmp       \tfalse\n",
      "MPI runner                          \t\n",
      "k-mers per sequence                 \t21\n",
      "Scale k-mers per sequence           \taa:0.000,nucl:0.200\n",
      "Adjust k-mer length                 \tfalse\n",
      "Shift hash                          \t67\n",
      "Include only extendable             \tfalse\n",
      "Skip repeating k-mers               \tfalse\n",
      "\n",
      "Set cluster sensitivity to -s 1.000000\n",
      "Set cluster mode GREEDY MEM\n",
      "Set cluster iterations to 1\n",
      "linclust function/function_redundant_db tmp/13263377272415860673/clu_redundancy tmp/13263377272415860673/linclust --cluster-mode 3 --max-iterations 1000 --similarity-type 2 --threads 256 --compressed 0 -v 3 --cluster-weight-threshold 0.9 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.9 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.5 --cov-mode 1 --max-seq-len 65535 --comp-bias-corr 0 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --alph-size aa:13,nucl:5 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 --mask-n-repeat 0 -k 0 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --rescore-mode 0 --filter-hits 0 --sort-results 0 --remove-tmp-files 0 --force-reuse 0 \n",
      "\n",
      "kmermatcher function/function_redundant_db tmp/13263377272415860673/linclust/7194583878587474280/pref --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --alph-size aa:13,nucl:5 --min-seq-id 0.9 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 --mask-n-repeat 0 --cov-mode 1 -k 0 -c 0.5 --max-seq-len 65535 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --threads 256 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "kmermatcher function/function_redundant_db tmp/13263377272415860673/linclust/7194583878587474280/pref --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --alph-size aa:13,nucl:5 --min-seq-id 0.9 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 --mask-n-repeat 0 --cov-mode 1 -k 0 -c 0.5 --max-seq-len 65535 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --threads 256 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Database size: 301300 type: Aminoacid\n",
      "Reduced amino acid alphabet: (A S T) (C) (D B N) (E Q Z) (F Y) (G) (H) (I V) (K R) (L J M) (P) (W) (X) \n",
      "\n",
      "Generate k-mers list for 1 split\n",
      "[=================================================================] 301.30K 2s 779ms\n",
      "Sort kmer 0h 0m 0s 843ms\n",
      "Sort by rep. sequence 0h 0m 0s 446ms\n",
      "Time for fill: 0h 0m 0s 58ms\n",
      "Time for merging to pref: 0h 0m 0s 2ms\n",
      "Time for processing: 0h 0m 4s 389ms\n",
      "rescorediagonal function/function_redundant_db function/function_redundant_db tmp/13263377272415860673/linclust/7194583878587474280/pref tmp/13263377272415860673/linclust/7194583878587474280/pref_rescore1 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --rescore-mode 0 --wrapped-scoring 0 --filter-hits 0 -e 0.001 -c 0.5 -a 0 --cov-mode 1 --min-seq-id 0.9 --min-aln-len 0 --seq-id-mode 0 --add-self-matches 0 --sort-results 0 --db-load-mode 0 --threads 256 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 301.30K 0s 511ms\n",
      "Time for merging to pref_rescore1: 0h 0m 0s 405ms\n",
      "Time for processing: 0h 0m 1s 722ms\n",
      "clust function/function_redundant_db tmp/13263377272415860673/linclust/7194583878587474280/pref_rescore1 tmp/13263377272415860673/linclust/7194583878587474280/pre_clust --cluster-mode 3 --max-iterations 1000 --similarity-type 2 --threads 256 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Greedy Low Mem\n",
      "Total time: 0h 0m 0s 200ms\n",
      "\n",
      "Size of the sequence database: 301300\n",
      "Size of the alignment database: 301300\n",
      "Number of clusters: 155280\n",
      "\n",
      "Writing results 0h 0m 0s 21ms\n",
      "Time for merging to pre_clust: 0h 0m 0s 2ms\n",
      "Time for processing: 0h 0m 0s 416ms\n",
      "createsubdb tmp/13263377272415860673/linclust/7194583878587474280/order_redundancy function/function_redundant_db tmp/13263377272415860673/linclust/7194583878587474280/input_step_redundancy -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step_redundancy: 0h 0m 0s 3ms\n",
      "Time for processing: 0h 0m 0s 57ms\n",
      "createsubdb tmp/13263377272415860673/linclust/7194583878587474280/order_redundancy tmp/13263377272415860673/linclust/7194583878587474280/pref tmp/13263377272415860673/linclust/7194583878587474280/pref_filter1 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to pref_filter1: 0h 0m 0s 2ms\n",
      "Time for processing: 0h 0m 0s 66ms\n",
      "filterdb tmp/13263377272415860673/linclust/7194583878587474280/pref_filter1 tmp/13263377272415860673/linclust/7194583878587474280/pref_filter2 --filter-file tmp/13263377272415860673/linclust/7194583878587474280/order_redundancy --threads 256 --compressed 0 -v 3 \n",
      "\n",
      "Filtering using file(s)\n",
      "[=================================================================] 155.28K 0s 211ms\n",
      "Time for merging to pref_filter2: 0h 0m 0s 384ms\n",
      "Time for processing: 0h 0m 1s 321ms\n",
      "rescorediagonal tmp/13263377272415860673/linclust/7194583878587474280/input_step_redundancy tmp/13263377272415860673/linclust/7194583878587474280/input_step_redundancy tmp/13263377272415860673/linclust/7194583878587474280/pref_filter2 tmp/13263377272415860673/linclust/7194583878587474280/pref_rescore2 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --rescore-mode 1 --wrapped-scoring 0 --filter-hits 1 -e 0.001 -c 0.5 -a 0 --cov-mode 1 --min-seq-id 0.9 --min-aln-len 0 --seq-id-mode 0 --add-self-matches 0 --sort-results 0 --db-load-mode 0 --threads 256 --compressed 0 -v 3 \n",
      "\n",
      "[================================================================] 155.28K =0s 360ms\n",
      "Time for merging to pref_rescore2: 0h 0m 0s 408ms\n",
      "Time for processing: 0h 0m 1s 557ms\n",
      "align tmp/13263377272415860673/linclust/7194583878587474280/input_step_redundancy tmp/13263377272415860673/linclust/7194583878587474280/input_step_redundancy tmp/13263377272415860673/linclust/7194583878587474280/pref_rescore2 tmp/13263377272415860673/linclust/7194583878587474280/aln --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.9 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.5 --cov-mode 1 --max-seq-len 65535 --comp-bias-corr 0 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 256 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 155280 type: Aminoacid\n",
      "Target database size: 155280 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 155.28K 33s 933ms\n",
      "Time for merging to aln: 0h 0m 0s 402ms\n",
      "361089 alignments calculated\n",
      "158885 sequence pairs passed the thresholds (0.440016 of overall calculated)\n",
      "1.023216 hits per query sequence\n",
      "Time for processing: 0h 0m 35s 276ms\n",
      "clust tmp/13263377272415860673/linclust/7194583878587474280/input_step_redundancy tmp/13263377272415860673/linclust/7194583878587474280/aln tmp/13263377272415860673/linclust/7194583878587474280/clust --cluster-mode 3 --max-iterations 1000 --similarity-type 2 --threads 256 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Greedy Low Mem\n",
      "Total time: 0h 0m 0s 75ms\n",
      "\n",
      "Size of the sequence database: 155280\n",
      "Size of the alignment database: 155280\n",
      "Number of clusters: 152045\n",
      "\n",
      "Writing results 0h 0m 0s 19ms\n",
      "Time for merging to clust: 0h 0m 0s 2ms\n",
      "Time for processing: 0h 0m 0s 231ms\n",
      "mergeclusters function/function_redundant_db tmp/13263377272415860673/clu_redundancy tmp/13263377272415860673/linclust/7194583878587474280/pre_clust tmp/13263377272415860673/linclust/7194583878587474280/clust --threads 256 --compressed 0 -v 3 \n",
      "\n",
      "Clustering step 1\n",
      "[=================================================================] 155.28K 0s 49ms\n",
      "Clustering step 2\n",
      "[=================================================================] 152.04K 0s 89ms\n",
      "Write merged clustering\n",
      "[=================================================================] 301.30K 0s 575ms\n",
      "Time for merging to clu_redundancy: 0h 0m 0s 409ms\n",
      "Time for processing: 0h 0m 1s 135ms\n",
      "createsubdb tmp/13263377272415860673/clu_redundancy function/function_redundant_db tmp/13263377272415860673/input_step_redundancy -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step_redundancy: 0h 0m 0s 2ms\n",
      "Time for processing: 0h 0m 0s 54ms\n",
      "prefilter tmp/13263377272415860673/input_step_redundancy tmp/13263377272415860673/input_step_redundancy tmp/13263377272415860673/pref_step0 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 1 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.5 --cov-mode 1 --comp-bias-corr 0 --comp-bias-corr-scale 1 --diag-score 0 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --mask-n-repeat 0 --min-ungapped-score 0 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 256 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 152045 type: Aminoacid\n",
      "Estimated memory consumption: 3G\n",
      "Target database size: 152045 type: Aminoacid\n",
      "Index table k-mer threshold: 154 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 152.04K 3s 884ms\n",
      "Index table: Masked residues: 178096\n",
      "Index table: fill\n",
      "[=================================================================] 152.04K 3s 27ms\n",
      "Index statistics\n",
      "Entries:          20598756\n",
      "DB size:          606 MB\n",
      "Avg k-mer size:   0.321856\n",
      "Top 10 k-mers\n",
      "    GPGGTL\t858\n",
      "    PGVFEV\t602\n",
      "    SRRWPI\t496\n",
      "    IDPHFT\t485\n",
      "    GGQRIA\t439\n",
      "    LRRGGD\t409\n",
      "    RRDQGR\t409\n",
      "    IMHDKT\t408\n",
      "    HVHKLL\t406\n",
      "    ATMGTD\t393\n",
      "Time for index table init: 0h 0m 7s 481ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 154\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 152045\n",
      "Target db start 1 to 152045\n",
      "[=================================================================] 152.04K 10s 499ms\n",
      "\n",
      "1.401256 k-mers per position\n",
      "1682 DB matches per sequence\n",
      "0 overflows\n",
      "16 sequences passed prefiltering per query sequence\n",
      "20 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step0: 0h 0m 0s 368ms\n",
      "Time for processing: 0h 0m 20s 849ms\n",
      "align tmp/13263377272415860673/input_step_redundancy tmp/13263377272415860673/input_step_redundancy tmp/13263377272415860673/pref_step0 tmp/13263377272415860673/aln_step0 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.9 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.5 --cov-mode 1 --max-seq-len 65535 --comp-bias-corr 0 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 256 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 152045 type: Aminoacid\n",
      "Target database size: 152045 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 152.04K 7m 10s 147ms\n",
      "Time for merging to aln_step0: 0h 0m 0s 401ms\n",
      "2543569 alignments calculated\n",
      "155833 sequence pairs passed the thresholds (0.061265 of overall calculated)\n",
      "1.024914 hits per query sequence\n",
      "Time for processing: 0h 7m 11s 358ms\n",
      "clust tmp/13263377272415860673/input_step_redundancy tmp/13263377272415860673/aln_step0 tmp/13263377272415860673/clu_step0 --cluster-mode 3 --max-iterations 1000 --similarity-type 2 --threads 256 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Greedy Low Mem\n",
      "Total time: 0h 0m 0s 71ms\n",
      "\n",
      "Size of the sequence database: 152045\n",
      "Size of the alignment database: 152045\n",
      "Number of clusters: 150860\n",
      "\n",
      "Writing results 0h 0m 0s 18ms\n",
      "Time for merging to clu_step0: 0h 0m 0s 2ms\n",
      "Time for processing: 0h 0m 0s 230ms\n",
      "mergeclusters function/function_redundant_db function/function_clusters_redundant tmp/13263377272415860673/clu_redundancy tmp/13263377272415860673/clu_step0 --threads 256 --compressed 0 -v 3 \n",
      "\n",
      "Clustering step 1\n",
      "[=================================================================] 152.04K 0s 54ms\n",
      "Clustering step 2\n",
      "[=================================================================] 150.86K 0s 120ms\n",
      "Write merged clustering\n",
      "[=================================================================] 301.30K 0s 568ms\n",
      "Time for merging to function_clusters_redundant: 0h 0m 0s 431ms\n",
      "Time for processing: 0h 0m 1s 276ms\n",
      "cluster component/component_redundant_db component/component_clusters_redundant tmp --min-seq-id 0.9 -c 0.5 --cov-mode 1 \n",
      "\n",
      "MMseqs Version:                     \t18.8cc5c\n",
      "Substitution matrix                 \taa:blosum62.out,nucl:nucleotide.out\n",
      "Seed substitution matrix            \taa:VTML80.out,nucl:nucleotide.out\n",
      "Sensitivity                         \t4\n",
      "k-mer length                        \t0\n",
      "Target search mode                  \t0\n",
      "k-score                             \tseq:2147483647,prof:2147483647\n",
      "Alphabet size                       \taa:21,nucl:5\n",
      "Max sequence length                 \t65535\n",
      "Max results per query               \t20\n",
      "Split database                      \t0\n",
      "Split mode                          \t2\n",
      "Split memory limit                  \t0\n",
      "Coverage threshold                  \t0.5\n",
      "Coverage mode                       \t1\n",
      "Compositional bias                  \t1\n",
      "Compositional bias scale            \t1\n",
      "Diagonal scoring                    \ttrue\n",
      "Exact k-mer matching                \t0\n",
      "Mask residues                       \t1\n",
      "Mask residues probability           \t0.9\n",
      "Mask lower case residues            \t0\n",
      "Mask lower letter repeating N times \t0\n",
      "Minimum diagonal score              \t15\n",
      "Selected taxa                       \t\n",
      "Include identical seq. id.          \tfalse\n",
      "Spaced k-mers                       \t1\n",
      "Preload mode                        \t0\n",
      "Pseudo count a                      \tsubstitution:1.100,context:1.400\n",
      "Pseudo count b                      \tsubstitution:4.100,context:5.800\n",
      "Spaced k-mer pattern                \t\n",
      "Local temporary path                \t\n",
      "Threads                             \t256\n",
      "Compressed                          \t0\n",
      "Verbosity                           \t3\n",
      "Add backtrace                       \tfalse\n",
      "Alignment mode                      \t3\n",
      "Alignment mode                      \t0\n",
      "Allow wrapped scoring               \tfalse\n",
      "E-value threshold                   \t0.001\n",
      "Seq. id. threshold                  \t0.9\n",
      "Min alignment length                \t0\n",
      "Seq. id. mode                       \t0\n",
      "Alternative alignments              \t0\n",
      "Max reject                          \t2147483647\n",
      "Max accept                          \t2147483647\n",
      "Score bias                          \t0\n",
      "Realign hits                        \tfalse\n",
      "Realign score bias                  \t-0.2\n",
      "Realign max seqs                    \t2147483647\n",
      "Correlation score weight            \t0\n",
      "Gap open cost                       \taa:11,nucl:5\n",
      "Gap extension cost                  \taa:1,nucl:2\n",
      "Zdrop                               \t40\n",
      "Rescore mode                        \t0\n",
      "Remove hits by seq. id. and coverage\tfalse\n",
      "Sort results                        \t0\n",
      "Cluster mode                        \t0\n",
      "Max connected component depth       \t1000\n",
      "Similarity type                     \t2\n",
      "Weight file name                    \t\n",
      "Cluster Weight threshold            \t0.9\n",
      "Single step clustering              \tfalse\n",
      "Cascaded clustering steps           \t3\n",
      "Cluster reassign                    \tfalse\n",
      "Remove temporary files              \tfalse\n",
      "Force restart with latest tmp       \tfalse\n",
      "MPI runner                          \t\n",
      "k-mers per sequence                 \t21\n",
      "Scale k-mers per sequence           \taa:0.000,nucl:0.200\n",
      "Adjust k-mer length                 \tfalse\n",
      "Shift hash                          \t67\n",
      "Include only extendable             \tfalse\n",
      "Skip repeating k-mers               \tfalse\n",
      "\n",
      "Set cluster sensitivity to -s 1.000000\n",
      "Set cluster mode GREEDY MEM\n",
      "Set cluster iterations to 1\n",
      "linclust component/component_redundant_db tmp/17736517636862053666/clu_redundancy tmp/17736517636862053666/linclust --cluster-mode 3 --max-iterations 1000 --similarity-type 2 --threads 256 --compressed 0 -v 3 --cluster-weight-threshold 0.9 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.9 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.5 --cov-mode 1 --max-seq-len 65535 --comp-bias-corr 0 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --alph-size aa:13,nucl:5 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 --mask-n-repeat 0 -k 0 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --rescore-mode 0 --filter-hits 0 --sort-results 0 --remove-tmp-files 0 --force-reuse 0 \n",
      "\n",
      "kmermatcher component/component_redundant_db tmp/17736517636862053666/linclust/8668545910647190509/pref --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --alph-size aa:13,nucl:5 --min-seq-id 0.9 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 --mask-n-repeat 0 --cov-mode 1 -k 0 -c 0.5 --max-seq-len 65535 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --threads 256 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "kmermatcher component/component_redundant_db tmp/17736517636862053666/linclust/8668545910647190509/pref --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --alph-size aa:13,nucl:5 --min-seq-id 0.9 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 --mask-n-repeat 0 --cov-mode 1 -k 0 -c 0.5 --max-seq-len 65535 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --threads 256 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Database size: 1 type: Aminoacid\n",
      "Reduced amino acid alphabet: (A S T) (C) (D B N) (E Q Z) (F Y) (G) (H) (I V) (K R) (L J M) (P) (W) (X) \n",
      "\n",
      "Generate k-mers list for 1 split\n",
      "[=================================================================] 1 0s 0ms\n",
      "Sort kmer 0h 0m 0s 0ms\n",
      "Sort by rep. sequence 0h 0m 0s 0ms\n",
      "Time for fill: 0h 0m 0s 0ms\n",
      "Time for merging to pref: 0h 0m 0s 2ms\n",
      "Time for processing: 0h 0m 0s 33ms\n",
      "rescorediagonal component/component_redundant_db component/component_redundant_db tmp/17736517636862053666/linclust/8668545910647190509/pref tmp/17736517636862053666/linclust/8668545910647190509/pref_rescore1 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --rescore-mode 0 --wrapped-scoring 0 --filter-hits 0 -e 0.001 -c 0.5 -a 0 --cov-mode 1 --min-seq-id 0.9 --min-aln-len 0 --seq-id-mode 0 --add-self-matches 0 --sort-results 0 --db-load-mode 0 --threads 256 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 1 0s 7ms\n",
      "Time for merging to pref_rescore1: 0h 0m 0s 390ms\n",
      "Time for processing: 0h 0m 1s 24ms\n",
      "clust component/component_redundant_db tmp/17736517636862053666/linclust/8668545910647190509/pref_rescore1 tmp/17736517636862053666/linclust/8668545910647190509/pre_clust --cluster-mode 3 --max-iterations 1000 --similarity-type 2 --threads 256 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Greedy Low Mem\n",
      "Total time: 0h 0m 0s 14ms\n",
      "\n",
      "Size of the sequence database: 1\n",
      "Size of the alignment database: 1\n",
      "Number of clusters: 1\n",
      "\n",
      "Writing results 0h 0m 0s 0ms\n",
      "Time for merging to pre_clust: 0h 0m 0s 2ms\n",
      "Time for processing: 0h 0m 0s 131ms\n",
      "createsubdb tmp/17736517636862053666/linclust/8668545910647190509/order_redundancy component/component_redundant_db tmp/17736517636862053666/linclust/8668545910647190509/input_step_redundancy -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step_redundancy: 0h 0m 0s 2ms\n",
      "Time for processing: 0h 0m 0s 14ms\n",
      "createsubdb tmp/17736517636862053666/linclust/8668545910647190509/order_redundancy tmp/17736517636862053666/linclust/8668545910647190509/pref tmp/17736517636862053666/linclust/8668545910647190509/pref_filter1 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to pref_filter1: 0h 0m 0s 2ms\n",
      "Time for processing: 0h 0m 0s 11ms\n",
      "filterdb tmp/17736517636862053666/linclust/8668545910647190509/pref_filter1 tmp/17736517636862053666/linclust/8668545910647190509/pref_filter2 --filter-file tmp/17736517636862053666/linclust/8668545910647190509/order_redundancy --threads 256 --compressed 0 -v 3 \n",
      "\n",
      "Filtering using file(s)\n",
      "[=================================================================] 1 0s 8ms\n",
      "Time for merging to pref_filter2: 0h 0m 0s 390ms\n",
      "Time for processing: 0h 0m 0s 942ms\n",
      "rescorediagonal tmp/17736517636862053666/linclust/8668545910647190509/input_step_redundancy tmp/17736517636862053666/linclust/8668545910647190509/input_step_redundancy tmp/17736517636862053666/linclust/8668545910647190509/pref_filter2 tmp/17736517636862053666/linclust/8668545910647190509/pref_rescore2 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --rescore-mode 1 --wrapped-scoring 0 --filter-hits 1 -e 0.001 -c 0.5 -a 0 --cov-mode 1 --min-seq-id 0.9 --min-aln-len 0 --seq-id-mode 0 --add-self-matches 0 --sort-results 0 --db-load-mode 0 --threads 256 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 1 0s 8ms\n",
      "Time for merging to pref_rescore2: 0h 0m 0s 435ms\n",
      "Time for processing: 0h 0m 1s 269ms\n",
      "align tmp/17736517636862053666/linclust/8668545910647190509/input_step_redundancy tmp/17736517636862053666/linclust/8668545910647190509/input_step_redundancy tmp/17736517636862053666/linclust/8668545910647190509/pref_rescore2 tmp/17736517636862053666/linclust/8668545910647190509/aln --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.9 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.5 --cov-mode 1 --max-seq-len 65535 --comp-bias-corr 0 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 256 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 1 type: Aminoacid\n",
      "Target database size: 1 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 1 0s 0ms\n",
      "Time for merging to aln: 0h 0m 0s 3ms\n",
      "1 alignments calculated\n",
      "1 sequence pairs passed the thresholds (1.000000 of overall calculated)\n",
      "1.000000 hits per query sequence\n",
      "Time for processing: 0h 0m 0s 105ms\n",
      "clust tmp/17736517636862053666/linclust/8668545910647190509/input_step_redundancy tmp/17736517636862053666/linclust/8668545910647190509/aln tmp/17736517636862053666/linclust/8668545910647190509/clust --cluster-mode 3 --max-iterations 1000 --similarity-type 2 --threads 256 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Greedy Low Mem\n",
      "Total time: 0h 0m 0s 15ms\n",
      "\n",
      "Size of the sequence database: 1\n",
      "Size of the alignment database: 1\n",
      "Number of clusters: 1\n",
      "\n",
      "Writing results 0h 0m 0s 0ms\n",
      "Time for merging to clust: 0h 0m 0s 3ms\n",
      "Time for processing: 0h 0m 0s 25ms\n",
      "mergeclusters component/component_redundant_db tmp/17736517636862053666/clu_redundancy tmp/17736517636862053666/linclust/8668545910647190509/pre_clust tmp/17736517636862053666/linclust/8668545910647190509/clust --threads 256 --compressed 0 -v 3 \n",
      "\n",
      "Clustering step 1\n",
      "[=================================================================] 1 0s 7ms\n",
      "Clustering step 2\n",
      "[=================================================================] 1 0s 22ms\n",
      "Write merged clustering\n",
      "[=================================================================] 1 0s 562ms\n",
      "Time for merging to clu_redundancy: 0h 0m 0s 394ms\n",
      "Time for processing: 0h 0m 1s 50ms\n",
      "createsubdb tmp/17736517636862053666/clu_redundancy component/component_redundant_db tmp/17736517636862053666/input_step_redundancy -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step_redundancy: 0h 0m 0s 2ms\n",
      "Time for processing: 0h 0m 0s 14ms\n",
      "prefilter tmp/17736517636862053666/input_step_redundancy tmp/17736517636862053666/input_step_redundancy tmp/17736517636862053666/pref_step0 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 1 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.5 --cov-mode 1 --comp-bias-corr 0 --comp-bias-corr-scale 1 --diag-score 0 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --mask-n-repeat 0 --min-ungapped-score 0 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 256 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 1 type: Aminoacid\n",
      "Estimated memory consumption: 977M\n",
      "Target database size: 1 type: Aminoacid\n",
      "Index table k-mer threshold: 154 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 1 0s 0ms\n",
      "Index table: Masked residues: 0\n",
      "No k-mer could be extracted for the database tmp/17736517636862053666/input_step_redundancy.\n",
      "Maybe the sequences length is less than 14 residues.\n",
      "Index table: fill\n",
      "[=================================================================] 1 0s 0ms\n",
      "Index statistics\n",
      "Entries:          0\n",
      "DB size:          488 MB\n",
      "Avg k-mer size:   0.000000\n",
      "Top 10 k-mers\n",
      "    AAAAAA\t0\n",
      "    AAAAAA\t0\n",
      "    AAAAAA\t0\n",
      "    AAAAAA\t0\n",
      "    AAAAAA\t0\n",
      "    AAAAAA\t0\n",
      "    AAAAAA\t0\n",
      "    AAAAAA\t0\n",
      "    AAAAAA\t0\n",
      "    AAAAAA\t0\n",
      "Time for index table init: 0h 0m 0s 463ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 154\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 1\n",
      "Target db start 1 to 1\n",
      "[=================================================================] 1 0s 0ms\n",
      "\n",
      "0.000000 k-mers per position\n",
      "0 DB matches per sequence\n",
      "0 overflows\n",
      "1 sequences passed prefiltering per query sequence\n",
      "1 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step0: 0h 0m 0s 3ms\n",
      "Time for processing: 0h 0m 2s 134ms\n",
      "align tmp/17736517636862053666/input_step_redundancy tmp/17736517636862053666/input_step_redundancy tmp/17736517636862053666/pref_step0 tmp/17736517636862053666/aln_step0 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.9 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.5 --cov-mode 1 --max-seq-len 65535 --comp-bias-corr 0 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 256 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 1 type: Aminoacid\n",
      "Target database size: 1 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 1 0s 0ms\n",
      "Time for merging to aln_step0: 0h 0m 0s 3ms\n",
      "1 alignments calculated\n",
      "1 sequence pairs passed the thresholds (1.000000 of overall calculated)\n",
      "1.000000 hits per query sequence\n",
      "Time for processing: 0h 0m 0s 12ms\n",
      "clust tmp/17736517636862053666/input_step_redundancy tmp/17736517636862053666/aln_step0 tmp/17736517636862053666/clu_step0 --cluster-mode 3 --max-iterations 1000 --similarity-type 2 --threads 256 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Greedy Low Mem\n",
      "Total time: 0h 0m 0s 13ms\n",
      "\n",
      "Size of the sequence database: 1\n",
      "Size of the alignment database: 1\n",
      "Number of clusters: 1\n",
      "\n",
      "Writing results 0h 0m 0s 0ms\n",
      "Time for merging to clu_step0: 0h 0m 0s 3ms\n",
      "Time for processing: 0h 0m 0s 22ms\n",
      "mergeclusters component/component_redundant_db component/component_clusters_redundant tmp/17736517636862053666/clu_redundancy tmp/17736517636862053666/clu_step0 --threads 256 --compressed 0 -v 3 \n",
      "\n",
      "Clustering step 1\n",
      "[=================================================================] 1 0s 9ms\n",
      "Clustering step 2\n",
      "[=================================================================] 1 0s 24ms\n",
      "Write merged clustering\n",
      "[=================================================================] 1 0s 494ms\n",
      "Time for merging to component_clusters_redundant: 0h 0m 0s 406ms\n",
      "Time for processing: 0h 0m 1s 94ms\n",
      "cluster process/process_redundant_db process/process_clusters_redundant tmp --min-seq-id 0.9 -c 0.5 --cov-mode 1 \n",
      "\n",
      "MMseqs Version:                     \t18.8cc5c\n",
      "Substitution matrix                 \taa:blosum62.out,nucl:nucleotide.out\n",
      "Seed substitution matrix            \taa:VTML80.out,nucl:nucleotide.out\n",
      "Sensitivity                         \t4\n",
      "k-mer length                        \t0\n",
      "Target search mode                  \t0\n",
      "k-score                             \tseq:2147483647,prof:2147483647\n",
      "Alphabet size                       \taa:21,nucl:5\n",
      "Max sequence length                 \t65535\n",
      "Max results per query               \t20\n",
      "Split database                      \t0\n",
      "Split mode                          \t2\n",
      "Split memory limit                  \t0\n",
      "Coverage threshold                  \t0.5\n",
      "Coverage mode                       \t1\n",
      "Compositional bias                  \t1\n",
      "Compositional bias scale            \t1\n",
      "Diagonal scoring                    \ttrue\n",
      "Exact k-mer matching                \t0\n",
      "Mask residues                       \t1\n",
      "Mask residues probability           \t0.9\n",
      "Mask lower case residues            \t0\n",
      "Mask lower letter repeating N times \t0\n",
      "Minimum diagonal score              \t15\n",
      "Selected taxa                       \t\n",
      "Include identical seq. id.          \tfalse\n",
      "Spaced k-mers                       \t1\n",
      "Preload mode                        \t0\n",
      "Pseudo count a                      \tsubstitution:1.100,context:1.400\n",
      "Pseudo count b                      \tsubstitution:4.100,context:5.800\n",
      "Spaced k-mer pattern                \t\n",
      "Local temporary path                \t\n",
      "Threads                             \t256\n",
      "Compressed                          \t0\n",
      "Verbosity                           \t3\n",
      "Add backtrace                       \tfalse\n",
      "Alignment mode                      \t3\n",
      "Alignment mode                      \t0\n",
      "Allow wrapped scoring               \tfalse\n",
      "E-value threshold                   \t0.001\n",
      "Seq. id. threshold                  \t0.9\n",
      "Min alignment length                \t0\n",
      "Seq. id. mode                       \t0\n",
      "Alternative alignments              \t0\n",
      "Max reject                          \t2147483647\n",
      "Max accept                          \t2147483647\n",
      "Score bias                          \t0\n",
      "Realign hits                        \tfalse\n",
      "Realign score bias                  \t-0.2\n",
      "Realign max seqs                    \t2147483647\n",
      "Correlation score weight            \t0\n",
      "Gap open cost                       \taa:11,nucl:5\n",
      "Gap extension cost                  \taa:1,nucl:2\n",
      "Zdrop                               \t40\n",
      "Rescore mode                        \t0\n",
      "Remove hits by seq. id. and coverage\tfalse\n",
      "Sort results                        \t0\n",
      "Cluster mode                        \t0\n",
      "Max connected component depth       \t1000\n",
      "Similarity type                     \t2\n",
      "Weight file name                    \t\n",
      "Cluster Weight threshold            \t0.9\n",
      "Single step clustering              \tfalse\n",
      "Cascaded clustering steps           \t3\n",
      "Cluster reassign                    \tfalse\n",
      "Remove temporary files              \tfalse\n",
      "Force restart with latest tmp       \tfalse\n",
      "MPI runner                          \t\n",
      "k-mers per sequence                 \t21\n",
      "Scale k-mers per sequence           \taa:0.000,nucl:0.200\n",
      "Adjust k-mer length                 \tfalse\n",
      "Shift hash                          \t67\n",
      "Include only extendable             \tfalse\n",
      "Skip repeating k-mers               \tfalse\n",
      "\n",
      "Set cluster sensitivity to -s 1.000000\n",
      "Set cluster mode GREEDY MEM\n",
      "Set cluster iterations to 1\n",
      "linclust process/process_redundant_db tmp/2756223674631297698/clu_redundancy tmp/2756223674631297698/linclust --cluster-mode 3 --max-iterations 1000 --similarity-type 2 --threads 256 --compressed 0 -v 3 --cluster-weight-threshold 0.9 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.9 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.5 --cov-mode 1 --max-seq-len 65535 --comp-bias-corr 0 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --alph-size aa:13,nucl:5 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 --mask-n-repeat 0 -k 0 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --rescore-mode 0 --filter-hits 0 --sort-results 0 --remove-tmp-files 0 --force-reuse 0 \n",
      "\n",
      "kmermatcher process/process_redundant_db tmp/2756223674631297698/linclust/878352698928025377/pref --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --alph-size aa:13,nucl:5 --min-seq-id 0.9 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 --mask-n-repeat 0 --cov-mode 1 -k 0 -c 0.5 --max-seq-len 65535 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --threads 256 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "kmermatcher process/process_redundant_db tmp/2756223674631297698/linclust/878352698928025377/pref --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --alph-size aa:13,nucl:5 --min-seq-id 0.9 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 --mask-n-repeat 0 --cov-mode 1 -k 0 -c 0.5 --max-seq-len 65535 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --threads 256 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Database size: 1 type: Aminoacid\n",
      "Reduced amino acid alphabet: (A S T) (C) (D B N) (E Q Z) (F Y) (G) (H) (I V) (K R) (L J M) (P) (W) (X) \n",
      "\n",
      "Generate k-mers list for 1 split\n",
      "[=================================================================] 1 0s 0ms\n",
      "Sort kmer 0h 0m 0s 0ms\n",
      "Sort by rep. sequence 0h 0m 0s 0ms\n",
      "Time for fill: 0h 0m 0s 0ms\n",
      "Time for merging to pref: 0h 0m 0s 2ms\n",
      "Time for processing: 0h 0m 0s 36ms\n",
      "rescorediagonal process/process_redundant_db process/process_redundant_db tmp/2756223674631297698/linclust/878352698928025377/pref tmp/2756223674631297698/linclust/878352698928025377/pref_rescore1 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --rescore-mode 0 --wrapped-scoring 0 --filter-hits 0 -e 0.001 -c 0.5 -a 0 --cov-mode 1 --min-seq-id 0.9 --min-aln-len 0 --seq-id-mode 0 --add-self-matches 0 --sort-results 0 --db-load-mode 0 --threads 256 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 1 0s 7ms\n",
      "Time for merging to pref_rescore1: 0h 0m 0s 421ms\n",
      "Time for processing: 0h 0m 1s 69ms\n",
      "clust process/process_redundant_db tmp/2756223674631297698/linclust/878352698928025377/pref_rescore1 tmp/2756223674631297698/linclust/878352698928025377/pre_clust --cluster-mode 3 --max-iterations 1000 --similarity-type 2 --threads 256 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Greedy Low Mem\n",
      "Total time: 0h 0m 0s 14ms\n",
      "\n",
      "Size of the sequence database: 1\n",
      "Size of the alignment database: 1\n",
      "Number of clusters: 1\n",
      "\n",
      "Writing results 0h 0m 0s 0ms\n",
      "Time for merging to pre_clust: 0h 0m 0s 2ms\n",
      "Time for processing: 0h 0m 0s 112ms\n",
      "createsubdb tmp/2756223674631297698/linclust/878352698928025377/order_redundancy process/process_redundant_db tmp/2756223674631297698/linclust/878352698928025377/input_step_redundancy -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step_redundancy: 0h 0m 0s 2ms\n",
      "Time for processing: 0h 0m 0s 14ms\n",
      "createsubdb tmp/2756223674631297698/linclust/878352698928025377/order_redundancy tmp/2756223674631297698/linclust/878352698928025377/pref tmp/2756223674631297698/linclust/878352698928025377/pref_filter1 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to pref_filter1: 0h 0m 0s 2ms\n",
      "Time for processing: 0h 0m 0s 11ms\n",
      "filterdb tmp/2756223674631297698/linclust/878352698928025377/pref_filter1 tmp/2756223674631297698/linclust/878352698928025377/pref_filter2 --filter-file tmp/2756223674631297698/linclust/878352698928025377/order_redundancy --threads 256 --compressed 0 -v 3 \n",
      "\n",
      "Filtering using file(s)\n",
      "[=================================================================] 1 0s 8ms\n",
      "Time for merging to pref_filter2: 0h 0m 0s 380ms\n",
      "Time for processing: 0h 0m 0s 942ms\n",
      "rescorediagonal tmp/2756223674631297698/linclust/878352698928025377/input_step_redundancy tmp/2756223674631297698/linclust/878352698928025377/input_step_redundancy tmp/2756223674631297698/linclust/878352698928025377/pref_filter2 tmp/2756223674631297698/linclust/878352698928025377/pref_rescore2 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --rescore-mode 1 --wrapped-scoring 0 --filter-hits 1 -e 0.001 -c 0.5 -a 0 --cov-mode 1 --min-seq-id 0.9 --min-aln-len 0 --seq-id-mode 0 --add-self-matches 0 --sort-results 0 --db-load-mode 0 --threads 256 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 1 0s 8ms\n",
      "Time for merging to pref_rescore2: 0h 0m 0s 377ms\n",
      "Time for processing: 0h 0m 1s 136ms\n",
      "align tmp/2756223674631297698/linclust/878352698928025377/input_step_redundancy tmp/2756223674631297698/linclust/878352698928025377/input_step_redundancy tmp/2756223674631297698/linclust/878352698928025377/pref_rescore2 tmp/2756223674631297698/linclust/878352698928025377/aln --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.9 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.5 --cov-mode 1 --max-seq-len 65535 --comp-bias-corr 0 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 256 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 1 type: Aminoacid\n",
      "Target database size: 1 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 1 0s 0ms\n",
      "Time for merging to aln: 0h 0m 0s 3ms\n",
      "1 alignments calculated\n",
      "1 sequence pairs passed the thresholds (1.000000 of overall calculated)\n",
      "1.000000 hits per query sequence\n",
      "Time for processing: 0h 0m 0s 110ms\n",
      "clust tmp/2756223674631297698/linclust/878352698928025377/input_step_redundancy tmp/2756223674631297698/linclust/878352698928025377/aln tmp/2756223674631297698/linclust/878352698928025377/clust --cluster-mode 3 --max-iterations 1000 --similarity-type 2 --threads 256 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Greedy Low Mem\n",
      "Total time: 0h 0m 0s 13ms\n",
      "\n",
      "Size of the sequence database: 1\n",
      "Size of the alignment database: 1\n",
      "Number of clusters: 1\n",
      "\n",
      "Writing results 0h 0m 0s 0ms\n",
      "Time for merging to clust: 0h 0m 0s 2ms\n",
      "Time for processing: 0h 0m 0s 22ms\n",
      "mergeclusters process/process_redundant_db tmp/2756223674631297698/clu_redundancy tmp/2756223674631297698/linclust/878352698928025377/pre_clust tmp/2756223674631297698/linclust/878352698928025377/clust --threads 256 --compressed 0 -v 3 \n",
      "\n",
      "Clustering step 1\n",
      "[=================================================================] 1 0s 7ms\n",
      "Clustering step 2\n",
      "[=================================================================] 1 0s 23ms\n",
      "Write merged clustering\n",
      "[=================================================================] 1 0s 450ms\n",
      "Time for merging to clu_redundancy: 0h 0m 0s 399ms\n",
      "Time for processing: 0h 0m 0s 939ms\n",
      "createsubdb tmp/2756223674631297698/clu_redundancy process/process_redundant_db tmp/2756223674631297698/input_step_redundancy -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step_redundancy: 0h 0m 0s 3ms\n",
      "Time for processing: 0h 0m 0s 17ms\n",
      "prefilter tmp/2756223674631297698/input_step_redundancy tmp/2756223674631297698/input_step_redundancy tmp/2756223674631297698/pref_step0 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 1 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.5 --cov-mode 1 --comp-bias-corr 0 --comp-bias-corr-scale 1 --diag-score 0 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --mask-n-repeat 0 --min-ungapped-score 0 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 256 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 1 type: Aminoacid\n",
      "Estimated memory consumption: 977M\n",
      "Target database size: 1 type: Aminoacid\n",
      "Index table k-mer threshold: 154 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 1 0s 0ms\n",
      "Index table: Masked residues: 0\n",
      "No k-mer could be extracted for the database tmp/2756223674631297698/input_step_redundancy.\n",
      "Maybe the sequences length is less than 14 residues.\n",
      "Index table: fill\n",
      "[=================================================================] 1 0s 0ms\n",
      "Index statistics\n",
      "Entries:          0\n",
      "DB size:          488 MB\n",
      "Avg k-mer size:   0.000000\n",
      "Top 10 k-mers\n",
      "    AAAAAA\t0\n",
      "    AAAAAA\t0\n",
      "    AAAAAA\t0\n",
      "    AAAAAA\t0\n",
      "    AAAAAA\t0\n",
      "    AAAAAA\t0\n",
      "    AAAAAA\t0\n",
      "    AAAAAA\t0\n",
      "    AAAAAA\t0\n",
      "    AAAAAA\t0\n",
      "Time for index table init: 0h 0m 0s 475ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 154\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 1\n",
      "Target db start 1 to 1\n",
      "[=================================================================] 1 0s 0ms\n",
      "\n",
      "0.000000 k-mers per position\n",
      "0 DB matches per sequence\n",
      "0 overflows\n",
      "1 sequences passed prefiltering per query sequence\n",
      "1 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step0: 0h 0m 0s 3ms\n",
      "Time for processing: 0h 0m 2s 153ms\n",
      "align tmp/2756223674631297698/input_step_redundancy tmp/2756223674631297698/input_step_redundancy tmp/2756223674631297698/pref_step0 tmp/2756223674631297698/aln_step0 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.9 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.5 --cov-mode 1 --max-seq-len 65535 --comp-bias-corr 0 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 256 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 1 type: Aminoacid\n",
      "Target database size: 1 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 1 0s 0ms\n",
      "Time for merging to aln_step0: 0h 0m 0s 5ms\n",
      "1 alignments calculated\n",
      "1 sequence pairs passed the thresholds (1.000000 of overall calculated)\n",
      "1.000000 hits per query sequence\n",
      "Time for processing: 0h 0m 0s 15ms\n",
      "clust tmp/2756223674631297698/input_step_redundancy tmp/2756223674631297698/aln_step0 tmp/2756223674631297698/clu_step0 --cluster-mode 3 --max-iterations 1000 --similarity-type 2 --threads 256 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Greedy Low Mem\n",
      "Total time: 0h 0m 0s 14ms\n",
      "\n",
      "Size of the sequence database: 1\n",
      "Size of the alignment database: 1\n",
      "Number of clusters: 1\n",
      "\n",
      "Writing results 0h 0m 0s 0ms\n",
      "Time for merging to clu_step0: 0h 0m 0s 2ms\n",
      "Time for processing: 0h 0m 0s 24ms\n",
      "mergeclusters process/process_redundant_db process/process_clusters_redundant tmp/2756223674631297698/clu_redundancy tmp/2756223674631297698/clu_step0 --threads 256 --compressed 0 -v 3 \n",
      "\n",
      "Clustering step 1\n",
      "[=================================================================] 1 0s 9ms\n",
      "Clustering step 2\n",
      "[=================================================================] 1 0s 24ms\n",
      "Write merged clustering\n",
      "[=================================================================] 1 0s 483ms\n",
      "Time for merging to process_clusters_redundant: 0h 0m 0s 375ms\n",
      "Time for processing: 0h 0m 1s 32ms\n",
      "Creating TSV files...\n",
      "createtsv function/function_redundant_db function/function_redundant_db function/function_clusters_redundant function/function_clusters_redundant.tsv \n",
      "\n",
      "MMseqs Version:                 \t18.8cc5c\n",
      "First sequence as representative\tfalse\n",
      "Target column                   \t1\n",
      "Add full header                 \tfalse\n",
      "Sequence source                 \t0\n",
      "Database output                 \tfalse\n",
      "Threads                         \t256\n",
      "Compressed                      \t0\n",
      "Verbosity                       \t3\n",
      "\n",
      "Time for merging to function_clusters_redundant.tsv: 0h 0m 0s 530ms\n",
      "Time for processing: 0h 0m 1s 269ms\n",
      "createtsv component/component_redundant_db component/component_redundant_db component/component_clusters_redundant component/component_clusters_redundant.tsv \n",
      "\n",
      "MMseqs Version:                 \t18.8cc5c\n",
      "First sequence as representative\tfalse\n",
      "Target column                   \t1\n",
      "Add full header                 \tfalse\n",
      "Sequence source                 \t0\n",
      "Database output                 \tfalse\n",
      "Threads                         \t256\n",
      "Compressed                      \t0\n",
      "Verbosity                       \t3\n",
      "\n",
      "Time for merging to component_clusters_redundant.tsv: 0h 0m 0s 519ms\n",
      "Time for processing: 0h 0m 1s 56ms\n",
      "createtsv process/process_redundant_db process/process_redundant_db process/process_clusters_redundant process/process_clusters_redundant.tsv \n",
      "\n",
      "MMseqs Version:                 \t18.8cc5c\n",
      "First sequence as representative\tfalse\n",
      "Target column                   \t1\n",
      "Add full header                 \tfalse\n",
      "Sequence source                 \t0\n",
      "Database output                 \tfalse\n",
      "Threads                         \t256\n",
      "Compressed                      \t0\n",
      "Verbosity                       \t3\n",
      "\n",
      "Time for merging to process_clusters_redundant.tsv: 0h 0m 0s 502ms\n",
      "Time for processing: 0h 0m 1s 143ms\n",
      "All tasks completed successfully.\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "CompletedProcess(args=['/shared/projects/deepmar/PlasmoFP_public/scripts/mmseq_cluster.sh', '/shared/projects/deepmar/data/bacteria_sequences/function_only/reduced_90', '/shared/projects/deepmar/data/bacteria_sequences/function_only/total_bacteria_function_for_reduce.fasta', '/shared/projects/deepmar/data/bacteria_sequences/function_only/empty.fasta', '/shared/projects/deepmar/data/bacteria_sequences/function_only/empty.fasta', '0.9'], returncode=0)"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "subprocess.run([\n",
    "    \"./mmseq_cluster.sh\",\n",
    "    \"/shared/projects/deepmar/data/bacteria_sequences/function_only/reduced_90\",\n",
    "     \"/shared/projects/deepmar/data/bacteria_sequences/function_only/total_bacteria_function_for_reduce.fasta\",\n",
    "    \"/shared/projects/deepmar/data/bacteria_sequences/function_only/empty.fasta\",    \n",
    "    \"/shared/projects/deepmar/data/bacteria_sequences/function_only/empty.fasta\",\n",
    "     \"0.9\"]\n",
    ", check=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Pick cluster representatives from reduced dataset (0.9)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(301300, 2)\n"
     ]
    }
   ],
   "source": [
    "bacteria_function_clusters = pd.read_csv('reduced_90/function/function_clusters_redundant.tsv', sep='\\t', header=None)\n",
    "bacteria_function_clusters.columns = [\"Cluster\", \"Entry\"]\n",
    "\n",
    "print(bacteria_function_clusters.shape)\n",
    "\n",
    "assert bacteria_function_clusters[\"Entry\"].isin(total_bacteria_function[\"Entry\"]).all()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(150860, 11)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/slurm_hmelcher_63507660.tmp/ipykernel_1950853/982002297.py:4: FutureWarning: DataFrameGroupBy.apply operated on the grouping columns. This behavior is deprecated, and in a future version of pandas the grouping columns will be excluded from the operation. Either pass `include_groups=False` to exclude the groupings or explicitly select the grouping columns after groupby to silence this warning.\n",
      "  total_bacteria_function_GO = total_bacteria_function_merged_grouped.apply(lambda x: x.loc[x['Gene Ontology (molecular function)'].str.findall(r\"GO:\\d+\").str.len().idxmax()])\n"
     ]
    }
   ],
   "source": [
    "total_bacteria_function_merged = pd.merge(total_bacteria_function, bacteria_function_clusters, on=\"Entry\") \n",
    "total_bacteria_function_merged_grouped = total_bacteria_function_merged.groupby(\"Cluster\")\n",
    "\n",
    "total_bacteria_function_GO = total_bacteria_function_merged_grouped.apply(lambda x: x.loc[x['Gene Ontology (molecular function)'].str.findall(r\"GO:\\d+\").str.len().idxmax()])\n",
    "\n",
    "print(total_bacteria_function_GO.shape)\n",
    "\n",
    "total_bacteria_function_GO_terms = total_bacteria_function_GO[\"Gene Ontology (molecular function)\"].str.findall(r\"GO:\\d+\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(150860, 11)\n"
     ]
    }
   ],
   "source": [
    "print(total_bacteria_function_GO.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "seq_records_bacteria_function_GO = create_seq_records(total_bacteria_function_GO)\n",
    "\n",
    "assert len(seq_records_bacteria_function_GO) == total_bacteria_function_GO.shape[0]\n",
    "\n",
    "function_path = \"reduced_90_function.fasta\"\n",
    "\n",
    "SeqIO.write(seq_records_bacteria_function_GO, function_path, \"fasta\")\n",
    "\n",
    "total_bacteria_function_GO.to_csv(\"reduced_90_function.tsv\", sep=\"\\t\", index=False)\n",
    "\n",
    "with open('reduced_90_expected_shapes.txt', 'w') as f:\n",
    "    print(\"Function:\", total_bacteria_function_GO.shape, file=f)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Propagate"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "total_bacteria_function_GO_raw_terms = total_bacteria_function_GO['Gene Ontology (molecular function)'].str.findall(r\"GO:\\d+\")\n",
    "\n",
    "total_bacteria_function_GO['Raw GO terms'] = total_bacteria_function_GO_raw_terms"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Unique GO terms in function: 3643\n"
     ]
    }
   ],
   "source": [
    "def get_unique_go_terms(go_terms):\n",
    "    return set([term for terms in go_terms for term in terms])\n",
    "\n",
    "unique_go_terms_function = get_unique_go_terms(total_bacteria_function_GO_raw_terms)\n",
    "\n",
    "print(\"Unique GO terms in function:\", len(unique_go_terms_function))\n",
    "\n",
    "#conver to lists \n",
    "unique_go_terms_function = list(unique_go_terms_function)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|| 73/73 [00:00<00:00, 241.76it/s]\n"
     ]
    }
   ],
   "source": [
    "obo_file_path = \"go-basic.obo\"\n",
    "graph = obonet.read_obo(obo_file_path)\n",
    "\n",
    "\n",
    "remove_edges = [(i, j, k) for i, j, k in graph.edges if not(k==\"is_a\" or k==\"part_of\")]\n",
    "\n",
    "graph.remove_edges_from(remove_edges)\n",
    "\n",
    "    \n",
    "# There should not be any cross-ontology edges, but we verify here\n",
    "crossont_edges = [(i, j, k) for i, j, k in graph.edges if\n",
    "                      graph.nodes[i]['namespace']!= graph.nodes[j]['namespace']]\n",
    "if len(crossont_edges)>0:\n",
    "        graph.remove_edges_from(crossont_edges)\n",
    "\n",
    "        \n",
    "graph_rev = graph.reverse(copy=True) # so edges point towards children\n",
    "all_go_terms_function = []\n",
    "\n",
    "def get_ancestors_from_obo(go_terms_batch, graph):\n",
    "    \"\"\"\n",
    "    Returns ancestors for a batch of GO terms from the OBO graph.\n",
    "    \"\"\"\n",
    "    ancestors_dict_batch = {}\n",
    "    json_dict_batch = {}\n",
    "    all_go_terms_batch = set(go_terms_batch)\n",
    "\n",
    "    for term in go_terms_batch:\n",
    "        if term not in graph:\n",
    "            ancestors_dict_batch[term] = None\n",
    "            json_dict_batch[term] = {\"is_obsolete\":True} # just assume this is the case\n",
    "            continue\n",
    "\n",
    "        ancestors = nx.ancestors(graph, term)\n",
    "        ancestors_dict_batch[term] = list(ancestors) if ancestors else None\n",
    "        \n",
    "        node_data = graph.nodes[term]\n",
    "        is_obsolete = node_data.get(\"is_obsolete\") == \"true\"\n",
    "\n",
    "        json_dict_batch[term] = {\n",
    "            \"id\": term,\n",
    "            \"is_obsolete\": is_obsolete,\n",
    "            \"ancestors\": [{\"id\": a, \"distance\": nx.shortest_path_length(graph, source=a, target=term)} for a in ancestors]\n",
    "        }\n",
    "\n",
    "        all_go_terms_batch.update(ancestors)\n",
    "\n",
    "    return json_dict_batch, ancestors_dict_batch, all_go_terms_batch\n",
    "\n",
    "batch_size = 50  \n",
    "\n",
    "def process_go_terms_in_batches_obo(go_terms, graph, batch_size=20):\n",
    "    json_dict = {}\n",
    "    ancestors_dict = {}\n",
    "    all_go_terms = set()\n",
    "\n",
    "    for i in tqdm(range(0, len(go_terms), batch_size)):\n",
    "        batch = go_terms[i:i + batch_size]\n",
    "        json_data_batch, ancestors_dict_batch, all_go_terms_batch = get_ancestors_from_obo(batch, graph)\n",
    "\n",
    "        if json_data_batch and ancestors_dict_batch:\n",
    "            json_dict.update(json_data_batch)\n",
    "            ancestors_dict.update(ancestors_dict_batch)\n",
    "            all_go_terms.update(all_go_terms_batch)\n",
    "\n",
    "    return json_dict, ancestors_dict, all_go_terms\n",
    "\n",
    "\n",
    "function_json_dict, function_ancestors_dict, function_all_go_terms = process_go_terms_in_batches_obo(unique_go_terms_function, graph_rev, batch_size=batch_size)\n",
    "\n",
    "with open('function_data_ancestors_dict.json', 'w') as f:\n",
    "    json.dump(function_ancestors_dict, f, indent=4)\n",
    "with open('function_data_json_dict.json', 'w') as f:\n",
    "    json.dump(function_json_dict, f, indent=4)\n",
    "\n",
    "\n",
    "\n",
    "assert len(function_json_dict) == len(unique_go_terms_function)\n",
    "assert len(function_ancestors_dict) == len(unique_go_terms_function)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Function Obsolete: 22\n"
     ]
    }
   ],
   "source": [
    "function_OBSELETE = [term for term in function_json_dict if function_json_dict[term][\"is_obsolete\"]]\n",
    "\n",
    "print(\"Function Obsolete:\", len(function_OBSELETE))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Adding ancestors: 100%|| 150860/150860 [00:00<00:00, 227813.67it/s]\n"
     ]
    }
   ],
   "source": [
    "def add_ancestors(go_terms_list, ancestors_dict):\n",
    "    all_ancestors = []\n",
    "    \n",
    "    for term_list in tqdm(go_terms_list, desc=\"Adding ancestors\"):\n",
    "        terms_return = set() \n",
    "        for t in term_list:\n",
    "            # Check if the term exists in the dictionary and has valid ancestors\n",
    "            if t in ancestors_dict and ancestors_dict[t] is not None:\n",
    "                # Add ancestors, excluding the original term itself\n",
    "                terms_return.update(a for a in ancestors_dict[t] if a != t)\n",
    "                \n",
    "            # I (Helena) don't think it's correct to exclude the original term\n",
    "            # I couldn't see a place where they were added back in\n",
    "            terms_return.update([t])\n",
    "        \n",
    "        # Append the filtered ancestors found for this protein (as a list)\n",
    "        all_ancestors.append(list(terms_return))\n",
    "\n",
    "    return all_ancestors\n",
    "\n",
    "total_bacteria_function_GO_raw_terms_propogated = add_ancestors(list(total_bacteria_function_GO_raw_terms), function_ancestors_dict)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Non-propogated function terms: 74\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "array(['citrate (Si)-synthase activity [GO:0004108]',\n",
       "       'aldo-keto reductase (NADPH) activity [GO:0004033]',\n",
       "       'electron transporter, transferring electrons within cytochrome b6/f complex of photosystem II activity [GO:0045158]',\n",
       "       'citrate (Re)-synthase activity [GO:0050450]',\n",
       "       'androsterone dehydrogenase (B-specific) activity [GO:0047042]',\n",
       "       'organic anion transmembrane transporter activity [GO:0008514]',\n",
       "       'electron transporter, transferring electrons within the cyclic electron transport pathway of photosynthesis activity [GO:0045156]'],\n",
       "      dtype=object)"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "non_propogated_function_terms = [i for i, x in enumerate(total_bacteria_function_GO_raw_terms_propogated) if len(x) <= 1]\n",
    "\n",
    "print(\"Non-propogated function terms:\", len(non_propogated_function_terms))\n",
    "# these are obsolete go terms.\n",
    "total_bacteria_function_GO.iloc[non_propogated_function_terms]['Gene Ontology (molecular function)'].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "total_bacteria_function_GO['Raw propagated GO terms'] = total_bacteria_function_GO_raw_terms_propogated"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "total_bacteria_function_GO = total_bacteria_function_GO[total_bacteria_function_GO['Raw propagated GO terms'].str.len() > 0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "seq_records_bacteria_function_GO_propagated = create_seq_records(total_bacteria_function_GO)\n",
    "\n",
    "assert len(seq_records_bacteria_function_GO_propagated) == total_bacteria_function_GO.shape[0]\n",
    "\n",
    "function_path = \"reduced_90_function_propagated.fasta\"\n",
    "\n",
    "SeqIO.write(seq_records_bacteria_function_GO_propagated, function_path, \"fasta\")\n",
    "\n",
    "total_bacteria_function_GO.to_csv(\"reduced_90_function_propagated.tsv\", sep=\"\\t\", index=False)\n",
    "\n",
    "with open('reduced_90_expected_shapes_propagated.txt', 'w') as f:\n",
    "    print(\"Function:\", total_bacteria_function_GO.shape, file=f)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now we can preform the sequence similarity splits for the train, dev, and test sets"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "! mkdir reduced_90_similarity_30"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true,
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Creating databases...\n",
      "createdb /shared/projects/deepmar/data/bacteria_sequences/function_only/reduced_90_function_propagated.fasta function/function_redundant_db \n",
      "\n",
      "MMseqs Version:                    \t18.8cc5c\n",
      "Database type                      \t0\n",
      "Shuffle input database             \ttrue\n",
      "Createdb mode                      \t0\n",
      "Write lookup file                  \t1\n",
      "Offset of numeric ids              \t0\n",
      "Threads                            \t256\n",
      "Compressed                         \t0\n",
      "Mask residues                      \t0\n",
      "Mask residues probability          \t0.9\n",
      "Mask lower case residues           \t0\n",
      "Mask lower letter repeating N times\t0\n",
      "Use GPU                            \t0\n",
      "Verbosity                          \t3\n",
      "\n",
      "Converting sequences\n",
      "[===============\n",
      "Time for merging to function_redundant_db_h: 0h 0m 0s 112ms\n",
      "Time for merging to function_redundant_db: 0h 0m 0s 146ms\n",
      "Database type: Aminoacid\n",
      "Time for processing: 0h 0m 0s 955ms\n",
      "createdb /shared/projects/deepmar/data/bacteria_sequences/function_only/empty.fasta component/component_redundant_db \n",
      "\n",
      "MMseqs Version:                    \t18.8cc5c\n",
      "Database type                      \t0\n",
      "Shuffle input database             \ttrue\n",
      "Createdb mode                      \t0\n",
      "Write lookup file                  \t1\n",
      "Offset of numeric ids              \t0\n",
      "Threads                            \t256\n",
      "Compressed                         \t0\n",
      "Mask residues                      \t0\n",
      "Mask residues probability          \t0.9\n",
      "Mask lower case residues           \t0\n",
      "Mask lower letter repeating N times\t0\n",
      "Use GPU                            \t0\n",
      "Verbosity                          \t3\n",
      "\n",
      "Converting sequences\n",
      "[\n",
      "Time for merging to component_redundant_db_h: 0h 0m 0s 73ms\n",
      "Time for merging to component_redundant_db: 0h 0m 0s 87ms\n",
      "Database type: Aminoacid\n",
      "Time for processing: 0h 0m 0s 381ms\n",
      "createdb /shared/projects/deepmar/data/bacteria_sequences/function_only/empty.fasta process/process_redundant_db \n",
      "\n",
      "MMseqs Version:                    \t18.8cc5c\n",
      "Database type                      \t0\n",
      "Shuffle input database             \ttrue\n",
      "Createdb mode                      \t0\n",
      "Write lookup file                  \t1\n",
      "Offset of numeric ids              \t0\n",
      "Threads                            \t256\n",
      "Compressed                         \t0\n",
      "Mask residues                      \t0\n",
      "Mask residues probability          \t0.9\n",
      "Mask lower case residues           \t0\n",
      "Mask lower letter repeating N times\t0\n",
      "Use GPU                            \t0\n",
      "Verbosity                          \t3\n",
      "\n",
      "Converting sequences\n",
      "[\n",
      "Time for merging to process_redundant_db_h: 0h 0m 0s 90ms\n",
      "Time for merging to process_redundant_db: 0h 0m 0s 59ms\n",
      "Database type: Aminoacid\n",
      "Time for processing: 0h 0m 0s 357ms\n",
      "Clustering sequences...\n",
      "Create directory tmp\n",
      "cluster function/function_redundant_db function/function_clusters_redundant tmp --min-seq-id 0.3 -c 0.5 --cov-mode 1 \n",
      "\n",
      "MMseqs Version:                     \t18.8cc5c\n",
      "Substitution matrix                 \taa:blosum62.out,nucl:nucleotide.out\n",
      "Seed substitution matrix            \taa:VTML80.out,nucl:nucleotide.out\n",
      "Sensitivity                         \t4\n",
      "k-mer length                        \t0\n",
      "Target search mode                  \t0\n",
      "k-score                             \tseq:2147483647,prof:2147483647\n",
      "Alphabet size                       \taa:21,nucl:5\n",
      "Max sequence length                 \t65535\n",
      "Max results per query               \t20\n",
      "Split database                      \t0\n",
      "Split mode                          \t2\n",
      "Split memory limit                  \t0\n",
      "Coverage threshold                  \t0.5\n",
      "Coverage mode                       \t1\n",
      "Compositional bias                  \t1\n",
      "Compositional bias scale            \t1\n",
      "Diagonal scoring                    \ttrue\n",
      "Exact k-mer matching                \t0\n",
      "Mask residues                       \t1\n",
      "Mask residues probability           \t0.9\n",
      "Mask lower case residues            \t0\n",
      "Mask lower letter repeating N times \t0\n",
      "Minimum diagonal score              \t15\n",
      "Selected taxa                       \t\n",
      "Include identical seq. id.          \tfalse\n",
      "Spaced k-mers                       \t1\n",
      "Preload mode                        \t0\n",
      "Pseudo count a                      \tsubstitution:1.100,context:1.400\n",
      "Pseudo count b                      \tsubstitution:4.100,context:5.800\n",
      "Spaced k-mer pattern                \t\n",
      "Local temporary path                \t\n",
      "Threads                             \t256\n",
      "Compressed                          \t0\n",
      "Verbosity                           \t3\n",
      "Add backtrace                       \tfalse\n",
      "Alignment mode                      \t3\n",
      "Alignment mode                      \t0\n",
      "Allow wrapped scoring               \tfalse\n",
      "E-value threshold                   \t0.001\n",
      "Seq. id. threshold                  \t0.3\n",
      "Min alignment length                \t0\n",
      "Seq. id. mode                       \t0\n",
      "Alternative alignments              \t0\n",
      "Max reject                          \t2147483647\n",
      "Max accept                          \t2147483647\n",
      "Score bias                          \t0\n",
      "Realign hits                        \tfalse\n",
      "Realign score bias                  \t-0.2\n",
      "Realign max seqs                    \t2147483647\n",
      "Correlation score weight            \t0\n",
      "Gap open cost                       \taa:11,nucl:5\n",
      "Gap extension cost                  \taa:1,nucl:2\n",
      "Zdrop                               \t40\n",
      "Rescore mode                        \t0\n",
      "Remove hits by seq. id. and coverage\tfalse\n",
      "Sort results                        \t0\n",
      "Cluster mode                        \t0\n",
      "Max connected component depth       \t1000\n",
      "Similarity type                     \t2\n",
      "Weight file name                    \t\n",
      "Cluster Weight threshold            \t0.9\n",
      "Single step clustering              \tfalse\n",
      "Cascaded clustering steps           \t3\n",
      "Cluster reassign                    \tfalse\n",
      "Remove temporary files              \tfalse\n",
      "Force restart with latest tmp       \tfalse\n",
      "MPI runner                          \t\n",
      "k-mers per sequence                 \t21\n",
      "Scale k-mers per sequence           \taa:0.000,nucl:0.200\n",
      "Adjust k-mer length                 \tfalse\n",
      "Shift hash                          \t67\n",
      "Include only extendable             \tfalse\n",
      "Skip repeating k-mers               \tfalse\n",
      "\n",
      "Set cluster sensitivity to -s 5.000000\n",
      "Set cluster mode GREEDY MEM\n",
      "Set cluster iterations to 3\n",
      "linclust function/function_redundant_db tmp/11983341506639745049/clu_redundancy tmp/11983341506639745049/linclust --cluster-mode 3 --max-iterations 1000 --similarity-type 2 --threads 256 --compressed 0 -v 3 --cluster-weight-threshold 0.9 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.3 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.5 --cov-mode 1 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --alph-size aa:13,nucl:5 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 --mask-n-repeat 0 -k 0 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --rescore-mode 0 --filter-hits 0 --sort-results 0 --remove-tmp-files 0 --force-reuse 0 \n",
      "\n",
      "kmermatcher function/function_redundant_db tmp/11983341506639745049/linclust/6301907503532421144/pref --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --alph-size aa:13,nucl:5 --min-seq-id 0.3 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 --mask-n-repeat 0 --cov-mode 1 -k 0 -c 0.5 --max-seq-len 65535 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --threads 256 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "kmermatcher function/function_redundant_db tmp/11983341506639745049/linclust/6301907503532421144/pref --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --alph-size aa:13,nucl:5 --min-seq-id 0.3 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 --mask-n-repeat 0 --cov-mode 1 -k 0 -c 0.5 --max-seq-len 65535 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --threads 256 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Database size: 150860 type: Aminoacid\n",
      "Reduced amino acid alphabet: (A S T) (C) (D B N) (E Q Z) (F Y) (G) (H) (I V) (K R) (L J M) (P) (W) (X) \n",
      "\n",
      "Generate k-mers list for 1 split\n",
      "[=================================================================] 150.86K 1s 307ms\n",
      "Sort kmer 0h 0m 0s 414ms\n",
      "Sort by rep. sequence 0h 0m 0s 152ms\n",
      "Time for fill: 0h 0m 0s 32ms\n",
      "Time for merging to pref: 0h 0m 0s 2ms\n",
      "Time for processing: 0h 0m 2s 66ms\n",
      "rescorediagonal function/function_redundant_db function/function_redundant_db tmp/11983341506639745049/linclust/6301907503532421144/pref tmp/11983341506639745049/linclust/6301907503532421144/pref_rescore1 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --rescore-mode 0 --wrapped-scoring 0 --filter-hits 0 -e 0.001 -c 0.5 -a 0 --cov-mode 1 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --add-self-matches 0 --sort-results 0 --db-load-mode 0 --threads 256 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 150.86K 0s 339ms\n",
      "Time for merging to pref_rescore1: 0h 0m 0s 472ms\n",
      "Time for processing: 0h 0m 1s 880ms\n",
      "clust function/function_redundant_db tmp/11983341506639745049/linclust/6301907503532421144/pref_rescore1 tmp/11983341506639745049/linclust/6301907503532421144/pre_clust --cluster-mode 3 --max-iterations 1000 --similarity-type 2 --threads 256 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Greedy Low Mem\n",
      "Total time: 0h 0m 0s 102ms\n",
      "\n",
      "Size of the sequence database: 150860\n",
      "Size of the alignment database: 150860\n",
      "Number of clusters: 76901\n",
      "\n",
      "Writing results 0h 0m 0s 11ms\n",
      "Time for merging to pre_clust: 0h 0m 0s 3ms\n",
      "Time for processing: 0h 0m 0s 255ms\n",
      "createsubdb tmp/11983341506639745049/linclust/6301907503532421144/order_redundancy function/function_redundant_db tmp/11983341506639745049/linclust/6301907503532421144/input_step_redundancy -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step_redundancy: 0h 0m 0s 3ms\n",
      "Time for processing: 0h 0m 0s 35ms\n",
      "createsubdb tmp/11983341506639745049/linclust/6301907503532421144/order_redundancy tmp/11983341506639745049/linclust/6301907503532421144/pref tmp/11983341506639745049/linclust/6301907503532421144/pref_filter1 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to pref_filter1: 0h 0m 0s 3ms\n",
      "Time for processing: 0h 0m 0s 39ms\n",
      "filterdb tmp/11983341506639745049/linclust/6301907503532421144/pref_filter1 tmp/11983341506639745049/linclust/6301907503532421144/pref_filter2 --filter-file tmp/11983341506639745049/linclust/6301907503532421144/order_redundancy --threads 256 --compressed 0 -v 3 \n",
      "\n",
      "Filtering using file(s)\n",
      "[=================================================================] 76.90K 0s 83ms\n",
      "Time for merging to pref_filter2: 0h 0m 0s 400ms\n",
      "Time for processing: 0h 0m 1s 633ms\n",
      "rescorediagonal tmp/11983341506639745049/linclust/6301907503532421144/input_step_redundancy tmp/11983341506639745049/linclust/6301907503532421144/input_step_redundancy tmp/11983341506639745049/linclust/6301907503532421144/pref_filter2 tmp/11983341506639745049/linclust/6301907503532421144/pref_rescore2 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --rescore-mode 1 --wrapped-scoring 0 --filter-hits 1 -e 0.001 -c 0.5 -a 0 --cov-mode 1 --min-seq-id 0.3 --min-aln-len 0 --seq-id-mode 0 --add-self-matches 0 --sort-results 0 --db-load-mode 0 --threads 256 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 76.90K 0s 154ms\n",
      "Time for merging to pref_rescore2: 0h 0m 0s 417ms\n",
      "Time for processing: 0h 0m 1s 928ms\n",
      "align tmp/11983341506639745049/linclust/6301907503532421144/input_step_redundancy tmp/11983341506639745049/linclust/6301907503532421144/input_step_redundancy tmp/11983341506639745049/linclust/6301907503532421144/pref_rescore2 tmp/11983341506639745049/linclust/6301907503532421144/aln --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.3 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.5 --cov-mode 1 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 256 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 76901 type: Aminoacid\n",
      "Target database size: 76901 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 76.90K 32s 814ms\n",
      "Time for merging to aln: 0h 0m 0s 383ms\n",
      "175160 alignments calculated\n",
      "170712 sequence pairs passed the thresholds (0.974606 of overall calculated)\n",
      "2.219893 hits per query sequence\n",
      "Time for processing: 0h 0m 34s 503ms\n",
      "clust tmp/11983341506639745049/linclust/6301907503532421144/input_step_redundancy tmp/11983341506639745049/linclust/6301907503532421144/aln tmp/11983341506639745049/linclust/6301907503532421144/clust --cluster-mode 3 --max-iterations 1000 --similarity-type 2 --threads 256 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Greedy Low Mem\n",
      "Total time: 0h 0m 0s 60ms\n",
      "\n",
      "Size of the sequence database: 76901\n",
      "Size of the alignment database: 76901\n",
      "Number of clusters: 41884\n",
      "\n",
      "Writing results 0h 0m 0s 6ms\n",
      "Time for merging to clust: 0h 0m 0s 3ms\n",
      "Time for processing: 0h 0m 0s 201ms\n",
      "mergeclusters function/function_redundant_db tmp/11983341506639745049/clu_redundancy tmp/11983341506639745049/linclust/6301907503532421144/pre_clust tmp/11983341506639745049/linclust/6301907503532421144/clust --threads 256 --compressed 0 -v 3 \n",
      "\n",
      "Clustering step 1\n",
      "[=================================================================] 76.90K 0s 38ms\n",
      "Clustering step 2\n",
      "[=================================================================] 41.88K 0s 68ms\n",
      "Write merged clustering\n",
      "[=================================================================] 150.86K 0s 699ms\n",
      "Time for merging to clu_redundancy: 0h 0m 0s 617ms\n",
      "Time for processing: 0h 0m 1s 448ms\n",
      "createsubdb tmp/11983341506639745049/clu_redundancy function/function_redundant_db tmp/11983341506639745049/input_step_redundancy -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step_redundancy: 0h 0m 0s 2ms\n",
      "Time for processing: 0h 0m 0s 28ms\n",
      "prefilter tmp/11983341506639745049/input_step_redundancy tmp/11983341506639745049/input_step_redundancy tmp/11983341506639745049/pref_step0 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 1 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.5 --cov-mode 1 --comp-bias-corr 0 --comp-bias-corr-scale 1 --diag-score 0 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --mask-n-repeat 0 --min-ungapped-score 0 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 256 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 41884 type: Aminoacid\n",
      "Estimated memory consumption: 1G\n",
      "Target database size: 41884 type: Aminoacid\n",
      "Index table k-mer threshold: 154 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 41.88K 1s 77ms\n",
      "Index table: Masked residues: 63938\n",
      "Index table: fill\n",
      "[=================================================================] 41.88K 0s 838ms\n",
      "Index statistics\n",
      "Entries:          5720025\n",
      "DB size:          521 MB\n",
      "Avg k-mer size:   0.089375\n",
      "Top 10 k-mers\n",
      "    GPGGTL\t176\n",
      "    SRRWPI\t76\n",
      "    PNGSFN\t73\n",
      "    AIGPGK\t71\n",
      "    GRNGTL\t70\n",
      "    RNAGRQ\t70\n",
      "    GGRRIA\t64\n",
      "    PTGLGH\t63\n",
      "    KPTMSG\t62\n",
      "    PNGSLN\t62\n",
      "Time for index table init: 0h 0m 2s 350ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 154\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 41884\n",
      "Target db start 1 to 41884\n",
      "[=================================================================] 41.88K 2s 360ms\n",
      "\n",
      "1.527927 k-mers per position\n",
      "260 DB matches per sequence\n",
      "0 overflows\n",
      "8 sequences passed prefiltering per query sequence\n",
      "5 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step0: 0h 0m 0s 375ms\n",
      "Time for processing: 0h 0m 7s 733ms\n",
      "align tmp/11983341506639745049/input_step_redundancy tmp/11983341506639745049/input_step_redundancy tmp/11983341506639745049/pref_step0 tmp/11983341506639745049/aln_step0 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.3 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.5 --cov-mode 1 --max-seq-len 65535 --comp-bias-corr 0 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 256 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 41884 type: Aminoacid\n",
      "Target database size: 41884 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 41.88K 1m 4s 820ms\n",
      "Time for merging to aln_step0: 0h 0m 0s 342ms\n",
      "336155 alignments calculated\n",
      "308868 sequence pairs passed the thresholds (0.918826 of overall calculated)\n",
      "7.374367 hits per query sequence\n",
      "Time for processing: 0h 1m 6s 238ms\n",
      "clust tmp/11983341506639745049/input_step_redundancy tmp/11983341506639745049/aln_step0 tmp/11983341506639745049/clu_step0 --cluster-mode 3 --max-iterations 1000 --similarity-type 2 --threads 256 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Greedy Low Mem\n",
      "Total time: 0h 0m 0s 56ms\n",
      "\n",
      "Size of the sequence database: 41884\n",
      "Size of the alignment database: 41884\n",
      "Number of clusters: 19436\n",
      "\n",
      "Writing results 0h 0m 0s 3ms\n",
      "Time for merging to clu_step0: 0h 0m 0s 2ms\n",
      "Time for processing: 0h 0m 0s 164ms\n",
      "createsubdb tmp/11983341506639745049/clu_step0 tmp/11983341506639745049/input_step_redundancy tmp/11983341506639745049/input_step1 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step1: 0h 0m 0s 2ms\n",
      "Time for processing: 0h 0m 0s 21ms\n",
      "prefilter tmp/11983341506639745049/input_step1 tmp/11983341506639745049/input_step1 tmp/11983341506639745049/pref_step1 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 3 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.5 --cov-mode 1 --comp-bias-corr 1 --comp-bias-corr-scale 1 --diag-score 1 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --mask-n-repeat 0 --min-ungapped-score 15 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 256 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 19436 type: Aminoacid\n",
      "Estimated memory consumption: 1G\n",
      "Target database size: 19436 type: Aminoacid\n",
      "Index table k-mer threshold: 136 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 19.44K 0s 571ms\n",
      "Index table: Masked residues: 40844\n",
      "Index table: fill\n",
      "[=================================================================] 19.44K 0s 730ms\n",
      "Index statistics\n",
      "Entries:          5482249\n",
      "DB size:          519 MB\n",
      "Avg k-mer size:   0.085660\n",
      "Top 10 k-mers\n",
      "    GPGGTL\t51\n",
      "    LLETLD\t31\n",
      "    GLGIGK\t26\n",
      "    FAKAKA\t25\n",
      "    GGKRIA\t24\n",
      "    AIGGGL\t24\n",
      "    VVGGGL\t24\n",
      "    GLGSAA\t22\n",
      "    GEQVAR\t22\n",
      "    GQQVAR\t22\n",
      "Time for index table init: 0h 0m 1s 726ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 136\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 19436\n",
      "Target db start 1 to 19436\n",
      "[=================================================================] 19.44K 5s 736ms\n",
      "\n",
      "18.457906 k-mers per position\n",
      "703 DB matches per sequence\n",
      "0 overflows\n",
      "6 sequences passed prefiltering per query sequence\n",
      "5 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step1: 0h 0m 0s 354ms\n",
      "Time for processing: 0h 0m 10s 373ms\n",
      "align tmp/11983341506639745049/input_step1 tmp/11983341506639745049/input_step1 tmp/11983341506639745049/pref_step1 tmp/11983341506639745049/aln_step1 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.3 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.5 --cov-mode 1 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 256 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 19436 type: Aminoacid\n",
      "Target database size: 19436 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 19.44K 13s 6ms\n",
      "Time for merging to aln_step1: 0h 0m 0s 325ms\n",
      "119423 alignments calculated\n",
      "61682 sequence pairs passed the thresholds (0.516500 of overall calculated)\n",
      "3.173595 hits per query sequence\n",
      "Time for processing: 0h 0m 14s 142ms\n",
      "clust tmp/11983341506639745049/input_step1 tmp/11983341506639745049/aln_step1 tmp/11983341506639745049/clu_step1 --cluster-mode 3 --max-iterations 1000 --similarity-type 2 --threads 256 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Greedy Low Mem\n",
      "Total time: 0h 0m 0s 26ms\n",
      "\n",
      "Size of the sequence database: 19436\n",
      "Size of the alignment database: 19436\n",
      "Number of clusters: 12213\n",
      "\n",
      "Writing results 0h 0m 0s 2ms\n",
      "Time for merging to clu_step1: 0h 0m 0s 2ms\n",
      "Time for processing: 0h 0m 0s 128ms\n",
      "createsubdb tmp/11983341506639745049/clu_step1 tmp/11983341506639745049/input_step1 tmp/11983341506639745049/input_step2 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step2: 0h 0m 0s 2ms\n",
      "Time for processing: 0h 0m 0s 19ms\n",
      "prefilter tmp/11983341506639745049/input_step2 tmp/11983341506639745049/input_step2 tmp/11983341506639745049/pref_step2 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 5 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.5 --cov-mode 1 --comp-bias-corr 1 --comp-bias-corr-scale 1 --diag-score 1 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --mask-n-repeat 0 --min-ungapped-score 15 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 256 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 12213 type: Aminoacid\n",
      "Estimated memory consumption: 1G\n",
      "Target database size: 12213 type: Aminoacid\n",
      "Index table k-mer threshold: 118 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 12.21K 0s 368ms\n",
      "Index table: Masked residues: 32582\n",
      "Index table: fill\n",
      "[=================================================================] 12.21K 0s 504ms\n",
      "Index statistics\n",
      "Entries:          3849644\n",
      "DB size:          510 MB\n",
      "Avg k-mer size:   0.060151\n",
      "Top 10 k-mers\n",
      "    AAAAAA\t29\n",
      "    AGAAAA\t22\n",
      "    AAAALA\t22\n",
      "    AAAAAG\t22\n",
      "    GPGGTL\t21\n",
      "    AAAAGA\t19\n",
      "    LAAAAA\t17\n",
      "    ALALAG\t17\n",
      "    VVGGGL\t17\n",
      "    AARAAA\t16\n",
      "Time for index table init: 0h 0m 1s 297ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 118\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 12213\n",
      "Target db start 1 to 12213\n",
      "[=================================================================] 12.21K 18s 478ms\n",
      "\n",
      "148.566311 k-mers per position\n",
      "3207 DB matches per sequence\n",
      "0 overflows\n",
      "15 sequences passed prefiltering per query sequence\n",
      "20 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step2: 0h 0m 0s 347ms\n",
      "Time for processing: 0h 0m 22s 934ms\n",
      "align tmp/11983341506639745049/input_step2 tmp/11983341506639745049/input_step2 tmp/11983341506639745049/pref_step2 tmp/11983341506639745049/aln_step2 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.3 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.5 --cov-mode 1 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 256 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 12213 type: Aminoacid\n",
      "Target database size: 12213 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 12.21K 7s 892ms\n",
      "Time for merging to aln_step2: 0h 0m 0s 334ms\n",
      "155733 alignments calculated\n",
      "20752 sequence pairs passed the thresholds (0.133254 of overall calculated)\n",
      "1.699173 hits per query sequence\n",
      "Time for processing: 0h 0m 9s 167ms\n",
      "clust tmp/11983341506639745049/input_step2 tmp/11983341506639745049/aln_step2 tmp/11983341506639745049/clu_step2 --cluster-mode 3 --max-iterations 1000 --similarity-type 2 --threads 256 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Greedy Low Mem\n",
      "Total time: 0h 0m 0s 19ms\n",
      "\n",
      "Size of the sequence database: 12213\n",
      "Size of the alignment database: 12213\n",
      "Number of clusters: 9826\n",
      "\n",
      "Writing results 0h 0m 0s 1ms\n",
      "Time for merging to clu_step2: 0h 0m 0s 2ms\n",
      "Time for processing: 0h 0m 0s 119ms\n",
      "mergeclusters function/function_redundant_db function/function_clusters_redundant tmp/11983341506639745049/clu_redundancy tmp/11983341506639745049/clu_step0 tmp/11983341506639745049/clu_step1 tmp/11983341506639745049/clu_step2 --threads 256 --compressed 0 -v 3 \n",
      "\n",
      "Clustering step 1\n",
      "[=================================================================] 41.88K 0s 32ms\n",
      "Clustering step 2\n",
      "[=================================================================] 19.44K 0s 69ms\n",
      "Clustering step 3\n",
      "[=================================================================] 12.21K 0s 88ms\n",
      "Clustering step 4\n",
      "[=================================================================] 9.83K 0s 106ms\n",
      "Write merged clustering\n",
      "[=================================================================] 150.86K 0s 663ms\n",
      "Time for merging to function_clusters_redundant: 0h 0m 0s 353ms\n",
      "Time for processing: 0h 0m 1s 209ms\n",
      "cluster component/component_redundant_db component/component_clusters_redundant tmp --min-seq-id 0.3 -c 0.5 --cov-mode 1 \n",
      "\n",
      "MMseqs Version:                     \t18.8cc5c\n",
      "Substitution matrix                 \taa:blosum62.out,nucl:nucleotide.out\n",
      "Seed substitution matrix            \taa:VTML80.out,nucl:nucleotide.out\n",
      "Sensitivity                         \t4\n",
      "k-mer length                        \t0\n",
      "Target search mode                  \t0\n",
      "k-score                             \tseq:2147483647,prof:2147483647\n",
      "Alphabet size                       \taa:21,nucl:5\n",
      "Max sequence length                 \t65535\n",
      "Max results per query               \t20\n",
      "Split database                      \t0\n",
      "Split mode                          \t2\n",
      "Split memory limit                  \t0\n",
      "Coverage threshold                  \t0.5\n",
      "Coverage mode                       \t1\n",
      "Compositional bias                  \t1\n",
      "Compositional bias scale            \t1\n",
      "Diagonal scoring                    \ttrue\n",
      "Exact k-mer matching                \t0\n",
      "Mask residues                       \t1\n",
      "Mask residues probability           \t0.9\n",
      "Mask lower case residues            \t0\n",
      "Mask lower letter repeating N times \t0\n",
      "Minimum diagonal score              \t15\n",
      "Selected taxa                       \t\n",
      "Include identical seq. id.          \tfalse\n",
      "Spaced k-mers                       \t1\n",
      "Preload mode                        \t0\n",
      "Pseudo count a                      \tsubstitution:1.100,context:1.400\n",
      "Pseudo count b                      \tsubstitution:4.100,context:5.800\n",
      "Spaced k-mer pattern                \t\n",
      "Local temporary path                \t\n",
      "Threads                             \t256\n",
      "Compressed                          \t0\n",
      "Verbosity                           \t3\n",
      "Add backtrace                       \tfalse\n",
      "Alignment mode                      \t3\n",
      "Alignment mode                      \t0\n",
      "Allow wrapped scoring               \tfalse\n",
      "E-value threshold                   \t0.001\n",
      "Seq. id. threshold                  \t0.3\n",
      "Min alignment length                \t0\n",
      "Seq. id. mode                       \t0\n",
      "Alternative alignments              \t0\n",
      "Max reject                          \t2147483647\n",
      "Max accept                          \t2147483647\n",
      "Score bias                          \t0\n",
      "Realign hits                        \tfalse\n",
      "Realign score bias                  \t-0.2\n",
      "Realign max seqs                    \t2147483647\n",
      "Correlation score weight            \t0\n",
      "Gap open cost                       \taa:11,nucl:5\n",
      "Gap extension cost                  \taa:1,nucl:2\n",
      "Zdrop                               \t40\n",
      "Rescore mode                        \t0\n",
      "Remove hits by seq. id. and coverage\tfalse\n",
      "Sort results                        \t0\n",
      "Cluster mode                        \t0\n",
      "Max connected component depth       \t1000\n",
      "Similarity type                     \t2\n",
      "Weight file name                    \t\n",
      "Cluster Weight threshold            \t0.9\n",
      "Single step clustering              \tfalse\n",
      "Cascaded clustering steps           \t3\n",
      "Cluster reassign                    \tfalse\n",
      "Remove temporary files              \tfalse\n",
      "Force restart with latest tmp       \tfalse\n",
      "MPI runner                          \t\n",
      "k-mers per sequence                 \t21\n",
      "Scale k-mers per sequence           \taa:0.000,nucl:0.200\n",
      "Adjust k-mer length                 \tfalse\n",
      "Shift hash                          \t67\n",
      "Include only extendable             \tfalse\n",
      "Skip repeating k-mers               \tfalse\n",
      "\n",
      "Set cluster sensitivity to -s 5.000000\n",
      "Set cluster mode GREEDY MEM\n",
      "Set cluster iterations to 3\n",
      "linclust component/component_redundant_db tmp/5251849818421459973/clu_redundancy tmp/5251849818421459973/linclust --cluster-mode 3 --max-iterations 1000 --similarity-type 2 --threads 256 --compressed 0 -v 3 --cluster-weight-threshold 0.9 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.3 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.5 --cov-mode 1 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --alph-size aa:13,nucl:5 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 --mask-n-repeat 0 -k 0 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --rescore-mode 0 --filter-hits 0 --sort-results 0 --remove-tmp-files 0 --force-reuse 0 \n",
      "\n",
      "kmermatcher component/component_redundant_db tmp/5251849818421459973/linclust/15506357871815186212/pref --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --alph-size aa:13,nucl:5 --min-seq-id 0.3 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 --mask-n-repeat 0 --cov-mode 1 -k 0 -c 0.5 --max-seq-len 65535 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --threads 256 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "kmermatcher component/component_redundant_db tmp/5251849818421459973/linclust/15506357871815186212/pref --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --alph-size aa:13,nucl:5 --min-seq-id 0.3 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 --mask-n-repeat 0 --cov-mode 1 -k 0 -c 0.5 --max-seq-len 65535 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --threads 256 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Database size: 1 type: Aminoacid\n",
      "Reduced amino acid alphabet: (A S T) (C) (D B N) (E Q Z) (F Y) (G) (H) (I V) (K R) (L J M) (P) (W) (X) \n",
      "\n",
      "Generate k-mers list for 1 split\n",
      "[=================================================================] 1 0s 0ms\n",
      "Sort kmer 0h 0m 0s 0ms\n",
      "Sort by rep. sequence 0h 0m 0s 0ms\n",
      "Time for fill: 0h 0m 0s 0ms\n",
      "Time for merging to pref: 0h 0m 0s 2ms\n",
      "Time for processing: 0h 0m 0s 33ms\n",
      "rescorediagonal component/component_redundant_db component/component_redundant_db tmp/5251849818421459973/linclust/15506357871815186212/pref tmp/5251849818421459973/linclust/15506357871815186212/pref_rescore1 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --rescore-mode 0 --wrapped-scoring 0 --filter-hits 0 -e 0.001 -c 0.5 -a 0 --cov-mode 1 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --add-self-matches 0 --sort-results 0 --db-load-mode 0 --threads 256 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 1 0s 8ms\n",
      "Time for merging to pref_rescore1: 0h 0m 0s 380ms\n",
      "Time for processing: 0h 0m 0s 932ms\n",
      "clust component/component_redundant_db tmp/5251849818421459973/linclust/15506357871815186212/pref_rescore1 tmp/5251849818421459973/linclust/15506357871815186212/pre_clust --cluster-mode 3 --max-iterations 1000 --similarity-type 2 --threads 256 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Greedy Low Mem\n",
      "Total time: 0h 0m 0s 14ms\n",
      "\n",
      "Size of the sequence database: 1\n",
      "Size of the alignment database: 1\n",
      "Number of clusters: 1\n",
      "\n",
      "Writing results 0h 0m 0s 0ms\n",
      "Time for merging to pre_clust: 0h 0m 0s 2ms\n",
      "Time for processing: 0h 0m 0s 112ms\n",
      "createsubdb tmp/5251849818421459973/linclust/15506357871815186212/order_redundancy component/component_redundant_db tmp/5251849818421459973/linclust/15506357871815186212/input_step_redundancy -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step_redundancy: 0h 0m 0s 2ms\n",
      "Time for processing: 0h 0m 0s 13ms\n",
      "createsubdb tmp/5251849818421459973/linclust/15506357871815186212/order_redundancy tmp/5251849818421459973/linclust/15506357871815186212/pref tmp/5251849818421459973/linclust/15506357871815186212/pref_filter1 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to pref_filter1: 0h 0m 0s 2ms\n",
      "Time for processing: 0h 0m 0s 10ms\n",
      "filterdb tmp/5251849818421459973/linclust/15506357871815186212/pref_filter1 tmp/5251849818421459973/linclust/15506357871815186212/pref_filter2 --filter-file tmp/5251849818421459973/linclust/15506357871815186212/order_redundancy --threads 256 --compressed 0 -v 3 \n",
      "\n",
      "Filtering using file(s)\n",
      "[=================================================================] 1 0s 8ms\n",
      "Time for merging to pref_filter2: 0h 0m 0s 360ms\n",
      "Time for processing: 0h 0m 0s 946ms\n",
      "rescorediagonal tmp/5251849818421459973/linclust/15506357871815186212/input_step_redundancy tmp/5251849818421459973/linclust/15506357871815186212/input_step_redundancy tmp/5251849818421459973/linclust/15506357871815186212/pref_filter2 tmp/5251849818421459973/linclust/15506357871815186212/pref_rescore2 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --rescore-mode 1 --wrapped-scoring 0 --filter-hits 1 -e 0.001 -c 0.5 -a 0 --cov-mode 1 --min-seq-id 0.3 --min-aln-len 0 --seq-id-mode 0 --add-self-matches 0 --sort-results 0 --db-load-mode 0 --threads 256 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 1 0s 15ms\n",
      "Time for merging to pref_rescore2: 0h 0m 0s 362ms\n",
      "Time for processing: 0h 0m 1s 180ms\n",
      "align tmp/5251849818421459973/linclust/15506357871815186212/input_step_redundancy tmp/5251849818421459973/linclust/15506357871815186212/input_step_redundancy tmp/5251849818421459973/linclust/15506357871815186212/pref_rescore2 tmp/5251849818421459973/linclust/15506357871815186212/aln --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.3 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.5 --cov-mode 1 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 256 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 1 type: Aminoacid\n",
      "Target database size: 1 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 1 0s 0ms\n",
      "Time for merging to aln: 0h 0m 0s 3ms\n",
      "1 alignments calculated\n",
      "1 sequence pairs passed the thresholds (1.000000 of overall calculated)\n",
      "1.000000 hits per query sequence\n",
      "Time for processing: 0h 0m 0s 92ms\n",
      "clust tmp/5251849818421459973/linclust/15506357871815186212/input_step_redundancy tmp/5251849818421459973/linclust/15506357871815186212/aln tmp/5251849818421459973/linclust/15506357871815186212/clust --cluster-mode 3 --max-iterations 1000 --similarity-type 2 --threads 256 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Greedy Low Mem\n",
      "Total time: 0h 0m 0s 13ms\n",
      "\n",
      "Size of the sequence database: 1\n",
      "Size of the alignment database: 1\n",
      "Number of clusters: 1\n",
      "\n",
      "Writing results 0h 0m 0s 0ms\n",
      "Time for merging to clust: 0h 0m 0s 2ms\n",
      "Time for processing: 0h 0m 0s 21ms\n",
      "mergeclusters component/component_redundant_db tmp/5251849818421459973/clu_redundancy tmp/5251849818421459973/linclust/15506357871815186212/pre_clust tmp/5251849818421459973/linclust/15506357871815186212/clust --threads 256 --compressed 0 -v 3 \n",
      "\n",
      "Clustering step 1\n",
      "[=================================================================] 1 0s 7ms\n",
      "Clustering step 2\n",
      "[=================================================================] 1 0s 21ms\n",
      "Write merged clustering\n",
      "[=================================================================] 1 0s 514ms\n",
      "Time for merging to clu_redundancy: 0h 0m 0s 375ms\n",
      "Time for processing: 0h 0m 0s 972ms\n",
      "createsubdb tmp/5251849818421459973/clu_redundancy component/component_redundant_db tmp/5251849818421459973/input_step_redundancy -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step_redundancy: 0h 0m 0s 4ms\n",
      "Time for processing: 0h 0m 0s 16ms\n",
      "prefilter tmp/5251849818421459973/input_step_redundancy tmp/5251849818421459973/input_step_redundancy tmp/5251849818421459973/pref_step0 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 1 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.5 --cov-mode 1 --comp-bias-corr 0 --comp-bias-corr-scale 1 --diag-score 0 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --mask-n-repeat 0 --min-ungapped-score 0 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 256 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 1 type: Aminoacid\n",
      "Estimated memory consumption: 977M\n",
      "Target database size: 1 type: Aminoacid\n",
      "Index table k-mer threshold: 154 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 1 0s 0ms\n",
      "Index table: Masked residues: 0\n",
      "No k-mer could be extracted for the database tmp/5251849818421459973/input_step_redundancy.\n",
      "Maybe the sequences length is less than 14 residues.\n",
      "Index table: fill\n",
      "[=================================================================] 1 0s 0ms\n",
      "Index statistics\n",
      "Entries:          0\n",
      "DB size:          488 MB\n",
      "Avg k-mer size:   0.000000\n",
      "Top 10 k-mers\n",
      "    AAAAAA\t0\n",
      "    AAAAAA\t0\n",
      "    AAAAAA\t0\n",
      "    AAAAAA\t0\n",
      "    AAAAAA\t0\n",
      "    AAAAAA\t0\n",
      "    AAAAAA\t0\n",
      "    AAAAAA\t0\n",
      "    AAAAAA\t0\n",
      "    AAAAAA\t0\n",
      "Time for index table init: 0h 0m 0s 466ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 154\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 1\n",
      "Target db start 1 to 1\n",
      "[=================================================================] 1 0s 0ms\n",
      "\n",
      "0.000000 k-mers per position\n",
      "0 DB matches per sequence\n",
      "0 overflows\n",
      "1 sequences passed prefiltering per query sequence\n",
      "1 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step0: 0h 0m 0s 3ms\n",
      "Time for processing: 0h 0m 2s 151ms\n",
      "align tmp/5251849818421459973/input_step_redundancy tmp/5251849818421459973/input_step_redundancy tmp/5251849818421459973/pref_step0 tmp/5251849818421459973/aln_step0 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.3 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.5 --cov-mode 1 --max-seq-len 65535 --comp-bias-corr 0 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 256 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 1 type: Aminoacid\n",
      "Target database size: 1 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 1 0s 0ms\n",
      "Time for merging to aln_step0: 0h 0m 0s 3ms\n",
      "1 alignments calculated\n",
      "1 sequence pairs passed the thresholds (1.000000 of overall calculated)\n",
      "1.000000 hits per query sequence\n",
      "Time for processing: 0h 0m 0s 11ms\n",
      "clust tmp/5251849818421459973/input_step_redundancy tmp/5251849818421459973/aln_step0 tmp/5251849818421459973/clu_step0 --cluster-mode 3 --max-iterations 1000 --similarity-type 2 --threads 256 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Greedy Low Mem\n",
      "Total time: 0h 0m 0s 14ms\n",
      "\n",
      "Size of the sequence database: 1\n",
      "Size of the alignment database: 1\n",
      "Number of clusters: 1\n",
      "\n",
      "Writing results 0h 0m 0s 0ms\n",
      "Time for merging to clu_step0: 0h 0m 0s 2ms\n",
      "Time for processing: 0h 0m 0s 22ms\n",
      "createsubdb tmp/5251849818421459973/clu_step0 tmp/5251849818421459973/input_step_redundancy tmp/5251849818421459973/input_step1 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step1: 0h 0m 0s 2ms\n",
      "Time for processing: 0h 0m 0s 14ms\n",
      "prefilter tmp/5251849818421459973/input_step1 tmp/5251849818421459973/input_step1 tmp/5251849818421459973/pref_step1 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 3 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.5 --cov-mode 1 --comp-bias-corr 1 --comp-bias-corr-scale 1 --diag-score 1 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --mask-n-repeat 0 --min-ungapped-score 15 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 256 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 1 type: Aminoacid\n",
      "Estimated memory consumption: 977M\n",
      "Target database size: 1 type: Aminoacid\n",
      "Index table k-mer threshold: 136 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 1 0s 0ms\n",
      "Index table: Masked residues: 0\n",
      "No k-mer could be extracted for the database tmp/5251849818421459973/input_step1.\n",
      "Maybe the sequences length is less than 14 residues.\n",
      "Index table: fill\n",
      "[=================================================================] 1 0s 0ms\n",
      "Index statistics\n",
      "Entries:          0\n",
      "DB size:          488 MB\n",
      "Avg k-mer size:   0.000000\n",
      "Top 10 k-mers\n",
      "    AAAAAA\t0\n",
      "    AAAAAA\t0\n",
      "    AAAAAA\t0\n",
      "    AAAAAA\t0\n",
      "    AAAAAA\t0\n",
      "    AAAAAA\t0\n",
      "    AAAAAA\t0\n",
      "    AAAAAA\t0\n",
      "    AAAAAA\t0\n",
      "    AAAAAA\t0\n",
      "Time for index table init: 0h 0m 0s 466ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 136\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 1\n",
      "Target db start 1 to 1\n",
      "[=================================================================] 1 0s 0ms\n",
      "\n",
      "0.000000 k-mers per position\n",
      "0 DB matches per sequence\n",
      "0 overflows\n",
      "1 sequences passed prefiltering per query sequence\n",
      "1 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step1: 0h 0m 0s 3ms\n",
      "Time for processing: 0h 0m 2s 148ms\n",
      "align tmp/5251849818421459973/input_step1 tmp/5251849818421459973/input_step1 tmp/5251849818421459973/pref_step1 tmp/5251849818421459973/aln_step1 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.3 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.5 --cov-mode 1 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 256 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 1 type: Aminoacid\n",
      "Target database size: 1 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 1 0s 0ms\n",
      "Time for merging to aln_step1: 0h 0m 0s 3ms\n",
      "1 alignments calculated\n",
      "1 sequence pairs passed the thresholds (1.000000 of overall calculated)\n",
      "1.000000 hits per query sequence\n",
      "Time for processing: 0h 0m 0s 11ms\n",
      "clust tmp/5251849818421459973/input_step1 tmp/5251849818421459973/aln_step1 tmp/5251849818421459973/clu_step1 --cluster-mode 3 --max-iterations 1000 --similarity-type 2 --threads 256 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Greedy Low Mem\n",
      "Total time: 0h 0m 0s 14ms\n",
      "\n",
      "Size of the sequence database: 1\n",
      "Size of the alignment database: 1\n",
      "Number of clusters: 1\n",
      "\n",
      "Writing results 0h 0m 0s 0ms\n",
      "Time for merging to clu_step1: 0h 0m 0s 2ms\n",
      "Time for processing: 0h 0m 0s 23ms\n",
      "createsubdb tmp/5251849818421459973/clu_step1 tmp/5251849818421459973/input_step1 tmp/5251849818421459973/input_step2 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step2: 0h 0m 0s 2ms\n",
      "Time for processing: 0h 0m 0s 15ms\n",
      "prefilter tmp/5251849818421459973/input_step2 tmp/5251849818421459973/input_step2 tmp/5251849818421459973/pref_step2 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 5 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.5 --cov-mode 1 --comp-bias-corr 1 --comp-bias-corr-scale 1 --diag-score 1 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --mask-n-repeat 0 --min-ungapped-score 15 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 256 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 1 type: Aminoacid\n",
      "Estimated memory consumption: 977M\n",
      "Target database size: 1 type: Aminoacid\n",
      "Index table k-mer threshold: 118 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 1 0s 0ms\n",
      "Index table: Masked residues: 0\n",
      "No k-mer could be extracted for the database tmp/5251849818421459973/input_step2.\n",
      "Maybe the sequences length is less than 14 residues.\n",
      "Index table: fill\n",
      "[=================================================================] 1 0s 0ms\n",
      "Index statistics\n",
      "Entries:          0\n",
      "DB size:          488 MB\n",
      "Avg k-mer size:   0.000000\n",
      "Top 10 k-mers\n",
      "    AAAAAA\t0\n",
      "    AAAAAA\t0\n",
      "    AAAAAA\t0\n",
      "    AAAAAA\t0\n",
      "    AAAAAA\t0\n",
      "    AAAAAA\t0\n",
      "    AAAAAA\t0\n",
      "    AAAAAA\t0\n",
      "    AAAAAA\t0\n",
      "    AAAAAA\t0\n",
      "Time for index table init: 0h 0m 0s 464ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 118\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 1\n",
      "Target db start 1 to 1\n",
      "[=================================================================] 1 0s 0ms\n",
      "\n",
      "0.000000 k-mers per position\n",
      "0 DB matches per sequence\n",
      "0 overflows\n",
      "1 sequences passed prefiltering per query sequence\n",
      "1 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step2: 0h 0m 0s 3ms\n",
      "Time for processing: 0h 0m 2s 136ms\n",
      "align tmp/5251849818421459973/input_step2 tmp/5251849818421459973/input_step2 tmp/5251849818421459973/pref_step2 tmp/5251849818421459973/aln_step2 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.3 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.5 --cov-mode 1 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 256 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 1 type: Aminoacid\n",
      "Target database size: 1 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 1 0s 0ms\n",
      "Time for merging to aln_step2: 0h 0m 0s 3ms\n",
      "1 alignments calculated\n",
      "1 sequence pairs passed the thresholds (1.000000 of overall calculated)\n",
      "1.000000 hits per query sequence\n",
      "Time for processing: 0h 0m 0s 12ms\n",
      "clust tmp/5251849818421459973/input_step2 tmp/5251849818421459973/aln_step2 tmp/5251849818421459973/clu_step2 --cluster-mode 3 --max-iterations 1000 --similarity-type 2 --threads 256 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Greedy Low Mem\n",
      "Total time: 0h 0m 0s 14ms\n",
      "\n",
      "Size of the sequence database: 1\n",
      "Size of the alignment database: 1\n",
      "Number of clusters: 1\n",
      "\n",
      "Writing results 0h 0m 0s 0ms\n",
      "Time for merging to clu_step2: 0h 0m 0s 2ms\n",
      "Time for processing: 0h 0m 0s 23ms\n",
      "mergeclusters component/component_redundant_db component/component_clusters_redundant tmp/5251849818421459973/clu_redundancy tmp/5251849818421459973/clu_step0 tmp/5251849818421459973/clu_step1 tmp/5251849818421459973/clu_step2 --threads 256 --compressed 0 -v 3 \n",
      "\n",
      "Clustering step 1\n",
      "[=================================================================] 1 0s 7ms\n",
      "Clustering step 2\n",
      "[=================================================================] 1 0s 23ms\n",
      "Clustering step 3\n",
      "[=================================================================] 1 0s 38ms\n",
      "Clustering step 4\n",
      "[=================================================================] 1 0s 55ms\n",
      "Write merged clustering\n",
      "[=================================================================] 1 0s 577ms\n",
      "Time for merging to component_clusters_redundant: 0h 0m 0s 361ms\n",
      "Time for processing: 0h 0m 1s 131ms\n",
      "cluster process/process_redundant_db process/process_clusters_redundant tmp --min-seq-id 0.3 -c 0.5 --cov-mode 1 \n",
      "\n",
      "MMseqs Version:                     \t18.8cc5c\n",
      "Substitution matrix                 \taa:blosum62.out,nucl:nucleotide.out\n",
      "Seed substitution matrix            \taa:VTML80.out,nucl:nucleotide.out\n",
      "Sensitivity                         \t4\n",
      "k-mer length                        \t0\n",
      "Target search mode                  \t0\n",
      "k-score                             \tseq:2147483647,prof:2147483647\n",
      "Alphabet size                       \taa:21,nucl:5\n",
      "Max sequence length                 \t65535\n",
      "Max results per query               \t20\n",
      "Split database                      \t0\n",
      "Split mode                          \t2\n",
      "Split memory limit                  \t0\n",
      "Coverage threshold                  \t0.5\n",
      "Coverage mode                       \t1\n",
      "Compositional bias                  \t1\n",
      "Compositional bias scale            \t1\n",
      "Diagonal scoring                    \ttrue\n",
      "Exact k-mer matching                \t0\n",
      "Mask residues                       \t1\n",
      "Mask residues probability           \t0.9\n",
      "Mask lower case residues            \t0\n",
      "Mask lower letter repeating N times \t0\n",
      "Minimum diagonal score              \t15\n",
      "Selected taxa                       \t\n",
      "Include identical seq. id.          \tfalse\n",
      "Spaced k-mers                       \t1\n",
      "Preload mode                        \t0\n",
      "Pseudo count a                      \tsubstitution:1.100,context:1.400\n",
      "Pseudo count b                      \tsubstitution:4.100,context:5.800\n",
      "Spaced k-mer pattern                \t\n",
      "Local temporary path                \t\n",
      "Threads                             \t256\n",
      "Compressed                          \t0\n",
      "Verbosity                           \t3\n",
      "Add backtrace                       \tfalse\n",
      "Alignment mode                      \t3\n",
      "Alignment mode                      \t0\n",
      "Allow wrapped scoring               \tfalse\n",
      "E-value threshold                   \t0.001\n",
      "Seq. id. threshold                  \t0.3\n",
      "Min alignment length                \t0\n",
      "Seq. id. mode                       \t0\n",
      "Alternative alignments              \t0\n",
      "Max reject                          \t2147483647\n",
      "Max accept                          \t2147483647\n",
      "Score bias                          \t0\n",
      "Realign hits                        \tfalse\n",
      "Realign score bias                  \t-0.2\n",
      "Realign max seqs                    \t2147483647\n",
      "Correlation score weight            \t0\n",
      "Gap open cost                       \taa:11,nucl:5\n",
      "Gap extension cost                  \taa:1,nucl:2\n",
      "Zdrop                               \t40\n",
      "Rescore mode                        \t0\n",
      "Remove hits by seq. id. and coverage\tfalse\n",
      "Sort results                        \t0\n",
      "Cluster mode                        \t0\n",
      "Max connected component depth       \t1000\n",
      "Similarity type                     \t2\n",
      "Weight file name                    \t\n",
      "Cluster Weight threshold            \t0.9\n",
      "Single step clustering              \tfalse\n",
      "Cascaded clustering steps           \t3\n",
      "Cluster reassign                    \tfalse\n",
      "Remove temporary files              \tfalse\n",
      "Force restart with latest tmp       \tfalse\n",
      "MPI runner                          \t\n",
      "k-mers per sequence                 \t21\n",
      "Scale k-mers per sequence           \taa:0.000,nucl:0.200\n",
      "Adjust k-mer length                 \tfalse\n",
      "Shift hash                          \t67\n",
      "Include only extendable             \tfalse\n",
      "Skip repeating k-mers               \tfalse\n",
      "\n",
      "Set cluster sensitivity to -s 5.000000\n",
      "Set cluster mode GREEDY MEM\n",
      "Set cluster iterations to 3\n",
      "linclust process/process_redundant_db tmp/5175508480126964613/clu_redundancy tmp/5175508480126964613/linclust --cluster-mode 3 --max-iterations 1000 --similarity-type 2 --threads 256 --compressed 0 -v 3 --cluster-weight-threshold 0.9 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.3 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.5 --cov-mode 1 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --alph-size aa:13,nucl:5 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 --mask-n-repeat 0 -k 0 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --rescore-mode 0 --filter-hits 0 --sort-results 0 --remove-tmp-files 0 --force-reuse 0 \n",
      "\n",
      "kmermatcher process/process_redundant_db tmp/5175508480126964613/linclust/5807011666729238576/pref --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --alph-size aa:13,nucl:5 --min-seq-id 0.3 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 --mask-n-repeat 0 --cov-mode 1 -k 0 -c 0.5 --max-seq-len 65535 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --threads 256 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "kmermatcher process/process_redundant_db tmp/5175508480126964613/linclust/5807011666729238576/pref --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --alph-size aa:13,nucl:5 --min-seq-id 0.3 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 --mask-n-repeat 0 --cov-mode 1 -k 0 -c 0.5 --max-seq-len 65535 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --threads 256 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Database size: 1 type: Aminoacid\n",
      "Reduced amino acid alphabet: (A S T) (C) (D B N) (E Q Z) (F Y) (G) (H) (I V) (K R) (L J M) (P) (W) (X) \n",
      "\n",
      "Generate k-mers list for 1 split\n",
      "[=================================================================] 1 0s 0ms\n",
      "Sort kmer 0h 0m 0s 0ms\n",
      "Sort by rep. sequence 0h 0m 0s 0ms\n",
      "Time for fill: 0h 0m 0s 0ms\n",
      "Time for merging to pref: 0h 0m 0s 2ms\n",
      "Time for processing: 0h 0m 0s 34ms\n",
      "rescorediagonal process/process_redundant_db process/process_redundant_db tmp/5175508480126964613/linclust/5807011666729238576/pref tmp/5175508480126964613/linclust/5807011666729238576/pref_rescore1 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --rescore-mode 0 --wrapped-scoring 0 --filter-hits 0 -e 0.001 -c 0.5 -a 0 --cov-mode 1 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --add-self-matches 0 --sort-results 0 --db-load-mode 0 --threads 256 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 1 0s 7ms\n",
      "Time for merging to pref_rescore1: 0h 0m 0s 416ms\n",
      "Time for processing: 0h 0m 1s 194ms\n",
      "clust process/process_redundant_db tmp/5175508480126964613/linclust/5807011666729238576/pref_rescore1 tmp/5175508480126964613/linclust/5807011666729238576/pre_clust --cluster-mode 3 --max-iterations 1000 --similarity-type 2 --threads 256 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Greedy Low Mem\n",
      "Total time: 0h 0m 0s 15ms\n",
      "\n",
      "Size of the sequence database: 1\n",
      "Size of the alignment database: 1\n",
      "Number of clusters: 1\n",
      "\n",
      "Writing results 0h 0m 0s 0ms\n",
      "Time for merging to pre_clust: 0h 0m 0s 2ms\n",
      "Time for processing: 0h 0m 0s 107ms\n",
      "createsubdb tmp/5175508480126964613/linclust/5807011666729238576/order_redundancy process/process_redundant_db tmp/5175508480126964613/linclust/5807011666729238576/input_step_redundancy -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step_redundancy: 0h 0m 0s 2ms\n",
      "Time for processing: 0h 0m 0s 13ms\n",
      "createsubdb tmp/5175508480126964613/linclust/5807011666729238576/order_redundancy tmp/5175508480126964613/linclust/5807011666729238576/pref tmp/5175508480126964613/linclust/5807011666729238576/pref_filter1 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to pref_filter1: 0h 0m 0s 2ms\n",
      "Time for processing: 0h 0m 0s 10ms\n",
      "filterdb tmp/5175508480126964613/linclust/5807011666729238576/pref_filter1 tmp/5175508480126964613/linclust/5807011666729238576/pref_filter2 --filter-file tmp/5175508480126964613/linclust/5807011666729238576/order_redundancy --threads 256 --compressed 0 -v 3 \n",
      "\n",
      "Filtering using file(s)\n",
      "[=================================================================] 1 0s 8ms\n",
      "Time for merging to pref_filter2: 0h 0m 0s 388ms\n",
      "Time for processing: 0h 0m 0s 976ms\n",
      "rescorediagonal tmp/5175508480126964613/linclust/5807011666729238576/input_step_redundancy tmp/5175508480126964613/linclust/5807011666729238576/input_step_redundancy tmp/5175508480126964613/linclust/5807011666729238576/pref_filter2 tmp/5175508480126964613/linclust/5807011666729238576/pref_rescore2 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --rescore-mode 1 --wrapped-scoring 0 --filter-hits 1 -e 0.001 -c 0.5 -a 0 --cov-mode 1 --min-seq-id 0.3 --min-aln-len 0 --seq-id-mode 0 --add-self-matches 0 --sort-results 0 --db-load-mode 0 --threads 256 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 1 0s 8ms\n",
      "Time for merging to pref_rescore2: 0h 0m 0s 406ms\n",
      "Time for processing: 0h 0m 1s 156ms\n",
      "align tmp/5175508480126964613/linclust/5807011666729238576/input_step_redundancy tmp/5175508480126964613/linclust/5807011666729238576/input_step_redundancy tmp/5175508480126964613/linclust/5807011666729238576/pref_rescore2 tmp/5175508480126964613/linclust/5807011666729238576/aln --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.3 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.5 --cov-mode 1 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 256 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 1 type: Aminoacid\n",
      "Target database size: 1 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 1 0s 0ms\n",
      "Time for merging to aln: 0h 0m 0s 3ms\n",
      "1 alignments calculated\n",
      "1 sequence pairs passed the thresholds (1.000000 of overall calculated)\n",
      "1.000000 hits per query sequence\n",
      "Time for processing: 0h 0m 0s 108ms\n",
      "clust tmp/5175508480126964613/linclust/5807011666729238576/input_step_redundancy tmp/5175508480126964613/linclust/5807011666729238576/aln tmp/5175508480126964613/linclust/5807011666729238576/clust --cluster-mode 3 --max-iterations 1000 --similarity-type 2 --threads 256 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Greedy Low Mem\n",
      "Total time: 0h 0m 0s 14ms\n",
      "\n",
      "Size of the sequence database: 1\n",
      "Size of the alignment database: 1\n",
      "Number of clusters: 1\n",
      "\n",
      "Writing results 0h 0m 0s 0ms\n",
      "Time for merging to clust: 0h 0m 0s 2ms\n",
      "Time for processing: 0h 0m 0s 23ms\n",
      "mergeclusters process/process_redundant_db tmp/5175508480126964613/clu_redundancy tmp/5175508480126964613/linclust/5807011666729238576/pre_clust tmp/5175508480126964613/linclust/5807011666729238576/clust --threads 256 --compressed 0 -v 3 \n",
      "\n",
      "Clustering step 1\n",
      "[=================================================================] 1 0s 6ms\n",
      "Clustering step 2\n",
      "[=================================================================] 1 0s 21ms\n",
      "Write merged clustering\n",
      "[=================================================================] 1 0s 547ms\n",
      "Time for merging to clu_redundancy: 0h 0m 0s 413ms\n",
      "Time for processing: 0h 0m 1s 63ms\n",
      "createsubdb tmp/5175508480126964613/clu_redundancy process/process_redundant_db tmp/5175508480126964613/input_step_redundancy -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step_redundancy: 0h 0m 0s 2ms\n",
      "Time for processing: 0h 0m 0s 13ms\n",
      "prefilter tmp/5175508480126964613/input_step_redundancy tmp/5175508480126964613/input_step_redundancy tmp/5175508480126964613/pref_step0 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 1 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.5 --cov-mode 1 --comp-bias-corr 0 --comp-bias-corr-scale 1 --diag-score 0 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --mask-n-repeat 0 --min-ungapped-score 0 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 256 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 1 type: Aminoacid\n",
      "Estimated memory consumption: 977M\n",
      "Target database size: 1 type: Aminoacid\n",
      "Index table k-mer threshold: 154 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 1 0s 0ms\n",
      "Index table: Masked residues: 0\n",
      "No k-mer could be extracted for the database tmp/5175508480126964613/input_step_redundancy.\n",
      "Maybe the sequences length is less than 14 residues.\n",
      "Index table: fill\n",
      "[=================================================================] 1 0s 0ms\n",
      "Index statistics\n",
      "Entries:          0\n",
      "DB size:          488 MB\n",
      "Avg k-mer size:   0.000000\n",
      "Top 10 k-mers\n",
      "    AAAAAA\t0\n",
      "    AAAAAA\t0\n",
      "    AAAAAA\t0\n",
      "    AAAAAA\t0\n",
      "    AAAAAA\t0\n",
      "    AAAAAA\t0\n",
      "    AAAAAA\t0\n",
      "    AAAAAA\t0\n",
      "    AAAAAA\t0\n",
      "    AAAAAA\t0\n",
      "Time for index table init: 0h 0m 0s 467ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 154\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 1\n",
      "Target db start 1 to 1\n",
      "[=================================================================] 1 0s 0ms\n",
      "\n",
      "0.000000 k-mers per position\n",
      "0 DB matches per sequence\n",
      "0 overflows\n",
      "1 sequences passed prefiltering per query sequence\n",
      "1 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step0: 0h 0m 0s 3ms\n",
      "Time for processing: 0h 0m 2s 136ms\n",
      "align tmp/5175508480126964613/input_step_redundancy tmp/5175508480126964613/input_step_redundancy tmp/5175508480126964613/pref_step0 tmp/5175508480126964613/aln_step0 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.3 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.5 --cov-mode 1 --max-seq-len 65535 --comp-bias-corr 0 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 256 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 1 type: Aminoacid\n",
      "Target database size: 1 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 1 0s 0ms\n",
      "Time for merging to aln_step0: 0h 0m 0s 3ms\n",
      "1 alignments calculated\n",
      "1 sequence pairs passed the thresholds (1.000000 of overall calculated)\n",
      "1.000000 hits per query sequence\n",
      "Time for processing: 0h 0m 0s 13ms\n",
      "clust tmp/5175508480126964613/input_step_redundancy tmp/5175508480126964613/aln_step0 tmp/5175508480126964613/clu_step0 --cluster-mode 3 --max-iterations 1000 --similarity-type 2 --threads 256 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Greedy Low Mem\n",
      "Total time: 0h 0m 0s 14ms\n",
      "\n",
      "Size of the sequence database: 1\n",
      "Size of the alignment database: 1\n",
      "Number of clusters: 1\n",
      "\n",
      "Writing results 0h 0m 0s 0ms\n",
      "Time for merging to clu_step0: 0h 0m 0s 2ms\n",
      "Time for processing: 0h 0m 0s 23ms\n",
      "createsubdb tmp/5175508480126964613/clu_step0 tmp/5175508480126964613/input_step_redundancy tmp/5175508480126964613/input_step1 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step1: 0h 0m 0s 3ms\n",
      "Time for processing: 0h 0m 0s 16ms\n",
      "prefilter tmp/5175508480126964613/input_step1 tmp/5175508480126964613/input_step1 tmp/5175508480126964613/pref_step1 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 3 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.5 --cov-mode 1 --comp-bias-corr 1 --comp-bias-corr-scale 1 --diag-score 1 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --mask-n-repeat 0 --min-ungapped-score 15 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 256 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 1 type: Aminoacid\n",
      "Estimated memory consumption: 977M\n",
      "Target database size: 1 type: Aminoacid\n",
      "Index table k-mer threshold: 136 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 1 0s 0ms\n",
      "Index table: Masked residues: 0\n",
      "No k-mer could be extracted for the database tmp/5175508480126964613/input_step1.\n",
      "Maybe the sequences length is less than 14 residues.\n",
      "Index table: fill\n",
      "[=================================================================] 1 0s 0ms\n",
      "Index statistics\n",
      "Entries:          0\n",
      "DB size:          488 MB\n",
      "Avg k-mer size:   0.000000\n",
      "Top 10 k-mers\n",
      "    AAAAAA\t0\n",
      "    AAAAAA\t0\n",
      "    AAAAAA\t0\n",
      "    AAAAAA\t0\n",
      "    AAAAAA\t0\n",
      "    AAAAAA\t0\n",
      "    AAAAAA\t0\n",
      "    AAAAAA\t0\n",
      "    AAAAAA\t0\n",
      "    AAAAAA\t0\n",
      "Time for index table init: 0h 0m 0s 477ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 136\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 1\n",
      "Target db start 1 to 1\n",
      "[=================================================================] 1 0s 0ms\n",
      "\n",
      "0.000000 k-mers per position\n",
      "0 DB matches per sequence\n",
      "0 overflows\n",
      "1 sequences passed prefiltering per query sequence\n",
      "1 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step1: 0h 0m 0s 3ms\n",
      "Time for processing: 0h 0m 2s 149ms\n",
      "align tmp/5175508480126964613/input_step1 tmp/5175508480126964613/input_step1 tmp/5175508480126964613/pref_step1 tmp/5175508480126964613/aln_step1 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.3 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.5 --cov-mode 1 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 256 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 1 type: Aminoacid\n",
      "Target database size: 1 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 1 0s 0ms\n",
      "Time for merging to aln_step1: 0h 0m 0s 3ms\n",
      "1 alignments calculated\n",
      "1 sequence pairs passed the thresholds (1.000000 of overall calculated)\n",
      "1.000000 hits per query sequence\n",
      "Time for processing: 0h 0m 0s 13ms\n",
      "clust tmp/5175508480126964613/input_step1 tmp/5175508480126964613/aln_step1 tmp/5175508480126964613/clu_step1 --cluster-mode 3 --max-iterations 1000 --similarity-type 2 --threads 256 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Greedy Low Mem\n",
      "Total time: 0h 0m 0s 13ms\n",
      "\n",
      "Size of the sequence database: 1\n",
      "Size of the alignment database: 1\n",
      "Number of clusters: 1\n",
      "\n",
      "Writing results 0h 0m 0s 0ms\n",
      "Time for merging to clu_step1: 0h 0m 0s 2ms\n",
      "Time for processing: 0h 0m 0s 22ms\n",
      "createsubdb tmp/5175508480126964613/clu_step1 tmp/5175508480126964613/input_step1 tmp/5175508480126964613/input_step2 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step2: 0h 0m 0s 3ms\n",
      "Time for processing: 0h 0m 0s 19ms\n",
      "prefilter tmp/5175508480126964613/input_step2 tmp/5175508480126964613/input_step2 tmp/5175508480126964613/pref_step2 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 5 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.5 --cov-mode 1 --comp-bias-corr 1 --comp-bias-corr-scale 1 --diag-score 1 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --mask-n-repeat 0 --min-ungapped-score 15 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 256 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 1 type: Aminoacid\n",
      "Estimated memory consumption: 977M\n",
      "Target database size: 1 type: Aminoacid\n",
      "Index table k-mer threshold: 118 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 1 0s 0ms\n",
      "Index table: Masked residues: 0\n",
      "No k-mer could be extracted for the database tmp/5175508480126964613/input_step2.\n",
      "Maybe the sequences length is less than 14 residues.\n",
      "Index table: fill\n",
      "[=================================================================] 1 0s 0ms\n",
      "Index statistics\n",
      "Entries:          0\n",
      "DB size:          488 MB\n",
      "Avg k-mer size:   0.000000\n",
      "Top 10 k-mers\n",
      "    AAAAAA\t0\n",
      "    AAAAAA\t0\n",
      "    AAAAAA\t0\n",
      "    AAAAAA\t0\n",
      "    AAAAAA\t0\n",
      "    AAAAAA\t0\n",
      "    AAAAAA\t0\n",
      "    AAAAAA\t0\n",
      "    AAAAAA\t0\n",
      "    AAAAAA\t0\n",
      "Time for index table init: 0h 0m 0s 475ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 118\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 1\n",
      "Target db start 1 to 1\n",
      "[=================================================================] 1 0s 0ms\n",
      "\n",
      "0.000000 k-mers per position\n",
      "0 DB matches per sequence\n",
      "0 overflows\n",
      "1 sequences passed prefiltering per query sequence\n",
      "1 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step2: 0h 0m 0s 4ms\n",
      "Time for processing: 0h 0m 2s 165ms\n",
      "align tmp/5175508480126964613/input_step2 tmp/5175508480126964613/input_step2 tmp/5175508480126964613/pref_step2 tmp/5175508480126964613/aln_step2 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.3 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.5 --cov-mode 1 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 256 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 1 type: Aminoacid\n",
      "Target database size: 1 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 1 0s 0ms\n",
      "Time for merging to aln_step2: 0h 0m 0s 3ms\n",
      "1 alignments calculated\n",
      "1 sequence pairs passed the thresholds (1.000000 of overall calculated)\n",
      "1.000000 hits per query sequence\n",
      "Time for processing: 0h 0m 0s 12ms\n",
      "clust tmp/5175508480126964613/input_step2 tmp/5175508480126964613/aln_step2 tmp/5175508480126964613/clu_step2 --cluster-mode 3 --max-iterations 1000 --similarity-type 2 --threads 256 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Greedy Low Mem\n",
      "Total time: 0h 0m 0s 13ms\n",
      "\n",
      "Size of the sequence database: 1\n",
      "Size of the alignment database: 1\n",
      "Number of clusters: 1\n",
      "\n",
      "Writing results 0h 0m 0s 0ms\n",
      "Time for merging to clu_step2: 0h 0m 0s 2ms\n",
      "Time for processing: 0h 0m 0s 22ms\n",
      "mergeclusters process/process_redundant_db process/process_clusters_redundant tmp/5175508480126964613/clu_redundancy tmp/5175508480126964613/clu_step0 tmp/5175508480126964613/clu_step1 tmp/5175508480126964613/clu_step2 --threads 256 --compressed 0 -v 3 \n",
      "\n",
      "Clustering step 1\n",
      "[=================================================================] 1 0s 9ms\n",
      "Clustering step 2\n",
      "[=================================================================] 1 0s 24ms\n",
      "Clustering step 3\n",
      "[=================================================================] 1 0s 39ms\n",
      "Clustering step 4\n",
      "[=================================================================] 1 0s 54ms\n",
      "Write merged clustering\n",
      "[=================================================================] 1 0s 565ms\n",
      "Time for merging to process_clusters_redundant: 0h 0m 0s 443ms\n",
      "Time for processing: 0h 0m 1s 187ms\n",
      "Creating TSV files...\n",
      "createtsv function/function_redundant_db function/function_redundant_db function/function_clusters_redundant function/function_clusters_redundant.tsv \n",
      "\n",
      "MMseqs Version:                 \t18.8cc5c\n",
      "First sequence as representative\tfalse\n",
      "Target column                   \t1\n",
      "Add full header                 \tfalse\n",
      "Sequence source                 \t0\n",
      "Database output                 \tfalse\n",
      "Threads                         \t256\n",
      "Compressed                      \t0\n",
      "Verbosity                       \t3\n",
      "\n",
      "Time for merging to function_clusters_redundant.tsv: 0h 0m 0s 483ms\n",
      "Time for processing: 0h 0m 1s 150ms\n",
      "createtsv component/component_redundant_db component/component_redundant_db component/component_clusters_redundant component/component_clusters_redundant.tsv \n",
      "\n",
      "MMseqs Version:                 \t18.8cc5c\n",
      "First sequence as representative\tfalse\n",
      "Target column                   \t1\n",
      "Add full header                 \tfalse\n",
      "Sequence source                 \t0\n",
      "Database output                 \tfalse\n",
      "Threads                         \t256\n",
      "Compressed                      \t0\n",
      "Verbosity                       \t3\n",
      "\n",
      "Time for merging to component_clusters_redundant.tsv: 0h 0m 0s 502ms\n",
      "Time for processing: 0h 0m 1s 62ms\n",
      "createtsv process/process_redundant_db process/process_redundant_db process/process_clusters_redundant process/process_clusters_redundant.tsv \n",
      "\n",
      "MMseqs Version:                 \t18.8cc5c\n",
      "First sequence as representative\tfalse\n",
      "Target column                   \t1\n",
      "Add full header                 \tfalse\n",
      "Sequence source                 \t0\n",
      "Database output                 \tfalse\n",
      "Threads                         \t256\n",
      "Compressed                      \t0\n",
      "Verbosity                       \t3\n",
      "\n",
      "Time for merging to process_clusters_redundant.tsv: 0h 0m 0s 470ms\n",
      "Time for processing: 0h 0m 1s 15ms\n",
      "All tasks completed successfully.\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "CompletedProcess(args=['/shared/projects/deepmar/PlasmoFP_public/scripts/mmseq_cluster.sh', '/shared/projects/deepmar/data/bacteria_sequences/function_only/reduced_90_similarity_30', '/shared/projects/deepmar/data/bacteria_sequences/function_only/reduced_90_function_propagated.fasta', '/shared/projects/deepmar/data/bacteria_sequences/function_only/empty.fasta', '/shared/projects/deepmar/data/bacteria_sequences/function_only/empty.fasta', '0.3'], returncode=0)"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "subprocess.run([\"./mmseq_cluster.sh\",\n",
    "    \"/shared/projects/deepmar/data/bacteria_sequences/function_only/reduced_90_similarity_30\",\n",
    "    \"/shared/projects/deepmar/data/bacteria_sequences/function_only/reduced_90_function_propagated.fasta\",\n",
    "    \"/shared/projects/deepmar/data/bacteria_sequences/function_only/empty.fasta\",\n",
    "    \"/shared/projects/deepmar/data/bacteria_sequences/function_only/empty.fasta\",\n",
    "    \"0.3\"]\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Train, val, test split"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "There are 9826 clusters in the function dataset.\n"
     ]
    }
   ],
   "source": [
    "function_mmseq_out = pd.read_csv(\"reduced_90_similarity_30/function/function_clusters_redundant.tsv\", sep='\\t', header=None)\n",
    "function_mmseq_out.columns = ['Cluster_2', 'Entry']\n",
    "\n",
    "\n",
    "print(\"There are {} clusters in the function dataset.\".format(function_mmseq_out['Cluster_2'].nunique()))\n",
    "\n",
    "assert len(function_mmseq_out) == len(total_bacteria_function_GO)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [],
   "source": [
    "#merge with the original datasets\n",
    "function_merged = pd.merge(function_mmseq_out, total_bacteria_function_GO, on='Entry')\n",
    "\n",
    "assert len(function_merged) == len(function_mmseq_out)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Processing rows: 100%|| 150860/150860 [00:07<00:00, 19078.82it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Identifying unique and multi-occurrence clusters\n",
      "ShuffleSplit for training, validation, and test sets from multi occurrence clusters\n",
      "Extracting rows for each set\n",
      "Moving clusters with missing terms from validation/test to training\n",
      "Validation terms not in training: set()\n",
      "Test terms not in training: set()\n",
      "Terms below threshold (50 occurrences): {'GO:0008761', 'GO:0018549', 'GO:0016815', 'GO:0047492', 'GO:0047922', 'GO:0015478', 'GO:1990829', 'GO:0016409', 'GO:0050540', 'GO:0052823', 'GO:0102682', 'GO:0036041', 'GO:0052855', 'GO:0015527', 'GO:0015643', 'GO:0047355', 'GO:0016647', 'GO:0003681', 'GO:0046625', 'GO:0043802', 'GO:0043837', 'GO:0051425', 'GO:0018493', 'GO:0018459', 'GO:1901359', 'GO:0047727', 'GO:0050833', 'GO:0004366', 'GO:0052880', 'GO:0070026', 'GO:0004574', 'GO:0051020', 'GO:0160105', 'GO:0047120', 'GO:0033940', 'GO:0051903', 'GO:0015117', 'GO:0015519', 'GO:0047971', 'GO:0051009', 'GO:0008719', 'GO:0050231', 'GO:0102933', 'GO:0000822', 'GO:0016430', 'GO:0016672', 'GO:0018620', 'GO:0043811', 'GO:0004747', 'GO:0052762', 'GO:0140905', 'GO:0008743', 'GO:0033848', 'GO:0005516', 'GO:0071172', 'GO:0106411', 'GO:0043828', 'GO:0070677', 'GO:0015418', 'GO:0036311', 'GO:0102306', 'GO:0047699', 'GO:0099094', 'GO:0030973', 'GO:0052588', 'GO:0050563', 'GO:0015205', 'GO:0071713', 'GO:0047817', 'GO:0034023', 'GO:0046025', 'GO:0050235', 'GO:0004649', 'GO:0015187', 'GO:0008861', 'GO:0061598', 'GO:0015091', 'GO:0015333', 'GO:0003882', 'GO:0033767', 'GO:0047269', 'GO:0008910', 'GO:0009678', 'GO:0034639', 'GO:0008876', 'GO:0047475', 'GO:0043872', 'GO:0008691', 'GO:0004623', 'GO:0031220', 'GO:0043365', 'GO:0018112', 'GO:0061627', 'GO:0018845', 'GO:0005230', 'GO:0016724', 'GO:0015594', 'GO:0016716', 'GO:0015369', 'GO:0015195', 'GO:0047651', 'GO:0047360', 'GO:0015212', 'GO:0004633', 'GO:0047487', 'GO:0046553', 'GO:0015315', 'GO:0008705', 'GO:0004345', 'GO:0018452', 'GO:0120092', 'GO:0089714', 'GO:0004591', 'GO:1990610', 'GO:0052717', 'GO:0047813', 'GO:0031219', 'GO:0160136', 'GO:0061133', 'GO:0050781', 'GO:0018525', 'GO:0047827', 'GO:0033850', 'GO:0004073', 'GO:0005047', 'GO:0009024', 'GO:0047267', 'GO:0004341', 'GO:0032052', 'GO:0008816', 'GO:0071913', 'GO:0071933', 'GO:0015215', 'GO:0047847', 'GO:0047856', 'GO:0140721', 'GO:0010945', 'GO:0019207', 'GO:0015355', 'GO:1902444', 'GO:0004757', 'GO:0047286', 'GO:0003881', 'GO:0160103', 'GO:0019159', 'GO:0001761', 'GO:0010326', 'GO:0030611', 'GO:0004856', 'GO:0022872', 'GO:0033712', 'GO:0033982', 'GO:0047560', 'GO:0018031', 'GO:0022883', 'GO:0036173', 'GO:0008736', 'GO:0033806', 'GO:0016864', 'GO:0018730', 'GO:0030341', 'GO:0018576', 'GO:0102571', 'GO:0008811', 'GO:0001406', 'GO:0051741', 'GO:0050607', 'GO:0008875', 'GO:0016504', 'GO:0050002', 'GO:0060241', 'GO:0017057', 'GO:0033931', 'GO:0102335', 'GO:0008309', 'GO:0018454', 'GO:0018760', 'GO:0050132', 'GO:0009026', 'GO:0106008', 'GO:0016158', 'GO:0047982', 'GO:0016768', 'GO:0051978', 'GO:0047570', 'GO:0033229', 'GO:0050484', 'GO:0018502', 'GO:0051742', 'GO:0018564', 'GO:0047580', 'GO:0030059', 'GO:0008512', 'GO:0047614', 'GO:0140613', 'GO:0004802', 'GO:0046316', 'GO:0043893', 'GO:0034072', 'GO:0050479', 'GO:0004729', 'GO:0043799', 'GO:0005344', 'GO:0016297', 'GO:0016152', 'GO:0050001', 'GO:0008674', 'GO:0047974', 'GO:0001098', 'GO:0047921', 'GO:0015112', 'GO:0046428', 'GO:0016977', 'GO:0004197', 'GO:0004743', 'GO:0070330', 'GO:0050081', 'GO:0015661', 'GO:0047742', 'GO:0008107', 'GO:0050532', 'GO:0008911', 'GO:0045127', 'GO:0102082', 'GO:0016625', 'GO:0004512', 'GO:0005537', 'GO:0033917', 'GO:0033846', 'GO:0034011', 'GO:0005436', 'GO:0047472', 'GO:0033734', 'GO:0002134', 'GO:0008807', 'GO:0050318', 'GO:0033717', 'GO:0047445', 'GO:0050473', 'GO:0047575', 'GO:0001846', 'GO:0018659', 'GO:0140827', 'GO:0047420', 'GO:0043919', 'GO:0004739', 'GO:0004733', 'GO:0008921', 'GO:0140618', 'GO:0019145', 'GO:0005280', 'GO:0018787', 'GO:0009927', 'GO:0048529', 'GO:0008989', 'GO:0052655', 'GO:0102285', 'GO:0050467', 'GO:0047810', 'GO:0004769', 'GO:0015530', 'GO:0047241', 'GO:0008972', 'GO:0160016', 'GO:0052657', 'GO:0008448', 'GO:0047029', 'GO:0003865', 'GO:0008793', 'GO:0004392', 'GO:0102031', 'GO:0030599', 'GO:0050559', 'GO:0008812', 'GO:0018860', 'GO:0120013', 'GO:1990404', 'GO:0030770', 'GO:0047609', 'GO:0016852', 'GO:0008766', 'GO:0160151', 'GO:0106408', 'GO:0008864', 'GO:0004488', 'GO:0015574', 'GO:0015067', 'GO:0008908', 'GO:0030337', 'GO:0043726', 'GO:0008695', 'GO:0043855', 'GO:0005186', 'GO:0030769', 'GO:0016980', 'GO:0018834', 'GO:0019164', 'GO:0022871', 'GO:0050617', 'GO:0000210', 'GO:0050606', 'GO:0008735', 'GO:0005368', 'GO:0106435', 'GO:0004508', 'GO:0033971', 'GO:0031127', 'GO:0051989', 'GO:0050152', 'GO:0045548', 'GO:0052716', 'GO:0001046', 'GO:0052929', 'GO:0031299', 'GO:0042410', 'GO:0050531', 'GO:0004492', 'GO:0005343', 'GO:0097077', 'GO:0018817', 'GO:0035485', 'GO:0016034', 'GO:0043838', 'GO:0061929', 'GO:0047067', 'GO:0047432', 'GO:0005287', 'GO:0070012', 'GO:0018818', 'GO:0070628', 'GO:0015528', 'GO:0047437', 'GO:0048256', 'GO:0047133', 'GO:0004462', 'GO:0043959', 'GO:0033705', 'GO:0047431', 'GO:0009044', 'GO:0015487', 'GO:0016993', 'GO:0047806', 'GO:0004412', 'GO:0019170', 'GO:0000293', 'GO:0015592', 'GO:0036380', 'GO:0033980', 'GO:0018849', 'GO:0004137', 'GO:0050587', 'GO:0003957', 'GO:0016937', 'GO:0052873', 'GO:0036373', 'GO:0019149', 'GO:0003958', 'GO:0050201', 'GO:0062147', 'GO:0052874', 'GO:0047916', 'GO:0009882', 'GO:0008849', 'GO:0043175', 'GO:0031403', 'GO:0042943', 'GO:0034257', 'GO:0102302', 'GO:0047818', 'GO:0047670', 'GO:0004737', 'GO:0047106', 'GO:0050528', 'GO:0061784', 'GO:0042973', 'GO:0008892', 'GO:0120227', 'GO:0033204', 'GO:0033920', 'GO:0043748', 'GO:0016663', 'GO:0008673', 'GO:0008817', 'GO:0102143', 'GO:0019154', 'GO:0008241', 'GO:0102198', 'GO:0019153', 'GO:0019161', 'GO:0043812', 'GO:0102525', 'GO:0015094', 'GO:0008894', 'GO:0016807', 'GO:0008378', 'GO:0008747', 'GO:0047930', 'GO:0050527', 'GO:0018669', 'GO:0016703', 'GO:0050417', 'GO:0047728', 'GO:0047043', 'GO:0050263', 'GO:0003964', 'GO:0034336', 'GO:0047903', 'GO:0005294', 'GO:0034022', 'GO:0070273', 'GO:0004848', 'GO:0047421', 'GO:1990002', 'GO:0015558', 'GO:0004155', 'GO:0043904', 'GO:0000099', 'GO:0050066', 'GO:0018764', 'GO:0140806', 'GO:0000908', 'GO:0008877', 'GO:0047516', 'GO:0050440', 'GO:0047071', 'GO:0000981', 'GO:0106343', 'GO:0120243', 'GO:0019147', 'GO:0030585', 'GO:0034026', 'GO:0004410', 'GO:0008146', 'GO:0000146', 'GO:0019135', 'GO:0022834', 'GO:0015446', 'GO:0036423', 'GO:0008868', 'GO:0052673', 'GO:0050119', 'GO:0004854', 'GO:0044715', 'GO:0070678', 'GO:0061135', 'GO:0033916', 'GO:0103020', 'GO:0031132', 'GO:0018523', 'GO:0050193', 'GO:0070968', 'GO:0090613', 'GO:0015437', 'GO:0018768', 'GO:0005477', 'GO:0050194', 'GO:0004144', 'GO:0102209', 'GO:1901567', 'GO:0016894', 'GO:0008934', 'GO:0004351', 'GO:0016416', 'GO:0047931', 'GO:0070284', 'GO:0008870', 'GO:0047547', 'GO:0003810', 'GO:0004420', 'GO:0018482', 'GO:0043781', 'GO:0009000', 'GO:0102299', 'GO:0047595', 'GO:0019168', 'GO:0009017', 'GO:0034071', 'GO:0050314', 'GO:0018507', 'GO:0018846', 'GO:0016715', 'GO:0022821', 'GO:0004528', 'GO:0003986', 'GO:0051212', 'GO:0050622', 'GO:0047880', 'GO:0005267', 'GO:0062146', 'GO:0030570', 'GO:0016827', 'GO:0033794', 'GO:1901982', 'GO:0018597', 'GO:0070404', 'GO:0030248', 'GO:0050500', 'GO:0032131', 'GO:0050371', 'GO:0106358', 'GO:0043884', 'GO:0102264', 'GO:0015095', 'GO:0047862', 'GO:0042944', 'GO:0015196', 'GO:0015113', 'GO:0050109', 'GO:0008327', 'GO:0005372', 'GO:0015489', 'GO:0004124', 'GO:0008919', 'GO:0008131', 'GO:0004664', 'GO:0050313', 'GO:0018633', 'GO:0003863', 'GO:0008912', 'GO:0051870', 'GO:0120536', 'GO:0051575', 'GO:0050840', 'GO:0042960', 'GO:0050053', 'GO:1990814', 'GO:0004715', 'GO:0008932', 'GO:0120225', 'GO:0050223', 'GO:0047126', 'GO:0097063', 'GO:0052868', 'GO:0004470', 'GO:0052764', 'GO:0050388', 'GO:0004575', 'GO:0050664', 'GO:0050039', 'GO:0015152', 'GO:0160013', 'GO:0047270', 'GO:0052934', 'GO:0018683', 'GO:0018114', 'GO:0050839', 'GO:0009030', 'GO:0005536', 'GO:0009027', 'GO:0160129', 'GO:0033695', 'GO:0010856', 'GO:0003972', 'GO:0005452', 'GO:0044716', 'GO:0015625', 'GO:0016682', 'GO:0015648', 'GO:0051907', 'GO:0003886', 'GO:0102169', 'GO:0050569', 'GO:0047571', 'GO:0048030', 'GO:0008884', 'GO:0016419', 'GO:0050346', 'GO:0004494', 'GO:0047934', 'GO:0008679', 'GO:0061712', 'GO:0101006', 'GO:0050317', 'GO:0034010', 'GO:0008456', 'GO:0031267', 'GO:0022869', 'GO:0042301', 'GO:0030492', 'GO:0061990', 'GO:1905502', 'GO:0032067', 'GO:0003831', 'GO:0004363', 'GO:0033983', 'GO:0004362', 'GO:0004473', 'GO:0033721', 'GO:0050175', 'GO:0070205', 'GO:0031216', 'GO:0018638', 'GO:0047711', 'GO:0036412', 'GO:0050054', 'GO:0030366', 'GO:0034029', 'GO:0032934', 'GO:0004364', 'GO:0008687', 'GO:0038024', 'GO:0015136', 'GO:0103074', 'GO:0018474', 'GO:0008902', 'GO:0018619', 'GO:0018859', 'GO:0102013', 'GO:0008667', 'GO:0033973', 'GO:0052734', 'GO:0050131', 'GO:0008898', 'GO:0004534', 'GO:0015387', 'GO:0018111', 'GO:0034909', 'GO:0031072', 'GO:0097100', 'GO:0005034', 'GO:0008860', 'GO:0044101', 'GO:0106286', 'GO:0019002', 'GO:0102164', 'GO:0000224', 'GO:0103116', 'GO:0018546', 'GO:0102971', 'GO:0036433', 'GO:0102986', 'GO:0034025', 'GO:0000215', 'GO:0050061', 'GO:0033923', 'GO:0030597', 'GO:0005290', 'GO:0004305', 'GO:0050355', 'GO:0043863', 'GO:0008831', 'GO:0033946', 'GO:0102252', 'GO:0015423', 'GO:0042900', 'GO:0030545', 'GO:0051109', 'GO:0017151', 'GO:0052601', 'GO:0036424', 'GO:0005254', 'GO:0102481', 'GO:0043778', 'GO:0003919', 'GO:0046820', 'GO:0016159', 'GO:0015930', 'GO:0047942', 'GO:0008745', 'GO:0043752', 'GO:0036469', 'GO:0033745', 'GO:0016216', 'GO:0051108', 'GO:0000009', 'GO:0047733', 'GO:0010487', 'GO:0050498', 'GO:0050338', 'GO:0080049', 'GO:0045302', 'GO:0047075', 'GO:0047829', 'GO:0034008', 'GO:0033296', 'GO:0018460', 'GO:0008460', 'GO:0047025', 'GO:0008970', 'GO:0050009', 'GO:0015044', 'GO:0016231', 'GO:0050242', 'GO:0050165', 'GO:0103026', 'GO:0004449', 'GO:0047298', 'GO:0090499', 'GO:0003877', 'GO:0120525', 'GO:0010334', 'GO:0008061', 'GO:0052660', 'GO:0047264', 'GO:0009486', 'GO:0102307', 'GO:0004573', 'GO:0030747', 'GO:0034237', 'GO:0052923', 'GO:0160141', 'GO:0004084', 'GO:0015211', 'GO:0106009', 'GO:0009010', 'GO:0043424', 'GO:0043139', 'GO:0008797', 'GO:0050627', 'GO:0002145', 'GO:0016530', 'GO:0047003', 'GO:0018677', 'GO:0034015', 'GO:0015207', 'GO:0004337', 'GO:0047739', 'GO:0018801', 'GO:0050605', 'GO:0097657', 'GO:0033910', 'GO:0050071', 'GO:0008863', 'GO:0022870', 'GO:0047548', 'GO:0160104', 'GO:0008836', 'GO:0004998', 'GO:0036054', 'GO:0051997', 'GO:0016018', 'GO:0052761', 'GO:0018695', 'GO:0047723', 'GO:0015436', 'GO:0015147', 'GO:0046917', 'GO:0008663', 'GO:0047302', 'GO:0033094', 'GO:0047533', 'GO:0102054', 'GO:0047919', 'GO:0071518', 'GO:0016689', 'GO:0034069', 'GO:0004239', 'GO:0061690', 'GO:0000703', 'GO:0061693', 'GO:0033727', 'GO:0047389', 'GO:0043818', 'GO:0008771', 'GO:0032448', 'GO:0070573', 'GO:0008428', 'GO:0008838', 'GO:0030283', 'GO:0047826', 'GO:0018511', 'GO:0102025', 'GO:0103031', 'GO:0019905', 'GO:0047023', 'GO:0034005', 'GO:0090483', 'GO:0018838', 'GO:0102100', 'GO:0046406', 'GO:0034785', 'GO:0004668', 'GO:0016713', 'GO:0047470', 'GO:0102673', 'GO:0015668', 'GO:0019864', 'GO:0008121', 'GO:0047943', 'GO:0045340', 'GO:0033966', 'GO:0008108', 'GO:0047444', 'GO:0004411', 'GO:0043786', 'GO:0004567', 'GO:0019887', 'GO:0015572', 'GO:0106409', 'GO:0015538', 'GO:0046810', 'GO:0018807', 'GO:0015226', 'GO:0047113', 'GO:0061751', 'GO:0050564', 'GO:0050144', 'GO:0046573', 'GO:0033897', 'GO:0018449', 'GO:0034875', 'GO:0047837', 'GO:0003993', 'GO:0061733', 'GO:0004576', 'GO:0008709', 'GO:0004314', 'GO:0090565', 'GO:0047632', 'GO:0008825', 'GO:0033232', 'GO:0004725', 'GO:0016691', 'GO:0018666', 'GO:0015503', 'GO:0000100', 'GO:0030975', 'GO:0003884', 'GO:0120242', 'GO:0050558', 'GO:0005249', 'GO:0004099', 'GO:0140825', 'GO:0046993', 'GO:0016432', 'GO:0070967', 'GO:0106350', 'GO:0120226', 'GO:0008446', 'GO:0015209', 'GO:0047652', 'GO:0047617', 'GO:0004772', 'GO:0008083', 'GO:0050076', 'GO:0052621', 'GO:0050292', 'GO:0106438', 'GO:0160117', 'GO:0050084', 'GO:0017172', 'GO:0052674', 'GO:0046920', 'GO:0051015', 'GO:0018774', 'GO:0015640', 'GO:0008737', 'GO:0015517', 'GO:0008106', 'GO:0015360', 'GO:0050268', 'GO:0008465', 'GO:0106344', 'GO:0048031', 'GO:0052577', 'GO:0052656', 'GO:0016692', 'GO:0043823', 'GO:0019158', 'GO:0052883', 'GO:0033764', 'GO:0102985', 'GO:0019901', 'GO:0050413', 'GO:0018740', 'GO:0016723', 'GO:0042602', 'GO:0008670', 'GO:0033752', 'GO:0047330', 'GO:0000006', 'GO:0102355', 'GO:0004869', 'GO:0015021', 'GO:0008517', 'GO:0052654', 'GO:0005542', 'GO:0052757', 'GO:0048028', 'GO:0004806', 'GO:0015265', 'GO:0008744', 'GO:0018861', 'GO:0030546', 'GO:0019156', 'GO:0015204', 'GO:0046526', 'GO:0000257', 'GO:0008964', 'GO:0047112', 'GO:0048502', 'GO:0019787', 'GO:0047200', 'GO:0033883', 'GO:0047463', 'GO:0003730', 'GO:0047569', 'GO:0016161', 'GO:0018477', 'GO:0015153', 'GO:0005244', 'GO:0016892', 'GO:0004034', 'GO:0004463', 'GO:0008126', 'GO:0018579', 'GO:0008917', 'GO:0015451', 'GO:1903136', 'GO:0061809', 'GO:0051669', 'GO:0102036', 'GO:0004458', 'GO:0009042', 'GO:0061594', 'GO:0047753', 'GO:0008240', 'GO:0047713', 'GO:0052597', 'GO:0008826', 'GO:0160107', 'GO:0052750', 'GO:0010698', 'GO:0015654', 'GO:0047631', 'GO:0047955', 'GO:0004081', 'GO:0018687', 'GO:0008923', 'GO:0004622', 'GO:0140826', 'GO:0042881', 'GO:0044374', 'GO:0140663', 'GO:0017124', 'GO:1990663', 'GO:0019534', 'GO:0004632', 'GO:0047866', 'GO:0043897', 'GO:0015046', 'GO:0015614', 'GO:0019165', 'GO:2001070', 'GO:0004316', 'GO:0008871', 'GO:0030743', 'GO:0015213', 'GO:0047443', 'GO:0090581', 'GO:0008715', 'GO:0047899', 'GO:0008336', 'GO:0004312', 'GO:0050555', 'GO:0050628', 'GO:0050044', 'GO:0004143', 'GO:0047954', 'GO:0062213', 'GO:0019003', 'GO:0047260', 'GO:0097098', 'GO:0009012', 'GO:0050545', 'GO:0004843', 'GO:0009824', 'GO:0043896', 'GO:0015368', 'GO:0047781', 'GO:0015128', 'GO:0022840', 'GO:0002953', 'GO:0032554', 'GO:0050334', 'GO:0160237', 'GO:0033904', 'GO:0010276', 'GO:0160142', 'GO:0034004', 'GO:0008430', 'GO:0004114', 'GO:0050470', 'GO:0047938', 'GO:0031218', 'GO:0047645', 'GO:0033957', 'GO:0048531', 'GO:0110068', 'GO:1902936', 'GO:0050618', 'GO:0008740', 'GO:0018623', 'GO:0103111', 'GO:0050538', 'GO:0036238', 'GO:0160137', 'GO:0102523', 'GO:0106356', 'GO:0047559', 'GO:0103012', 'GO:0016531', 'GO:0004037', 'GO:0052690', 'GO:0016156', 'GO:0004837', 'GO:0016418', 'GO:0018675', 'GO:0003978', 'GO:0047722', 'GO:0052875', 'GO:0015192', 'GO:0046524', 'GO:0052693', 'GO:0043894', 'GO:0106361', 'GO:0008294', 'GO:0103011', 'GO:0089702', 'GO:0071972', 'GO:0010176', 'GO:0052738', 'GO:0043918', 'GO:0018663', 'GO:0018796', 'GO:0061513', 'GO:0050609', 'GO:0034986', 'GO:0046353', 'GO:0008311', 'GO:0018690', 'GO:0080061', 'GO:0050149', 'GO:0015123', 'GO:0016966', 'GO:0043885', 'GO:0016719', 'GO:0004138', 'GO:0010852', 'GO:0008730', 'GO:0050573', 'GO:0047265', 'GO:0080030', 'GO:0102478', 'GO:0052751', 'GO:0089715', 'GO:0042959', 'GO:0070280', 'GO:0008906', 'GO:0033748', 'GO:0004343', 'GO:0018484', 'GO:0015568', 'GO:0102155', 'GO:0050490', 'GO:0120014', 'GO:0008191', 'GO:0033825', 'GO:0017174', 'GO:0042979', 'GO:0004631', 'GO:0043733', 'GO:0046421', 'GO:0097697', 'GO:0004339', 'GO:0016263', 'GO:0106363', 'GO:0102276', 'GO:0004311', 'GO:0102211', 'GO:0035368', 'GO:0043842', 'GO:0160210', 'GO:0002952', 'GO:0004446', 'GO:0050304', 'GO:0042834', 'GO:0016825', 'GO:0061630', 'GO:0106316', 'GO:0043737', 'GO:0032560', 'GO:0008444', 'GO:0033969', 'GO:0018762', 'GO:0160140', 'GO:0015024', 'GO:0043777', 'GO:0008936', 'GO:0019107', 'GO:0050299', 'GO:0050297', 'GO:0008885', 'GO:0102266', 'GO:0016274', 'GO:0003825', 'GO:0051007', 'GO:0098848', 'GO:0106099', 'GO:0008688', 'GO:0015038', 'GO:0018513', 'GO:0047693', 'GO:0043136', 'GO:0050487', 'GO:0047110', 'GO:0004777', 'GO:0071967', 'GO:0102318', 'GO:0004321', 'GO:0008973', 'GO:0004161', 'GO:0019144', 'GO:0032217', 'GO:0016699', 'GO:0016714', 'GO:0031404', 'GO:0015615', 'GO:0045301', 'GO:0050357', 'GO:0042806', 'GO:0043834', 'GO:0047794', 'GO:0052595', 'GO:0047895', 'GO:0050537', 'GO:0004888', 'GO:0141122', 'GO:0047529', 'GO:0050486', 'GO:0047100', 'GO:0016210', 'GO:0022832', 'GO:0015148', 'GO:0018626', 'GO:1990817', 'GO:0016929', 'GO:0015159', 'GO:0047789', 'GO:0022842', 'GO:0102391', 'GO:0008671', 'GO:0033942', 'GO:0047456', 'GO:0008251', 'GO:0032357', 'GO:0015438', 'GO:0061659', 'GO:0004756', 'GO:0004586', 'GO:0033978', 'GO:0036468', 'GO:0106226', 'GO:0018529', 'GO:0004555', 'GO:0009675', 'GO:0015168', 'GO:0070538', 'GO:0047809', 'GO:0003838', 'GO:0018739', 'GO:0047635', 'GO:0031386', 'GO:0103118', 'GO:0003953', 'GO:0140658', 'GO:0018798', 'GO:0061928', 'GO:0140784', 'GO:0047688', 'GO:0097256', 'GO:0035925', 'GO:0047600', 'GO:0004021', 'GO:0008452', 'GO:0047143', 'GO:0043957', 'GO:0043337', 'GO:0140272', 'GO:0140315', 'GO:0016848', 'GO:0052661', 'GO:0008700', 'GO:0008725', 'GO:0102168', 'GO:0035598', 'GO:0004531', 'GO:0015160', 'GO:0047448', 'GO:0003960', 'GO:0004471', 'GO:0170057', 'GO:0010855', 'GO:0010461', 'GO:0015391', 'GO:0004097', 'GO:0004617', 'GO:0120147', 'GO:0043895', 'GO:0016213', 'GO:0004787', 'GO:0008756', 'GO:0008851', 'GO:0052729', 'GO:0005297', 'GO:0140696', 'GO:0004402', 'GO:0050225', 'GO:0032182', 'GO:0004852', 'GO:0047878', 'GO:0018582', 'GO:0018759', 'GO:0046583', 'GO:0016630', 'GO:0050218', 'GO:0051219', 'GO:0106261', 'GO:0071617', 'GO:0004558', 'GO:0042287', 'GO:0019177', 'GO:0015092', 'GO:0050415', 'GO:0004451', 'GO:0005546', 'GO:0016463', 'GO:0016162', 'GO:0120284', 'GO:0016157', 'GO:0043792', 'GO:0033735', 'GO:0035438', 'GO:0042945', 'GO:0045735', 'GO:0005388', 'GO:0015441', 'GO:0043865', 'GO:0035250', 'GO:0004475', 'GO:0000700', 'GO:0004833', 'GO:0018535', 'GO:0051871', 'GO:0106329', 'GO:0004035', 'GO:0015188', 'GO:0097718', 'GO:0003994', 'GO:0140303', 'GO:0047838', 'GO:0047910', 'GO:0008824', 'GO:0018504', 'GO:0045442', 'GO:0008995', 'GO:0052881', 'GO:0008789', 'GO:0004106', 'GO:0050572', 'GO:0036308', 'GO:0018113', 'GO:0005548', 'GO:0043755', 'GO:0016867', 'GO:0003871', 'GO:0004436', 'GO:0005096', 'GO:0008953', 'GO:0051908', 'GO:0017110', 'GO:0045303', 'GO:0003785', 'GO:0047738', 'GO:0042907', 'GO:0009881', 'GO:0018800', 'GO:0045552', 'GO:0043130', 'GO:0004108', 'GO:0010291', 'GO:0071885', 'GO:0009496', 'GO:0140701', 'GO:0004467', 'GO:0008195', 'GO:0050485', 'GO:0102009', 'GO:0032422', 'GO:0015210', 'GO:0004566', 'GO:0047830', 'GO:0008790', 'GO:0047349', 'GO:0008828', 'GO:0015416', 'GO:0010854', 'GO:0043776', 'GO:0004605', 'GO:0015254', 'GO:0043822', 'GO:0045484', 'GO:0008672', 'GO:0008983', 'GO:0050520', 'GO:0047834', 'GO:0043887', 'GO:0009389', 'GO:0051003', 'GO:0033854', 'GO:0003885', 'GO:0099621', 'GO:0047643', 'GO:0015234', 'GO:0102210', 'GO:0015422', 'GO:0043718', 'GO:0061593', 'GO:0008872', 'GO:0047556', 'GO:0071667', 'GO:0018580', 'GO:0050060', 'GO:0019783', 'GO:0043857', 'GO:1990362', 'GO:0061459', 'GO:0008754', 'GO:0004058', 'GO:0008117', 'GO:0000309', 'GO:0004409', 'GO:0019180', 'GO:0019210', 'GO:0047689', 'GO:0010853', 'GO:0008980', 'GO:0015097', 'GO:0050241', 'GO:0005085', 'GO:0030270', 'GO:0106418', 'GO:0008854', 'GO:0019172', 'GO:0000721', 'GO:0044689', 'GO:1990238', 'GO:0016824', 'GO:0050519', 'GO:0031956', 'GO:0047004', 'GO:0008678', 'GO:0047868', 'GO:0043364', 'GO:0004466', 'GO:0120562', 'GO:0004658', 'GO:0017089', 'GO:0047808', 'GO:0090582', 'GO:0036348', 'GO:0034832', 'GO:0047436', 'GO:0140581', 'GO:0047372', 'GO:0008138', 'GO:0030788', 'GO:0030350', 'GO:0008728', 'GO:0047111', 'GO:0015506', 'GO:0034057', 'GO:0018530', 'GO:0015606', 'GO:0031176', 'GO:0043754', 'GO:0004044', 'GO:0070051', 'GO:0018858', 'GO:0102526', 'GO:0052706', 'GO:0005471', 'GO:0080016', 'GO:0015410', 'GO:0050023', 'GO:0018676', 'GO:0008312', 'GO:0000900', 'GO:0008888', 'GO:0046992', 'GO:0031409', 'GO:0047695', 'GO:0050279', 'GO:0099618', 'GO:0160138', 'GO:0004150', 'GO:0032567', 'GO:0015341', 'GO:0034700', 'GO:0008742', 'GO:0009028', 'GO:0042958', 'GO:0016262', 'GO:0004039', 'GO:0004503', 'GO:0004354', 'GO:0005527', 'GO:0051996', 'GO:0004616', 'GO:0097599', 'GO:0000102', 'GO:0033890', 'GO:0005199', 'GO:0033930', 'GO:0009016', 'GO:0044103', 'GO:1990464', 'GO:0003849', 'GO:0015155', 'GO:0043273', 'GO:0043751', 'GO:0003848', 'GO:0102313', 'GO:0102527', 'GO:0004776', 'GO:0004760', 'GO:0033812', 'GO:0004398', 'GO:0003711', 'GO:0030267', 'GO:0015473', 'GO:0030272', 'GO:0061503', 'GO:0062204', 'GO:0061463', 'GO:0030295', 'GO:0019209', 'GO:0003714', 'GO:1901236', 'GO:0016899', 'GO:0004065', 'GO:0018813', 'GO:0008867', 'GO:0003963', 'GO:0052736', 'GO:0003983', 'GO:0044325', 'GO:0019146', 'GO:0005347', 'GO:0030739', 'GO:0050278', 'GO:0090587', 'GO:0061522', 'GO:0022841', 'GO:0050599', 'GO:0019120', 'GO:0004491', 'GO:0035460', 'GO:0016697', 'GO:0008847', 'GO:0019113', 'GO:0090585', 'GO:0102162', 'GO:0008959', 'GO:0018578', 'GO:0140313', 'GO:0046962', 'GO:0031073', 'GO:0043530', 'GO:0034611', 'GO:0052816', 'GO:0140771', 'GO:0030410', 'GO:0018671', 'GO:0019139', 'GO:0008873', 'GO:0033929', 'GO:0008531', 'GO:0015927', 'GO:0047395', 'GO:0042134', 'GO:0008727', 'GO:0099619', 'GO:0070492', 'GO:0033265', 'GO:0018618', 'GO:0004067', 'GO:0001884', 'GO:0015575', 'GO:0004502', 'GO:0015485', 'GO:0032791', 'GO:0035496', 'GO:0031955', 'GO:0047356', 'GO:0009036', 'GO:0090612', 'GO:0106366', 'GO:0004557', 'GO:0008490', 'GO:0004015', 'GO:0052692', 'GO:0047590', 'GO:0070224', 'GO:0047196', 'GO:0030596', 'GO:0008801', 'GO:0010436', 'GO:0050112', 'GO:0047584', 'GO:0047553', 'GO:0015535', 'GO:0030701', 'GO:0015658', 'GO:0008920', 'GO:0140643', 'GO:0061599', 'GO:1903981', 'GO:0008514', 'GO:0034545', 'GO:0052773', 'GO:0004060', 'GO:0004559', 'GO:0015377', 'GO:0008134', 'GO:0050188', 'GO:0034480', 'GO:0004450', 'GO:0009374', 'GO:0004346', 'GO:0016649', 'GO:0032552', 'GO:0004301', 'GO:0141207', 'GO:0052916', 'GO:0004075', 'GO:0043176', 'GO:0008890', 'GO:0016985', 'GO:0102246', 'GO:0046905', 'GO:0106345', 'GO:1900750', 'GO:0033914', 'GO:0008554', 'GO:0019807', 'GO:0050499', 'GO:0102982', 'GO:0051018', 'GO:0004071', 'GO:0018509', 'GO:0004371', 'GO:0033877', 'GO:0015245', 'GO:0046558', 'GO:0043875', 'GO:0043866', 'GO:0102262', 'GO:0019531', 'GO:0001968', 'GO:0019116', 'GO:0033294', 'GO:0047121', 'GO:0140032', 'GO:0018531', 'GO:0004063', 'GO:0043801', 'GO:0030733', 'GO:1901611', 'GO:0030598', 'GO:0002057', 'GO:0043858', 'GO:0050446', 'GO:0033988', 'GO:0046570', 'GO:0018467', 'GO:0004775', 'GO:0008442', 'GO:1901515', 'GO:0103023', 'GO:0030378', 'GO:0047900', 'GO:0102319', 'GO:0102635', 'GO:0004514', 'GO:0018775', 'GO:0015516', 'GO:0004160', 'GO:0015408', 'GO:0033719', 'GO:0047530', 'GO:0018826', 'GO:0015137', 'GO:0044875', 'GO:0004795', 'GO:0019806', 'GO:0047850', 'GO:0019143', 'GO:0042393', 'GO:0047831', 'GO:0050215', 'GO:0034722', 'GO:0008713', 'GO:0043779', 'GO:0047681', 'GO:0042879', 'GO:0102224', 'GO:0018662', 'GO:0102810', 'GO:0050549', 'GO:0047074', 'GO:0008441', 'GO:0043906', 'GO:0047769', 'GO:0050483', 'GO:0018583', 'GO:0005302', 'GO:0008974', 'GO:0004230', 'GO:0034634', 'GO:0043825', 'GO:0050530', 'GO:0003688', 'GO:0033164', 'GO:0060589', 'GO:0043824', 'GO:0034597', 'GO:0033997', 'GO:0019784', 'GO:0004794', 'GO:0018812', 'GO:0005353', 'GO:0050510', 'GO:0016622', 'GO:0102294', 'GO:0015445', 'GO:0015104', 'GO:0033990', 'GO:0016595', 'GO:0106321', 'GO:0032266', 'GO:0044594', 'GO:0018461', 'GO:0004156', 'GO:0000062', 'GO:0017042', 'GO:0033740', 'GO:0008814', 'GO:0004564', 'GO:0008941', 'GO:0004746', 'GO:0033999', 'GO:0004368', 'GO:0047474', 'GO:0015923', 'GO:0032574', 'GO:0047736', 'GO:0033736', 'GO:0019862', 'GO:0102943', 'GO:0003868', 'GO:0008111', 'GO:0047044', 'GO:0004383', 'GO:0018541', 'GO:0033816', 'GO:0019874', 'GO:0018822', 'GO:0016751', 'GO:0055041', 'GO:0047694', 'GO:0047724', 'GO:0003834', 'GO:0047459', 'GO:0008734', 'GO:0047577', 'GO:0035516', 'GO:0004422', 'GO:0043920', 'GO:0004131', 'GO:0008417', 'GO:0034001', 'GO:0047206', 'GO:0035439', 'GO:0140103', 'GO:0050492', 'GO:0047078', 'GO:0046565', 'GO:0004135', 'GO:0050163', 'GO:0015577', 'GO:0009883', 'GO:0047995', 'GO:0046789', 'GO:0004842', 'GO:0016851', 'GO:0003841', 'GO:0019187', 'GO:0008073', 'GO:0008664', 'GO:0047334', 'GO:0008798', 'GO:0106355', 'GO:0008996', 'GO:0019808', 'GO:0047244', 'GO:0052877', 'GO:0043199', 'GO:0034736', 'GO:0050157', 'GO:0043860', 'GO:0046812', 'GO:0004104', 'GO:0043892', 'GO:0003974', 'GO:0033284', 'GO:0047799', 'GO:0030613', 'GO:0030600', 'GO:0047300', 'GO:0005298', 'GO:0050368', 'GO:0016759', 'GO:0004738', 'GO:0035312', 'GO:0015172', 'GO:0090584', 'GO:0050451', 'GO:2001065', 'GO:0036356', 'GO:0034028', 'GO:0004401', 'GO:0047985', 'GO:0140328', 'GO:0061596', 'GO:0102334', 'GO:0050135', 'GO:0090614', 'GO:0047059', 'GO:0019900', 'GO:0043750', 'GO:0046990', 'GO:0050547', 'GO:0051861', 'GO:0047935', 'GO:0018763', 'GO:0070819', 'GO:0004476', 'GO:0019179', 'GO:0004587', 'GO:0008398', 'GO:0140035', 'GO:0008260', 'GO:0042925', 'GO:0008738', 'GO:0018498', 'GO:0015276', 'GO:0106281', 'GO:0004018', 'GO:0008979', 'GO:0047418', 'GO:0017116', 'GO:1901981', 'GO:0018456', 'GO:0017118', 'GO:0015109', 'GO:0043798', 'GO:0001227', 'GO:0106312', 'GO:0015426', 'GO:0062158', 'GO:0008772', 'GO:0061595', 'GO:0016223', 'GO:0050625', 'GO:0047307', 'GO:0035591', 'GO:0018551', 'GO:0033828', 'GO:0070042', 'GO:0034938', 'GO:0015066', 'GO:0035539', 'GO:0015166', 'GO:0047282', 'GO:0061787', 'GO:0047679', 'GO:0070905', 'GO:0003842', 'GO:0045439', 'GO:0030612', 'GO:2001062', 'GO:0008804', 'GO:0052861', 'GO:0016989', 'GO:0000149', 'GO:0061980', 'GO:0015413', 'GO:0051002', 'GO:0071111', 'GO:0008785', 'GO:0019150', 'GO:0140306', 'GO:0009392', 'GO:0015928', 'GO:0036200', 'GO:0042054', 'GO:0008886', 'GO:0004064', 'GO:0002020', 'GO:0045436', 'GO:0009013', 'GO:0008965', 'GO:0018462', 'GO:0050340', 'GO:0042608', 'GO:0050129', 'GO:0004421', 'GO:0033723', 'GO:0018785', 'GO:0141148', 'GO:0140282', 'GO:0016174', 'GO:0018628', 'GO:0032843', 'GO:0047747', 'GO:0018627', 'GO:0047828', 'GO:0032556', 'GO:0017108', 'GO:0047721', 'GO:0004489', 'GO:0016664', 'GO:0047422', 'GO:0050361', 'GO:0070626', 'GO:0015225', 'GO:0140226', 'GO:0047732', 'GO:0009034', 'GO:0008901', 'GO:0004713', 'GO:0003712', 'GO:0050093', 'GO:0043714', 'GO:0015221', 'GO:0004485', 'GO:1901235', 'GO:0030409', 'GO:0033758', 'GO:0009039', 'GO:0141100', 'GO:0050611', 'GO:0062144', 'GO:0015189', 'GO:0047656', 'GO:0034784', 'GO:0043731', 'GO:0070290', 'GO:0001000', 'GO:0050626', 'GO:0140088', 'GO:0120533', 'GO:0005247', 'GO:0102029', 'GO:0015526', 'GO:0050021', 'GO:0048001', 'GO:0050656', 'GO:0047411', 'GO:0015183', 'GO:0015620', 'GO:0140087', 'GO:0050514', 'GO:0102550', 'GO:0047449', 'GO:0008668', 'GO:0102998', 'GO:0052866', 'GO:0008470', 'GO:0030429', 'GO:0005375', 'GO:0044750', 'GO:0015386', 'GO:0005275', 'GO:0022858', 'GO:1901612', 'GO:0047471', 'GO:0004582', 'GO:0047440', 'GO:0052636', 'GO:0015499', 'GO:0080130', 'GO:0008551', 'GO:0004613', 'GO:1990883', 'GO:0008494', 'GO:0043178', 'GO:0140740', 'GO:0008768', 'GO:0047519', 'GO:0033975', 'GO:0061605', 'GO:0160149', 'GO:0016427', 'GO:0008741', 'GO:0008887', 'GO:0018784', 'GO:0008909', 'GO:0070004', 'GO:0015414', 'GO:0018799', 'GO:0050583', 'GO:0016994', 'GO:0005242', 'GO:0071890', 'GO:0050096', 'GO:0050272', 'GO:0015233', 'GO:0015199', 'GO:0001681', 'GO:0008815', 'GO:0071771', 'GO:0090566', 'GO:0008273', 'GO:0018836', 'GO:0008453', 'GO:0043721', 'GO:0047849', 'GO:0004517', 'GO:0018478', 'GO:0052618', 'GO:0052730', 'GO:0015129', 'GO:0043817', 'GO:0047416', 'GO:0008999', 'GO:0043810', 'GO:0036382', 'GO:0102921', 'GO:0003939', 'GO:0008791', 'GO:0034228', 'GO:0043803', 'GO:0004126', 'GO:0003913', 'GO:0016623', 'GO:0047869', 'GO:0034038', 'GO:0003979', 'GO:0033188', 'GO:0004791', 'GO:0050575', 'GO:0032558', 'GO:0005342', 'GO:0106019', 'GO:0050401', 'GO:0102127', 'GO:0050565', 'GO:0042242', 'GO:0102770', 'GO:0001727', 'GO:0050192', 'GO:0015518', 'GO:0004805', 'GO:0030791', 'GO:0018706', 'GO:0030266', 'GO:0018668', 'GO:0031320', 'GO:0017169', 'GO:0016273', 'GO:0047488', 'GO:0015514', 'GO:0036307', 'GO:0047480', 'GO:0004123', 'GO:0071916', 'GO:1901973', 'GO:0050416', 'GO:0050121', 'GO:0004174', 'GO:0036199', 'GO:0015424', 'GO:0008110', 'GO:0003682', 'GO:0141093', 'GO:0047061', 'GO:0047911', 'GO:0015193', 'GO:0050138', 'GO:0004352', 'GO:0033732', 'GO:0015230', 'GO:0042970', 'GO:0016165', 'GO:0033776', 'GO:0047348', 'GO:0043800', 'GO:0047417', 'GO:0004308', 'GO:0050309', 'GO:0004750', 'GO:0102317', 'GO:0003941', 'GO:0004788', 'GO:0003973', 'GO:0030798', 'GO:0140679', 'GO:0005432', 'GO:0005272', 'GO:0033912', 'GO:0047266', 'GO:0009973', 'GO:0004620', 'GO:0005539', 'GO:0015045', 'GO:0042171', 'GO:0120233', 'GO:0050580', 'GO:0008918', 'GO:0047035', 'GO:0018803', 'GO:0047297', 'GO:0015651', 'GO:0015578', 'GO:0008707', 'GO:0018765', 'GO:0032136', 'GO:0019875', 'GO:0004127', 'GO:0160139', 'GO:0043958', 'GO:0043214', 'GO:0050100', 'GO:0047646', 'GO:0001618', 'GO:0043916', 'GO:0015085', 'GO:0050429', 'GO:0106242', 'GO:0050082', 'GO:0016153', 'GO:0050151', 'GO:1990827', 'GO:0005354', 'GO:0008473', 'GO:0009384', 'GO:0018570', 'GO:0047512', 'GO:0016420', 'GO:0043961', 'GO:0003756', 'GO:0003680', 'GO:0047294', 'GO:0043773', 'GO:1904121', 'GO:0034856', 'GO:0090599', 'GO:0008806', 'GO:0030572', 'GO:0047400', 'GO:0030772', 'GO:0044610', 'GO:0008830', 'GO:0052598', 'GO:0097266', 'GO:0050236', 'GO:0008986', 'GO:0102561', 'GO:0033608', 'GO:0008092', 'GO:0004146', 'GO:0090586', 'GO:0046589', 'GO:0032135', 'GO:0033797', 'GO:0070717', 'GO:0033905', 'GO:0004134', 'GO:0004069', 'GO:0036383', 'GO:0043876', 'GO:0015565', 'GO:0018601', 'GO:0036055', 'GO:0030614', 'GO:0036243', 'GO:0004053', 'GO:0140394', 'GO:0050303', 'GO:0015250', 'GO:0071968', 'GO:0018788', 'GO:0046555', 'GO:0008675', 'GO:0002196', 'GO:0008957', 'GO:0018756', 'GO:0043880', 'GO:0043883', 'GO:0030340', 'GO:0016760', 'GO:0001730', 'GO:0061749', 'GO:0046982', 'GO:0050114', 'GO:0050585', 'GO:0099620', 'GO:0015379', 'GO:0004614', 'GO:0008925', 'GO:0034000', 'GO:0047441', 'GO:0044603', 'GO:0047370', 'GO:0047490', 'GO:0062145', 'GO:0004353', 'GO:0050062', 'GO:2001071', 'GO:0036427', 'GO:0047371', 'GO:0004499', 'GO:0052913', 'GO:0106362', 'GO:0045145', 'GO:0016979', 'GO:0004111', 'GO:0102919', 'GO:0030699', 'GO:0097351', 'GO:0004090', 'GO:0036397', 'GO:0018741', 'GO:0050300', 'GO:0004530', 'GO:0140132', 'GO:0050133', 'GO:0009035', 'GO:0008714', 'GO:0043743', 'GO:0036524', 'GO:0004377', 'GO:0070025', 'GO:0047929', 'GO:0050394', 'GO:0036374', 'GO:0009977', 'GO:0050577', 'GO:0004758', 'GO:0036440', 'GO:0047554', 'GO:0043879', 'GO:0102388', 'GO:0000150', 'GO:0042279', 'GO:0008994', 'GO:0047091', 'GO:0016166', 'GO:0008732', 'GO:0008177', 'GO:0003839', 'GO:0034068', 'GO:0008752', 'GO:0050348', 'GO:0016414', 'GO:0047761', 'GO:0004322', 'GO:0042084', 'GO:0033677', 'GO:0098531', 'GO:0047683', 'GO:0015125', 'GO:0052911', 'GO:0052876', 'GO:0015216', 'GO:0047555', 'GO:0102007', 'GO:0033927', 'GO:0004169', 'GO:0047022', 'GO:0015551', 'GO:0047001', 'GO:0031992', 'GO:0106251', 'GO:0102223', 'GO:0047489', 'GO:0070335', 'GO:0004860', 'GO:0000295', 'GO:0031405', 'GO:0047536', 'GO:0052619', 'GO:0019151', 'GO:0005048', 'GO:0102096', 'GO:0003948', 'GO:0018489', 'GO:0003850', 'GO:0015174', 'GO:0031402', 'GO:0010851', 'GO:0047876', 'GO:0018457', 'GO:0018632', 'GO:0003779', 'GO:0050620', 'GO:0018684', 'GO:0019992', 'GO:0034200', 'GO:0050116', 'GO:0070991', 'GO:0018850', 'GO:0004493', 'GO:0005346', 'GO:0015231', 'GO:0015208', 'GO:0051060', 'GO:0090588', 'GO:0015108', 'GO:0015133', 'GO:0008843', 'GO:0008808', 'GO:0004767', 'GO:0015099', 'GO:0042957', 'GO:0050043', 'GO:0008271', 'GO:0047178', 'GO:0004115', 'GO:0051392', 'GO:0061677', 'GO:0005351', 'GO:0047482', 'GO:0018658', 'GO:0030371', 'GO:0015134', 'GO:0102195', 'GO:0008993', 'GO:0018729', 'GO:0008395', 'GO:0050797', 'GO:0000810', 'GO:0015496', 'GO:0005518', 'GO:0008720', 'GO:0034027', 'GO:0015049', 'GO:0050262', 'GO:0047777', 'GO:0102984', 'GO:0004033', 'GO:0035091', 'GO:0004165', 'GO:0018738', 'GO:0050118', 'GO:0008515', 'GO:0047540', 'GO:0000026', 'GO:0052749', 'GO:0052745', 'GO:0033963', 'GO:0022843', 'GO:0046994', 'GO:0048018', 'GO:0047798', 'GO:0000014', 'GO:0047310', 'GO:0047602', 'GO:1904091', 'GO:0042085', 'GO:0004607', 'GO:0008703', 'GO:0000701', 'GO:0018602', 'GO:0008879', 'GO:0103045', 'GO:0080025', 'GO:0050503', 'GO:0015665', 'GO:0098627', 'GO:0003858', 'GO:0004665', 'GO:0002135', 'GO:0140616', 'GO:0050286', 'GO:0003904', 'GO:0001108', 'GO:0071522', 'GO:0070818', 'GO:0050182', 'GO:0160222', 'GO:0016898', 'GO:0050290', 'GO:0017083', 'GO:0016632', 'GO:0103015', 'GO:0035374', 'GO:0050695', 'GO:0043720', 'GO:0018819', 'GO:0050613', 'GO:0106274', 'GO:0001530', 'GO:0004456', 'GO:0015184', 'GO:0047434', 'GO:0005528', 'GO:0015649', 'GO:0004551', 'GO:0050029', 'GO:0008977', 'GO:0043856', 'GO:0016784', 'GO:0004046', 'GO:0004419', 'GO:0102067', 'GO:0001653', 'GO:0042897', 'GO:0032557', 'GO:0004089', 'GO:0004344', 'GO:0050363', 'GO:0008519', 'GO:0045182', 'GO:0001848', 'GO:0008770', 'GO:0046574', 'GO:0140297', 'GO:0031217', 'GO:0047403', 'GO:0008662', 'GO:1903135', 'GO:0019825', 'GO:0044600', 'GO:0050022', 'GO:0120558', 'GO:0015143', 'GO:0017166', 'GO:0008922', 'GO:0047820', 'GO:0101005', 'GO:0042972', 'GO:0061602', 'GO:0050127', 'GO:0004589', 'GO:0018652', 'GO:0004435', 'GO:0047585', 'GO:0044729', 'GO:0047198', 'GO:0047257', 'GO:0008904', 'GO:0018667', 'GO:0008681', 'GO:0008641', 'GO:0043814', 'GO:0033853', 'GO:0048040', 'GO:0008677', 'GO:0018835', 'GO:0052737', 'GO:0035527', 'GO:0005363', 'GO:0071160', 'GO:0035375', 'GO:0061473', 'GO:0016696', 'GO:0052915', 'GO:0008748', 'GO:0015612', 'GO:0042169', 'GO:0098847', 'GO:0016618', 'GO:0008942', 'GO:0047154', 'GO:0030395', 'GO:0050020', 'GO:0018030', 'GO:0106141', 'GO:0005345', 'GO:1901238', 'GO:0047125', 'GO:0050385', 'GO:0004159', 'GO:0033863', 'GO:0004072', 'GO:0046870', 'GO:0097023', 'GO:0015932', 'GO:0008721', 'GO:0015119', 'GO:1902670', 'GO:0016995', 'GO:0001072', 'GO:0016412', 'GO:0050270', 'GO:0140750', 'GO:0004157', 'GO:0110153', 'GO:0030911', 'GO:0001716', 'GO:0005217', 'GO:0046997', 'GO:0050582', 'GO:0009018', 'GO:0015567', 'GO:0008998', 'GO:0080032', 'GO:0015531', 'GO:0047989', 'GO:0018648', 'GO:0015173', 'GO:0047473', 'GO:0120518', 'GO:0043744', 'GO:0008697', 'GO:0004638', 'GO:0052882', 'GO:0035888', 'GO:0016997', 'GO:1901474', 'GO:0050578', 'GO:0003875', 'GO:0005504', 'GO:0047419', 'GO:0043765', 'GO:0070694', 'GO:0102497', 'GO:0005391', 'GO:0047453', 'GO:0047304', 'GO:0033713', 'GO:0018823', 'GO:0035529', 'GO:0004846', 'GO:0050378', 'GO:0047280', 'GO:0004596', 'GO:0008127', 'GO:0050256', 'GO:0016795', 'GO:0070009', 'GO:0047804', 'GO:0050172', 'GO:0008484', 'GO:0050097', 'GO:0120509', 'GO:0106313', 'GO:0015296', 'GO:0043785', 'GO:0018554', 'GO:0043907', 'GO:0050825', 'GO:0050126', 'GO:0061785', 'GO:0015142', 'GO:0004133', 'GO:0102296', 'GO:0050086', 'GO:0016285', 'GO:0008478', 'GO:0047904', 'GO:0046522', 'GO:0045135', 'GO:0034024', 'GO:0019809', 'GO:0050232', 'GO:0015667', 'GO:0033958', 'GO:0047686', 'GO:0009038', 'GO:0047407', 'GO:0019826', 'GO:0009025', 'GO:0052832', 'GO:0018685', 'GO:0008767', 'GO:0047324', 'GO:0043208', 'GO:0035514', 'GO:0034458', 'GO:0019904', 'GO:0015185', 'GO:0047116', 'GO:0043722', 'GO:0008805', 'GO:1990411', 'GO:0018624', 'GO:0005262', 'GO:0050281', 'GO:0047296', 'GO:0015214', 'GO:0047045', 'GO:0005337', 'GO:1990594', 'GO:0019152', 'GO:0033847', 'GO:0050049', 'GO:0015444', 'GO:0033831', 'GO:1901702', 'GO:0000010', 'GO:0005324', 'GO:0005291', 'GO:0036370', 'GO:0050255', 'GO:0008939', 'GO:0141102', 'GO:0022885', 'GO:0008786', 'GO:0050269', 'GO:0016212', 'GO:0004145', 'GO:0009001', 'GO:0016215', 'GO:0015505', 'GO:0047611', 'GO:0018522', 'GO:0004181', 'GO:0004650', 'GO:0030695', 'GO:0019103', 'GO:0003950', 'GO:0004132', 'GO:0004657', 'GO:0008781', 'GO:0010354', 'GO:0022877', 'GO:0050176', 'GO:0045437', 'GO:0016717', 'GO:0005201', 'GO:0044547', 'GO:0008413', 'GO:0008788', 'GO:0008609', 'GO:0015483', 'GO:0018645', 'GO:0050179', 'GO:0008310', 'GO:0003961', 'GO:0036355', 'GO:0004062', 'GO:0047129', 'GO:0001070', 'GO:0048307', 'GO:0047589', 'GO:0018510', 'GO:0140691', 'GO:0008829', 'GO:0052680', 'GO:0018480', 'GO:0004136', 'GO:0003910', 'GO:0050089', 'GO:0010242', 'GO:0080031', 'GO:0034593', 'GO:0004061', 'GO:0050501', 'GO:0070728', 'GO:0052910', 'GO:0015411', 'GO:0070225', 'GO:0008660', 'GO:0052624', 'GO:0046790', 'GO:0042937', 'GO:0043236', 'GO:0004774', 'GO:0015180', 'GO:0050525', 'GO:0003796', 'GO:0160220', 'GO:0050450', 'GO:0046026', 'GO:0055056', 'GO:0103068', 'GO:0033882', 'GO:0052927', 'GO:1990929', 'GO:0016229', 'GO:0004198', 'GO:0033922', 'GO:0046509', 'GO:0052834', 'GO:0032778', 'GO:0008733', 'GO:0043532', 'GO:0050123', 'GO:0016206', 'GO:0033951', 'GO:0032451', 'GO:0008239', 'GO:0004328', 'GO:0045300', 'GO:0032564', 'GO:0031459', 'GO:0047725', 'GO:0004630', 'GO:0015156', 'GO:0015186', 'GO:0043717', 'GO:0050332', 'GO:0008308', 'GO:0050040', 'GO:0008684', 'GO:0008750', 'GO:0004113', 'GO:0052733', 'GO:0033795', 'GO:0005371', 'GO:0018506', 'GO:0033763', 'GO:0005221', 'GO:0035515', 'GO:0050207', 'GO:0003908', 'GO:0043682', 'GO:0018773', 'GO:0047748', 'GO:0018710', 'GO:0102193', 'GO:1990189', 'GO:0004032', 'GO:0047086', 'GO:0110130', 'GO:0110050', 'GO:0072545', 'GO:0141047', 'GO:0047405', 'GO:0010385', 'GO:0052684', 'GO:0005534', 'GO:0033219', 'GO:0046564', 'GO:0047680', 'GO:0034338', 'GO:0047070', 'GO:0033765', 'GO:0015616', 'GO:0050025', 'GO:0015481', 'GO:0106357', 'GO:0033918', 'GO:0052912', 'GO:0019810', 'GO:0008865', 'GO:0047770', 'GO:0018505', 'GO:0003917', 'GO:0052851', 'GO:0018577', 'GO:0004149', 'GO:0047669', 'GO:0000064', 'GO:0090589', 'GO:0102547', 'GO:0045152', 'GO:0010279', 'GO:0015106', 'GO:0034596', 'GO:0047663', 'GO:0015068', 'GO:0050301', 'GO:0047042', 'GO:0046556', 'GO:0008201', 'GO:0015020', 'GO:0140699', 'GO:0080007', 'GO:0043877', 'GO:0004655', 'GO:0003689', 'GO:0009046', 'GO:0004066', 'GO:0072582', 'GO:0015191', 'GO:0015419', 'GO:0103043', 'GO:0102482', 'GO:0000405', 'GO:0008429', 'GO:0008704', 'GO:0048039', 'GO:0000286', 'GO:0008296', 'GO:0120547', 'GO:0090564', 'GO:0047462', 'GO:0052933', 'GO:0018753', 'GO:0005412', 'GO:0140906', 'GO:0008376', 'GO:0120542', 'GO:0008889', 'GO:0050074', 'GO:0061770', 'GO:0043899', 'GO:0047124', 'GO:0051748', 'GO:0004477', 'GO:0008835', 'GO:0030848', 'GO:0047980', 'GO:0018805', 'GO:0015069', 'GO:0106415', 'GO:0009905', 'GO:0047303', 'GO:0008787', 'GO:0033939', 'GO:0046936', 'GO:0010333', 'GO:0004448', 'GO:0043873', 'GO:0015546', 'GO:0004423', 'GO:0022878', 'GO:0047605', 'GO:0010484', 'GO:0015080', 'GO:0033925', 'GO:0004355', 'GO:0102573', 'GO:0070336', 'GO:0008866', 'GO:0047905', 'GO:0016454', 'GO:0004077', 'GO:0046405', 'GO:0033691', 'GO:0000907', 'GO:0004722', 'GO:0001851', 'GO:0004338', 'GO:0030789', 'GO:0004438', 'GO:0070406', 'GO:0008951', 'GO:0050421', 'GO:0000121', 'GO:0050056', 'GO:0004736', 'GO:0015554', 'GO:0004782', 'GO:0052849', 'GO:0000334', 'GO:0050032', 'GO:0106437', 'GO:0030414', 'GO:0016411', 'GO:0051116', 'GO:0008118', 'GO:0008755', 'GO:0140911', 'GO:0050228', 'GO:0017061', 'GO:1990275', 'GO:0009679', 'GO:0003962', 'GO:0005547', 'GO:0015105', 'GO:0008809', 'GO:0042289', 'GO:0015647', 'GO:0008820', 'GO:0015540', 'GO:0008143', 'GO:0160097', 'GO:0015217', 'GO:0140487', 'GO:0160182', 'GO:0140769', 'GO:0080176', 'GO:0140175', 'GO:0004830', 'GO:0047622', 'GO:0050189', 'GO:0052614', 'GO:0033881', 'GO:0008949', 'GO:0050526', 'GO:0015093', 'GO:0047435', 'GO:0047013', 'GO:0030058', 'GO:0008869', 'GO:0003854', 'GO:0043870', 'GO:0033720', 'GO:0004505', 'GO:0008841', 'GO:0004629', 'GO:0015553', 'GO:0033968', 'GO:0017040', 'GO:0050237', 'GO:0015218', 'GO:0015501', 'GO:0004735', 'GO:0008794', 'GO:0032428', 'GO:0051063', 'GO:0047046', 'GO:0050077', 'GO:0000977', 'GO:0050393', 'GO:0120241', 'GO:0004568', 'GO:0033976', 'GO:0043909', 'GO:0008373', 'GO:0004369', 'GO:0042562', 'GO:0047936', 'GO:0050180', 'GO:0016669', 'GO:0009040', 'GO:0052679', 'GO:0008696', 'GO:0008960', 'GO:0008263', 'GO:0050208', 'GO:0050083', 'GO:0008823', 'GO:0018744', 'GO:0003870', 'GO:0070005', 'GO:0005496', 'GO:0015150', 'GO:0050319', 'GO:0008775', 'GO:0008903', 'GO:0160206', 'GO:0005350', 'GO:0102530', 'GO:0018786', 'GO:0045550', 'GO:0003938', 'GO:0140030', 'GO:0005304', 'GO:0102039', 'GO:0008297', 'GO:0047991', 'GO:0008706', 'GO:0071161', 'GO:0004602', 'GO:0004838', 'GO:0019148', 'GO:0008913', 'GO:0047972', 'GO:0008874', 'GO:0035731', 'GO:0018625', 'GO:0004867', 'GO:0018479', 'GO:0043325', 'GO:0061634', 'GO:0061579', 'GO:0042912', 'GO:0003914', 'GO:1901683', 'GO:0004001', 'GO:0102099', 'GO:0033792', 'GO:0106322', 'GO:0008779', 'GO:0001073', 'GO:0033907', 'GO:0047917', 'GO:0004130', 'GO:0047238', 'GO:0033972', 'GO:0004122', 'GO:0141098', 'GO:0034432', 'GO:0015169', 'GO:0015350', 'GO:0047343', 'GO:0042577', 'GO:0008813', 'GO:0070039', 'GO:0050574', 'GO:0015194', 'GO:0004793', 'GO:0018814', 'GO:0062192', 'GO:0047228', 'GO:0015430', 'GO:0018485', 'GO:0015220', 'GO:0140326', 'GO:0015190', 'GO:0005469', 'GO:0018824', 'GO:0018771', 'GO:0004866', 'GO:0015638', 'GO:0008301', 'GO:0050586', 'GO:0008962', 'GO:0047527', 'GO:0034899', 'GO:0033952', 'GO:0052744', 'GO:0052924', 'GO:0016670', 'GO:0140751', 'GO:0034386', 'GO:0043295', 'GO:0044605', 'GO:0008115', 'GO:0043821', 'GO:0030552', 'GO:0070497', 'GO:0044606', 'GO:0018592', 'GO:0004348', 'GO:0047840', 'GO:0008935', 'GO:0016207', 'GO:0062148', 'GO:0032794', 'GO:0044667', 'GO:0042931', 'GO:0001883', 'GO:0046559', 'GO:0047700', 'GO:0102212', 'GO:0004742', 'GO:0048472', 'GO:0008184', 'GO:0031172', 'GO:0016403', 'GO:0033994', 'GO:0008559', 'GO:0018750', 'GO:0047102', 'GO:0032356', 'GO:0050608', 'GO:0060698', 'GO:0019865', 'GO:0030729'}\n",
      "Filtering out 2947 terms from the training set.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/slurm_hmelcher_63507660.tmp/ipykernel_1950853/420060375.py:81: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  train_df[propagated_terms_col] = train_df[propagated_terms_col].apply(lambda terms: [term for term in terms if term not in terms_below_threshold])\n",
      "/tmp/slurm_hmelcher_63507660.tmp/ipykernel_1950853/420060375.py:85: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  val_df[propagated_terms_col] = val_df[propagated_terms_col].apply(lambda terms: [term for term in terms if term not in terms_below_threshold])\n",
      "/tmp/slurm_hmelcher_63507660.tmp/ipykernel_1950853/420060375.py:86: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  test_df[propagated_terms_col] = test_df[propagated_terms_col].apply(lambda terms: [term for term in terms if term not in terms_below_threshold])\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Filtering validation and test sets to remove terms below training threshold.\n",
      "Filtered function training set terms with less than 50 occurrences: {}\n",
      "Filtered function training set total terms: 1098\n",
      "Function training set shape: (129887, 14)\n",
      "Function validation set shape: (10804, 14)\n",
      "Function test set shape: (10130, 14)\n",
      "----\n"
     ]
    }
   ],
   "source": [
    "def prepare_data_and_split(go_terms_df, clusters_col='NA', terms_col='NA'):\n",
    "    cluster_to_terms = defaultdict(list)\n",
    "    term_to_clusters = defaultdict(set)\n",
    "\n",
    "    for _, row in tqdm(go_terms_df.iterrows(), total=go_terms_df.shape[0], desc=\"Processing rows\"):\n",
    "        cluster, terms = row[clusters_col], row[terms_col]\n",
    "        cluster_to_terms[cluster].extend(terms)\n",
    "        for term in terms:\n",
    "            term_to_clusters[term].add(cluster)\n",
    "\n",
    "    print(\"Identifying unique and multi-occurrence clusters\")\n",
    "\n",
    "    unique_clusters = {cluster for cluster, terms in cluster_to_terms.items()\n",
    "                       if all(len(term_to_clusters[term]) == 1 for term in terms)}\n",
    "    multi_clusters = set(cluster_to_terms.keys()) - unique_clusters\n",
    "\n",
    "    mlb = MultiLabelBinarizer()\n",
    "    all_terms = [terms for terms in cluster_to_terms.values()]\n",
    "    mlb.fit(all_terms)\n",
    "\n",
    "    print(\"ShuffleSplit for training, validation, and test sets from multi occurrence clusters\")\n",
    "    ss = ShuffleSplit(n_splits=1, test_size=0.1, random_state=38)\n",
    "    multi_clusters_list = list(multi_clusters)\n",
    "    train_val_idx, test_idx = next(ss.split(multi_clusters_list))\n",
    "\n",
    "    train_val_clusters = [multi_clusters_list[i] for i in train_val_idx]\n",
    "    ss_train_val = ShuffleSplit(n_splits=1, test_size=0.1, random_state=38)\n",
    "    train_idx, val_idx = next(ss_train_val.split(train_val_clusters))\n",
    "\n",
    "    train_clusters = [train_val_clusters[i] for i in train_idx] + list(unique_clusters)\n",
    "    val_clusters = [train_val_clusters[i] for i in val_idx]\n",
    "    test_clusters = [multi_clusters_list[i] for i in test_idx]\n",
    "\n",
    "    print(\"Extracting rows for each set\")\n",
    "    train_df = go_terms_df[go_terms_df[clusters_col].isin(train_clusters)]\n",
    "    val_df = go_terms_df[go_terms_df[clusters_col].isin(val_clusters)]\n",
    "    test_df = go_terms_df[go_terms_df[clusters_col].isin(test_clusters)]\n",
    "\n",
    "    train_terms = set(sum(train_df[terms_col].tolist(), []))\n",
    "    val_terms = set(sum(val_df[terms_col].tolist(), []))\n",
    "    test_terms = set(sum(test_df[terms_col].tolist(), []))\n",
    "\n",
    "    missing_val_terms = val_terms - train_terms\n",
    "    missing_test_terms = test_terms - train_terms\n",
    "\n",
    "    print(\"Moving clusters with missing terms from validation/test to training\")\n",
    "    if missing_val_terms:\n",
    "        val_clusters_to_move = [cluster for cluster in val_clusters if any(term in missing_val_terms for term in cluster_to_terms[cluster])]\n",
    "        train_clusters.extend(val_clusters_to_move)\n",
    "        val_clusters = list(set(val_clusters) - set(val_clusters_to_move))\n",
    "\n",
    "    if missing_test_terms:\n",
    "        test_clusters_to_move = [cluster for cluster in test_clusters if any(term in missing_test_terms for term in cluster_to_terms[cluster])]\n",
    "        train_clusters.extend(test_clusters_to_move)\n",
    "        test_clusters = list(set(test_clusters) - set(test_clusters_to_move))\n",
    "\n",
    "    train_df = go_terms_df[go_terms_df[clusters_col].isin(train_clusters)]\n",
    "    val_df = go_terms_df[go_terms_df[clusters_col].isin(val_clusters)]\n",
    "    test_df = go_terms_df[go_terms_df[clusters_col].isin(test_clusters)]\n",
    "\n",
    "    train_terms = set(sum(train_df[terms_col].tolist(), []))\n",
    "    val_terms = set(sum(val_df[terms_col].tolist(), []))\n",
    "    test_terms = set(sum(test_df[terms_col].tolist(), []))\n",
    "\n",
    "    missing_val_terms = val_terms - train_terms\n",
    "    missing_test_terms = test_terms - train_terms\n",
    "\n",
    "    print(\"Validation terms not in training:\", missing_val_terms)\n",
    "    print(\"Test terms not in training:\", missing_test_terms)\n",
    "\n",
    "    propagated_terms_col = 'Raw propagated GO terms'  \n",
    "\n",
    "    min_training_instances = 50 \n",
    "    train_terms_counter = Counter([term for terms in train_df[propagated_terms_col].tolist() for term in terms])\n",
    "    terms_below_threshold = {term for term, count in train_terms_counter.items() if count < min_training_instances}\n",
    "\n",
    "    print(f\"Terms below threshold ({min_training_instances} occurrences):\", terms_below_threshold)\n",
    "\n",
    "    if terms_below_threshold:\n",
    "        print(f\"Filtering out {len(terms_below_threshold)} terms from the training set.\")\n",
    "        train_df[propagated_terms_col] = train_df[propagated_terms_col].apply(lambda terms: [term for term in terms if term not in terms_below_threshold])\n",
    "        train_df = train_df[train_df[propagated_terms_col].apply(len) > 0]  # Remove rows with no terms after filtering\n",
    "\n",
    "    print(\"Filtering validation and test sets to remove terms below training threshold.\")\n",
    "    val_df[propagated_terms_col] = val_df[propagated_terms_col].apply(lambda terms: [term for term in terms if term not in terms_below_threshold])\n",
    "    test_df[propagated_terms_col] = test_df[propagated_terms_col].apply(lambda terms: [term for term in terms if term not in terms_below_threshold])\n",
    "\n",
    "    val_df = val_df[val_df[propagated_terms_col].apply(len) > 0]\n",
    "    test_df = test_df[test_df[propagated_terms_col].apply(len) > 0]\n",
    "\n",
    "    filtered_train_counts = Counter([term for terms in train_df[propagated_terms_col].tolist() for term in terms])\n",
    "    filtered_train_counter_terms_less_than_X = {term: count for term, count in filtered_train_counts.items() if count < min_training_instances}\n",
    "    print(\"Filtered function training set terms with less than {} occurrences:\".format(min_training_instances), filtered_train_counter_terms_less_than_X)\n",
    "    print(\"Filtered function training set total terms:\", len(filtered_train_counts))\n",
    "\n",
    "    return train_df, val_df, test_df, mlb\n",
    "\n",
    "\n",
    "function_train_df, function_val_df, function_test_df, function_mlb = prepare_data_and_split(function_merged, clusters_col='Cluster_2', terms_col='Raw GO terms')\n",
    "print(\"Function training set shape:\", function_train_df.shape)\n",
    "print(\"Function validation set shape:\", function_val_df.shape)\n",
    "print(\"Function test set shape:\", function_test_df.shape)\n",
    "print(\"----\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of GO terms in the function training set: 1885828\n",
      "Total number of unique GO terms in the function training set: 1098\n"
     ]
    }
   ],
   "source": [
    "temp_terms = function_train_df['Raw propagated GO terms'].tolist()\n",
    "temp_terms= [item for sublist in temp_terms for item in sublist]\n",
    "print(\"Total number of GO terms in the function training set:\", len(temp_terms))\n",
    "print(\"Total number of unique GO terms in the function training set:\", len(set(temp_terms)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Terms in validation set but not in training set: set()\n",
      "Terms in test set but not in training set: set()\n"
     ]
    }
   ],
   "source": [
    "#are there any terms that are not in the training set but are in the validation set and test set\n",
    "\n",
    "temp_terms_train = function_train_df['Raw propagated GO terms'].tolist()\n",
    "temp_terms_train = [item for sublist in temp_terms_train for item in sublist]\n",
    "temp_terms_val = function_val_df['Raw propagated GO terms'].tolist()\n",
    "temp_terms_val = [item for sublist in temp_terms_val for item in sublist]\n",
    "temp_terms_test = function_test_df['Raw propagated GO terms'].tolist()\n",
    "temp_terms_test = [item for sublist in temp_terms_test for item in sublist]\n",
    "\n",
    "print(\"Terms in validation set but not in training set:\", set(temp_terms_val) - set(temp_terms_train))\n",
    "print(\"Terms in test set but not in training set:\", set(temp_terms_test) - set(temp_terms_train))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mkdir: cannot create directory processed_data_90_30: File exists\n"
     ]
    }
   ],
   "source": [
    "! mkdir processed_data_90_30"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "os.chdir(\"processed_data_90_30\")\n",
    "function_train_df.to_csv(\"function_train.tsv\", sep=\"\\t\", index=False)\n",
    "function_val_df.to_csv(\"function_val.tsv\", sep=\"\\t\", index=False)\n",
    "function_test_df.to_csv(\"function_test.tsv\", sep=\"\\t\", index=False)\n",
    "\n",
    "#write file that has contains how many unique GO terms are in the training set\n",
    "function_train_terms = function_train_df['Raw propagated GO terms'].tolist()\n",
    "function_train_terms = [item for sublist in function_train_terms for item in sublist]\n",
    "function_train_terms_unique = len(set(function_train_terms))\n",
    "\n",
    "with open(\"number_of_terms_trained_on.txt\", \"w\") as f:\n",
    "    print(\"Function:\", len(set(function_train_terms)), file=f)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [],
   "source": [
    "with open(\"df_shapes.txt\", \"w\") as f:\n",
    "    print(\"Function train:\", function_train_df.shape, file=f)\n",
    "    print(\"Function val:\", function_val_df.shape, file=f)\n",
    "    print(\"Function test:\", function_test_df.shape, file=f)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [],
   "source": [
    "function_merged.to_csv(\"presplit_function_merged.tsv\", sep=\"\\t\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "ename": "AssertionError",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[31m---------------------------------------------------------------------------\u001b[39m",
      "\u001b[31mAssertionError\u001b[39m                            Traceback (most recent call last)",
      "\u001b[36mCell\u001b[39m\u001b[36m \u001b[39m\u001b[32mIn[37]\u001b[39m\u001b[32m, line 1\u001b[39m\n\u001b[32m----> \u001b[39m\u001b[32m1\u001b[39m \u001b[38;5;28;01massert\u001b[39;00m \u001b[38;5;28mlen\u001b[39m(function_train_df) + \u001b[38;5;28mlen\u001b[39m(function_val_df) + \u001b[38;5;28mlen\u001b[39m(function_test_df) == \u001b[38;5;28mlen\u001b[39m(function_merged)\n",
      "\u001b[31mAssertionError\u001b[39m: "
     ]
    }
   ],
   "source": [
    "assert len(function_train_df) + len(function_val_df) + len(function_test_df) == len(function_merged)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [],
   "source": [
    "#create fasta files \n",
    "seq_records_function_train = create_seq_records(function_train_df)\n",
    "seq_records_function_val = create_seq_records(function_val_df)\n",
    "seq_records_function_test = create_seq_records(function_test_df)\n",
    "\n",
    "\n",
    "assert len(seq_records_function_train) == function_train_df.shape[0]\n",
    "assert len(seq_records_function_val) == function_val_df.shape[0]\n",
    "assert len(seq_records_function_test) == function_test_df.shape[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "10130"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#write out the fasta files\n",
    "SeqIO.write(seq_records_function_train, \"function_train.fasta\", \"fasta\")\n",
    "SeqIO.write(seq_records_function_val, \"function_val.fasta\", \"fasta\")\n",
    "SeqIO.write(seq_records_function_test, \"function_test.fasta\", \"fasta\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Get embeddings"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/shared/ifbstor1/projects/deepmar/conda_environments/conda_envs/PlasmoFP/lib/python3.12/site-packages/tqdm/auto.py:21: TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html\n",
      "  from .autonotebook import tqdm as notebook_tqdm\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Embedding generation started!\n",
      "Output directory: embeddings\n",
      "Log file: embeddings/embedding_generation.log\n",
      "Verbose mode: OFF\n",
      "============================================================\n",
      "Using device: cuda\n",
      "2026-01-16 12:06:56,566 - INFO - Using device: cuda\n",
      "Found 1 FASTA file(s) to process:\n",
      "   1. reduced_90_function.fasta\n",
      "\n",
      "2026-01-16 12:06:56,568 - INFO - Found 1 FASTA file(s) to process\n",
      "Loading ProtT5 tokenizer and model...\n",
      "2026-01-16 12:06:56,570 - INFO - Loading ProtT5 tokenizer and model...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "You are using the default legacy behaviour of the <class 'transformers.models.t5.tokenization_t5.T5Tokenizer'>. This is expected, and simply means that the `legacy` (previous) behavior will be used so nothing changes for you. If you want to use the new behaviour, set `legacy=False`. This should only be set if you understand what it means, and thoroughly read the reason why this was added as explained in https://github.com/huggingface/transformers/pull/24565\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading TM-Vec model...\n",
      "2026-01-16 12:07:51,249 - INFO - Loading TM-Vec model...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Lightning automatically upgraded your loaded checkpoint from v1.5.8 to v2.6.0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint ../../../../../../projects/deepmar/data/tmvec_model_weights/tm_vec_cath_model.ckpt`\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Models loaded successfully on device: cuda\n",
      "2026-01-16 12:07:53,046 - INFO - Models loaded successfully on device: cuda\n",
      "Starting processing...\n",
      "============================================================\n",
      "[1/1] Processing: reduced_90_function.fasta\n",
      "Processing file: reduced_90_function.fasta\n",
      "2026-01-16 12:07:53,046 - INFO - Processing file: reduced_90_function.fasta\n",
      "Found 150860 sequences in reduced_90_function.fasta\n",
      "2026-01-16 12:07:59,058 - INFO - Found 150860 sequences in reduced_90_function.fasta\n",
      "Generating embeddings... (this may take a while)\n",
      "2026-01-16 12:07:59,059 - INFO - Generating embeddings...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 23%|       | 35207/150860 [1:31:09<4:30:08,  7.14it/s] "
     ]
    }
   ],
   "source": [
    "%run ../generate_embeddings/generate_embeddings.py \\\n",
    "    --input ./reduced_90_function.fasta \\\n",
    "    --output ./embeddings/ \\\n",
    "    --output_format npz --tm_vec_model /shared/projects/deepmar/data/tmvec_model_weights/tm_vec_cath_model.ckpt --device cuda --tm_vec_config /shared/projects/deepmar/data/tmvec_model_weights/tm_vec_cath_model_params.json"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [],
   "source": [
    "os.chdir(\"..\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [],
   "source": [
    "function_all_np = np.load(\"embeddings/reduced_90_function_embeddings.npy\", allow_pickle=True)\n",
    "function_all_df = pd.read_csv(\"reduced_90_function.tsv\", sep='\\t')\n",
    "assert function_all_np.shape[0] == function_all_df.shape[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [],
   "source": [
    "function_train_df= pd.read_csv(\"processed_data_90_30/function_train.tsv\", sep=\"\\t\")\n",
    "function_val_df.to_csv(\"processed_data_90_30/function_val.tsv\", sep=\"\\t\")\n",
    "function_test_df.to_csv(\"processed_data_90_30/function_test.tsv\", sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [],
   "source": [
    "function_all_dict = dict(zip(function_all_df['Entry'], function_all_np))\n",
    "function_train_np = np.array([function_all_dict[entry] for entry in function_train_df['Entry']])\n",
    "function_val_np = np.array([function_all_dict[entry] for entry in function_val_df['Entry']])\n",
    "function_test_np = np.array([function_all_dict[entry] for entry in function_test_df['Entry']])\n",
    "\n",
    "\n",
    "assert function_train_np.shape[0] == function_train_df.shape[0]\n",
    "assert function_val_np.shape[0] == function_val_df.shape[0]\n",
    "assert function_test_np.shape[0] == function_test_df.shape[0]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [],
   "source": [
    "np.save(\"processed_data_90_30/function_train.npy\", function_train_np)\n",
    "np.save(\"processed_data_90_30/function_val.npy\", function_val_np)\n",
    "np.save(\"processed_data_90_30/function_test.npy\", function_test_np)\n",
    "\n",
    "\n",
    "with open(\"expected_shapes.txt\", \"w\") as f:\n",
    "    print(\"Function train:\", function_train_np.shape, file=f)\n",
    "    print(\"Function val:\", function_val_np.shape, file=f)\n",
    "    print(\"Function test:\", function_test_np.shape, file=f)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Generate IA TSV using the training data "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [],
   "source": [
    "total_bacteria_function_GO = pd.read_csv(\"reduced_90_function_propagated.tsv\", sep=\"\\t\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Function Obselete: 22\n"
     ]
    }
   ],
   "source": [
    "with open('function_data_json_dict.json') as json_data:\n",
    "    function_all_json_dict = json.load(json_data)\n",
    "\n",
    "function_OBSELETE = [term for term in function_all_json_dict if function_all_json_dict[term][\"is_obsolete\"]]\n",
    "\n",
    "print(\"Function Obselete:\", len(function_OBSELETE))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [],
   "source": [
    "function_train_df_subset = total_bacteria_function_GO[['Entry', 'Raw propagated GO terms']].copy()\n",
    "function_train_df_subset.columns = ['EntryID', 'term']\n",
    "\n",
    "# Adding aspect to each respective dataframe\n",
    "function_train_df_subset['aspect'] = 'MFO'\n",
    "\n",
    "# Convert stringified list  real list\n",
    "function_train_df_subset['term'] = (\n",
    "    function_train_df_subset['term']\n",
    "        .str.strip('[]')\n",
    "        .str.replace(\"'\", \"\", regex=False)\n",
    "        .str.split(', ')\n",
    ")\n",
    "exploded_df = function_train_df_subset.explode('term')\n",
    "\n",
    "\n",
    "#REMOVE ALL_OBSELETE_TERMS\n",
    "exploded_df = exploded_df[~exploded_df['term'].isin(function_OBSELETE)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>EntryID</th>\n",
       "      <th>term</th>\n",
       "      <th>aspect</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A0A009IHW8</td>\n",
       "      <td>GO:0016798</td>\n",
       "      <td>MFO</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A0A009IHW8</td>\n",
       "      <td>GO:0061809</td>\n",
       "      <td>MFO</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A0A009IHW8</td>\n",
       "      <td>GO:0050135</td>\n",
       "      <td>MFO</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A0A009IHW8</td>\n",
       "      <td>GO:0016799</td>\n",
       "      <td>MFO</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A0A009IHW8</td>\n",
       "      <td>GO:0016787</td>\n",
       "      <td>MFO</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      EntryID        term aspect\n",
       "0  A0A009IHW8  GO:0016798    MFO\n",
       "0  A0A009IHW8  GO:0061809    MFO\n",
       "0  A0A009IHW8  GO:0050135    MFO\n",
       "0  A0A009IHW8  GO:0016799    MFO\n",
       "0  A0A009IHW8  GO:0016787    MFO"
      ]
     },
     "execution_count": 76,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "exploded_df.head()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [],
   "source": [
    "exploded_df.to_csv(\"terms_for_IA_calculation.tsv\", sep=\"\\t\", index=False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Counting Terms\n",
      "0 []\n",
      "0 []\n",
      "0 []\n",
      "Computing Information Accretion\n",
      "Saving to file IA_all.tsv\n"
     ]
    }
   ],
   "source": [
    "%run ia.py -a terms_for_IA_calculation.tsv -o IA_all.tsv -g go-basic.obo"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "PlasmoFP",
   "language": "python",
   "name": "hmelcher"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
